# Classification of cells into types

Given that a core goal of the Human Cell Atlas is to advance knowledge about _all_ human cell types, [@wikidata:Q46368626] the definition of "cell type" becomes essential. 
Although a number of views exist [@wikidata:Q87649649; @doi:10.1242/dev.169854; @doi:10.1242/dev.178996; @wikidata:Q47565008; @wikidata:Q99418657; @wikidata:Q98665291; @wikidata:Q93086971; @wikidata:Q26770736; @wikidata:Q35688096; @wikidata:Q98633613; @wikidata:Q99418649; @wikidata:Q40436539; @wikidata:Q108723646], there is no formal, commonly agreed upon defintion of cell type. 
A 2017 article on the Human Cell Atlas mentions[@wikidata:Q47565008]: 

"Descriptors such as 'cell type' and 'cell state' can be difficult to define at the moment. An integrative, systematic effort by many teams of scientists working together and bringing different expertise to the problem could dramatically sharpen our terminology, and revolutionize the way we see our cells, tissues and organs. We invite you to join the effort." 

The article further highlights both the current gap in knowledge and the need for a community effort to work in finding definitions. 

One consequence of a lack of a definition is that there is no commonly agreed number of cell types, and not even on the order of magnitude. 
As of November 2021, the leading answers in the Google Search Engine for the question "How many different cell types are found in the human body?" all point to around 200 different types [@url:https://askabiologist.asu.edu/questions/human-cell-types, https://www.researchgate.net/post/How-many-cell-types-in-a-human-body-How-about-the-number-of-cell-cycles-in-each-species] [@url:https://www.kenhub.com/en/library/anatomy/types-of-cells-in-the-human-body], an estimate that is agreed upon by Bionumbers, a database of useful biological numbers [@wikidata:Q24643881] [@url:https://bionumbers.hms.harvard.edu/bionumber.aspx?id=103626].
A list of cell types in the adult human body on Wikipedia also amounts to around a couple hundred cell types [@url:https://en.wikipedia.org/w/index.php?title=List_of_distinct_cell_types_in_the_adult_human_body&oldid=1044853788].
However, the Cell Ontology has catalogued more than 2,311 cell types of interest for the Human Cell Atlas [@wikidata:Q107373831], increasing the estimate by at least one order of magnitude. 
Additionally, with an estimate of 37 trillion cells on average per human body [@wikidata:Q34037445] and an ever-increasing report of new cell types/clusters in single-cell transcriptomics [@wikidata:Q103034964], it is reasonable that the number of relevant cell types is much larger.
The Human Cell Atlas project itself does not commit to any estimates of numbers of cell types due to the sheer difficulty of estimating a number given current knowledge. (Aviv Regev; reply to a question in the general HCA conference) 

<!-- Adapted from https://github.com/lubianat/technotype/blob/9f428cbcf8e8a8d2843faac684275fb0c37914ed/content/02.introduction.md BEGIN-->

Even though there is no agreement, different views on cell types are maturing. 
One core line of thought to define "cell type" is based on the cell type as an evolutionary unit. 
That definition enables the drawing of parallels, from the evolution of other biological entities (such as genes, proteins, and species) to cell types' evolution. 
Models of how multicellular life works greatly benefit from concepts as sister types (sharing a single ancestor), cell type homology ( sharing a common evolutionary origin), and cell type convergence (executing similar functions without direct evolutionary links) [@pmid:18927580; @doi:10.1038/nrg.2016.127]

Another school of thought is based on attractors: regions of dynamical stability in a feature space. [@wikidata:Q37475461; @wikidata:Q91908361] 
In this theory, "basins of attraction" direct cell phenotypes, providing points in, say, a gene expression space towards which different cells "move" their expression programs. 
This dynamic view sees each cell type corresponding to "a self-stabilizing regulatory program, which acts to maintain and restore the cell type-specific program of gene expression." [@wikidata:Q90361677] 
It aligns itself with dynamic systems theory, and some authors go as far as to say that "Lacking the idea of attractors we have no clear idea of what a cell type is." [@wikidata:Q91316993]

As much as different species concepts coexist [@doi:10.1080/10635150701701083], the quest to define cell types may take various forms.
The challenge of representing cell types in evolution is conceptually different from representing cell types in biomedical experimentation. 
In that second direction, the groundwork of the Cell Ontology [@wikidata:Q21184168; @doi:10.1186/1471-2105-12-6; @doi:10.1186/s13326-016-0088-7] and CELDA [@wikidata:Q21284308] and the contributions of the International Workshop on Cells in Experimental Life Sciences series [@doi:10.1186/s12859-017-1976-2; @doi:10.1186/s12859-019-2721-9] are notable. 

<!-- Text from https://github.com/lubianat/technotype/blob/9f428cbcf8e8a8d2843faac684275fb0c37914ed/content/02.introduction.md END-->

## Single-cell transcriptomics

Even though many sources of knowledge contribute to our understanding of cell types [@wikidata:Q109717199], arguably single-cell transcriptomics is the workhorse for current efforts of the HUman cell Atlas, with an increasing amount of published studies. [@wikidata:Q109717199]
Current scRNA-seq data analyses often rely on unsupervised clustering of cells followed by the assignment of cell-type labels to clusters.
For the clustering, bioinformaticians tailor parameter sets to a target resolution, i.e., the level of detail used to detect cell identities. [@wikidata:Q64974172] [@wikidata:Q46368626]
 When the clustering is finished, the groups of cells are annotated with class labels, representing the underlying biology in a language we can understand. [@wikidata:Q107158224]


Instead of assigning expression gates from pre-defined markers, as is the standard for flow-cytometry analysis, single-cell RNA-seq analysis pipelines usually start from _de novo_ clustering of cells followed by cluster annotation. [@wikidata:Q64974172]
While it is clear that clusters and cell types are different concepts [@wikidata:Q64974172], often cluster labels are treated as cell types. 
There are several ways to cluster cells to find groups of similarity, but arguably the current default is derived from the methodology proposed by PhenoGraph. [@wikidata:Q30975629]
The protocol is to calculate the distances between cells in a reduced PCA space (with the number of dimensions chosen by the experimenters), followed by constructing a k-nearest-neighbours network. 
Each cell is a node connected by _k_ (another parameter) edges to other cells. 
Once the network is built, network modules (i.e. cell clusters) are commonly found using the Louvain algorithm, published in 2008 by researchers of the Université Catholique de Louvain in Belgium. [@wikidata:Q29305711]
The cell clusters found by the PhenoGraph (or any other) algorithm are then labelled by domain experts, often based on genes differentially expressed on each cluster, so-called "markers". [@wikidata:Q64974172]

While it is possible to manually investigate the identities of which clusters, automatic methods have been developed to aid in the task. [@wikidata:Q107158224]
One approach ("marker-based automatic annotation") bases itself on crossing clusters markers in the analyzed dataset with previous knowledge from databases like PanglaoDB [@wikidata:Q63664483] and CellMarker [@wikidata:Q56984510] [@wikidata:Q107158224].
Another approach (reference-based automatic cell annotation) relies on base, expert-annotated datasets as references from which labels are transferred to the dataset of interest. [@wikidata:Q107158224] 
Other methods bypass the clustering step and focus on labelling the individual cells, which avoids lumping different cells together but requires a high amount of reads per individual cells to be efficient. [@wikidata:Q107158224] 
Clarke et al.'s recent review and tutorial [@wikidata:Q107158224] provides an extensive account of current techniques. 

Of note, even though a range of methods is available, most techniques and publications do not use standard identifiers for cell types. 
This is in contradiction with the acknowledgement by the community of the advantages of using identifiers the ad using standard identifiers, such as those provided by the Cell Ontology. [@wikidata:Q107158224] [@wikidata:Q33037209] [@wikidata:Q107373831] [@wikidata:Q36067763] [@wikidata:Q52625486]  [@wikidata:Q49192555].
Nevertheless, projects that use Cell Ontology identifiers for single-cell RNA-seq data are appearing [@wikidata:Q109755180], including python and R packages (e.g. Besca [@wikidata:Q104450593], OnClass [@wikidata:Q108929315] and ontoProc[@wikidata:Q101074371]), data management projects and reference datasets, (e.g. Tabula Muris [@url:https://tabula-muris.ds.czbiohub.org/] and Tabula Sapiens [@url:https://tabula-sapiens-portal.ds.czbiohub.org/celltypes] Azimuth map [@url:https://azimuth.hubmapconsortium.org/] and HubMap's ASCT+B Tables [@wikidata:Q109755184]) and annotation platforms (e.g. the Cell Annotation Platform [@url:http://celltype.info] and CellTypist [@wikidata:Q107363182].

As elegantly put by Meehan et al. [@wikidata:Q34978215] the Cell Ontology is a "manually constructed computer-readable resource that links cell types by different relationships".
it was first described in 2005 by Jonathan Bard, Seung Y Rhee† and Michael Ashburner [@wikidata:Q21184168] and was oriented at creating an "organism-independent classification of cells".
It followed criteria that included function, histology, lineage and ploidy and providing "Cell-type unique identifiers (ID) that can be incorporated into any database holding cell-type-associated knowledge." 
It also had a didactic goal in itself, as the authors mention [@wikidata:Q21184168]: 
"It is designed to be useful in the sense that a researcher should be able to find, in a rapid and intuitive way, any cell type in any of the major model organisms and, having found it, learn a considerable amount about that cell type and its relationships to other biological objects."
The collaborative project gradually evolved and changed its design and scope to fit new needs. 
By 2011, a need for computable definitions led to an advance in the number and quality of immune cell types represented in CL. [@wikidata:Q33786317] 
It also included the addition of species-specific cell types to handle better marker-based definitions, usually presented at the species level. [@wikidata:Q33786317] 
Further developments over the years included both technical improvements and the addition of new cell types. 
By the time of the last official CL publication, in 2016, it contained approximately 2,200 classes.   [@wikidata:Q36067763]  
Currently, the Cell Ontology is growing as a resource for the Human Cell Atlas and in providing identifiers for cell types [@wikidata:Q109755180] and presents over 2500 classes of cells (Figure @fig:cl_counts)

![Cell Ontology counts per release. Some releases around 2014 are not available anymore, thus yielding 0 counts.](https://github.com/lubianat/cell_ontology_count/blob/master/results/classes_per_release.png?raw=true){#fig:cl_counts width="85 %"}

While purely theoretical developments have their value, refining the cell type theory in the context of knowledge management arguably will directly influence how the products of the Human Cell Atlas will impact modern science. 
One reason is that formal systems enable automation of knowledge integration and can feed intelligent systems that aid current research practices. 
In the following chapter, it will be discussed how computer-based knowledge processing can influence life-sciences research and discuss techniques and platforms to advance the frontier. 