## Methodology {.page_break_before}

This project's methodology resembles practical research-action practices [@wikidata:Q108479295]. 
The "action" facet is done by contributing to projects in the Human Cell Atlas and knowledge management context. 

Research in 3 forms:
    - Philosophical investigation on knowledge representations of cell types, both in formal logic and in academic literature
    - Applied investigations of database integration and data quality in the context of Wikidata and biomedical ontologies
    - Data-driven biomedical research targeted at hypothesis generation and literature-based discovery using knowledge at the level of cell-type

All research branches are linked to the improvement of knowledge management in biomedical sciences, focusing on the Human Cell Atlas. 
The methods included the development and application of a framework for an organized reading of the scientific literature, providing contact with the different facets of biocuration and Human Cell Atlas-related research. 

## Organized reading

As much of the project is based directly on published research, we developed a reading framework, described in detail in the results section. 
The framework is based on GitHub and includes Python scripts, a file organizing the reading list, and another documenting the reading history in RDF.
Notes and additional information are saved in a GitHub repository, and the structured information powers a live website with analytics on the users recent readings. 
The source code for Wikidata Bib is available at <https://github.com/lubianat/wikidata_bib/tree/template> and notes on my readings can currently be accessed at <https://lubianat.github.io/wikidata_bib/>.

Additionally, the methodology included a discipline of reading that entails the daily task of reading 2 articles, one about "cell types" and another about "biocuration". 
The articles are obtained by a mixed manual and automatic approach, including a la carte selection of articles to read alongside Wikidata queries for Cell, Nature, Science and eLife papers about single-cell transcriptomics (query: <https://w.wiki/4LHr>) and papers on biocuration (query: <https://w.wiki/4LHi>). 

### Biocuration of cell classes for Wikidata

For each article about cell types read, cell types previously absent on Wikidata are added via a combination of curation in a Google Spreadsheet and a custom Python script (https://github.com/lubianat/wikidata_markers/tree/master/curation_of_classes).  

## Wikidata updates

We have updated the Wikidata database with new entities, relations and triples. 
Creation of new entities was done  in the Graphical User Interface (<https://www.wikidata.org/wiki/Special:NewItem>), via the batch Quickstatements tool (<https://quickstatements.toolforge.org/#/>) and the Wikidata Integrator python library (<https://github.com/SuLab/WikidataIntegrator>). 

Properties, which link items to values need community approval. Under the scope of this PhD project, we have gotten the community approval for the properties : 
Cell Ontology ID (<https://www.wikidata.org/wiki/Property:P7963>) used to link cell types to their IDs in the Cell Ontology and has marker (<https://www.wikidata.org/wiki/Property:P8872>) used to link cell types to genes and proteins considered their markers.

The property acceptance cycle takes at least one week and is  open for opinions by any Wikidata user.
All the information regarding property proposals is available at <https://www.wikidata.org/wiki/Wikidata:Property_proposal>.

## Cell Ontology participation

As part of the research-action process, I have joined the Cell Ontology working group. 
I participate in the monthly meetings and sporadic workshops, learning and contributing to the discussions. 
Additionally, I contribute to the ontology development, actively engaging in the Cell Ontology GitHub repository (<https://github.com/obophenotype/cell-ontology>) and contributing with new terms and assertions. 
I edit the ontology with the software for ontology editing Protégé v. 5.5.0 [@url:https://protege.stanford.edu].

### Status of cell type info on Wikidata and the Cell Ontology

Status of cell type information on Wikidata was accessed via SPARQL queries combined with processing in python and is available at <https://colab.research.google.com/drive/1GvQXOs51_U8icdGMtKXMeLOXKM8pXWet#scrollTo=szvBWI9zr_AA>.

Counts of cell classes in the Cell Ontology were performed via regex matching on Cell Ontology releases following the code available at <https://github.com/lubianat/cell_ontology_count>. 
