<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Tiago Lubiana" />
  <meta name="dcterms.date" content="2022-01-04" />
  <meta name="keywords" content="Wikidata, Human Cell Atlas, ontology, Cell Ontology" />
  <title>Building a biological knowledge graph via Wikidata with a focus on the Human Cell Atlas</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      {   }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span.al { color: #ff0000; font-weight: bold; } /* Alert */
    code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
    code span.at { color: #7d9029; } /* Attribute */
    code span.bn { color: #40a070; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #4070a0; } /* Char */
    code span.cn { color: #880000; } /* Constant */
    code span.co { color: #60a0b0; font-style: italic; } /* Comment */
    code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
    code span.do { color: #ba2121; font-style: italic; } /* Documentation */
    code span.dt { color: #902000; } /* DataType */
    code span.dv { color: #40a070; } /* DecVal */
    code span.er { color: #ff0000; font-weight: bold; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #40a070; } /* Float */
    code span.fu { color: #06287e; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
    code span.kw { color: #007020; font-weight: bold; } /* Keyword */
    code span.op { color: #666666; } /* Operator */
    code span.ot { color: #007020; } /* Other */
    code span.pp { color: #bc7a00; } /* Preprocessor */
    code span.sc { color: #4070a0; } /* SpecialChar */
    code span.ss { color: #bb6688; } /* SpecialString */
    code span.st { color: #4070a0; } /* String */
    code span.va { color: #19177c; } /* Variable */
    code span.vs { color: #4070a0; } /* VerbatimString */
    code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Building a biological knowledge graph via Wikidata with a focus on the Human Cell Atlas" />
  <meta name="citation_title" content="Building a biological knowledge graph via Wikidata with a focus on the Human Cell Atlas" />
  <meta property="og:title" content="Building a biological knowledge graph via Wikidata with a focus on the Human Cell Atlas" />
  <meta property="twitter:title" content="Building a biological knowledge graph via Wikidata with a focus on the Human Cell Atlas" />
  <meta name="dc.date" content="2022-01-04" />
  <meta name="citation_publication_date" content="2022-01-04" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Tiago Lubiana" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of São Paulo" />
  <meta name="citation_author_institution" content="Ronin Institute" />
  <meta name="citation_author_orcid" content="0000-0003-2473-2313" />
  <meta name="twitter:creator" content="@lubianat" />
  <link rel="canonical" href="https://lubianat.github.io/quali_phd/" />
  <meta property="og:url" content="https://lubianat.github.io/quali_phd/" />
  <meta property="twitter:url" content="https://lubianat.github.io/quali_phd/" />
  <meta name="citation_fulltext_html_url" content="https://lubianat.github.io/quali_phd/" />
  <meta name="citation_pdf_url" content="https://lubianat.github.io/quali_phd/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://lubianat.github.io/quali_phd/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://lubianat.github.io/quali_phd/v/f962072fcd0685f369693089c4c2d0b96e523fd1/" />
  <meta name="manubot_html_url_versioned" content="https://lubianat.github.io/quali_phd/v/f962072fcd0685f369693089c4c2d0b96e523fd1/" />
  <meta name="manubot_pdf_url_versioned" content="https://lubianat.github.io/quali_phd/v/f962072fcd0685f369693089c4c2d0b96e523fd1/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Building a biological knowledge graph via Wikidata with a focus on the Human Cell Atlas</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://lubianat.github.io/quali_phd/v/f962072fcd0685f369693089c4c2d0b96e523fd1/">permalink</a>)
was automatically generated
from <a href="https://github.com/lubianat/quali_phd/tree/f962072fcd0685f369693089c4c2d0b96e523fd1">lubianat/quali_phd@f962072</a>
on January 4, 2022.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><strong>Tiago Lubiana</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-2473-2313">0000-0003-2473-2313</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/lubianat">lubianat</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/lubianat">lubianat</a><br>
<small>
School of Pharmaceutical Sciences, University of São Paulo; Ronin Institute
· Funded by Grant #2019/26284-1 from the São Paulo Research Foundation (FAPESP).
</small></li>
</ul>
<h1 id="abstract">Abstract</h1>
<p>The Human Cell Atlas (HCA) is an international effort to characterize every cell type of the human body.
HCA members are producing cell-level data from virtually all human tissue using techniques such as single-cell RNA sequencing, mass cytometry, and multiplexed in situ hybridizations.
The wealth knowledge stemming from the wealth of data needs management to maximize the societal benefit of HCA.
While semantic technologies have emerged as key players in the data interoperability ecosystem, there are still gaps to bridge.
Wikidata, a sister project of Wikipedia for structured data, is surfacing as a hub in the semantic web for multiple types of information and bridging experts and the people.
The connection of the Human Cell Atlas to Wikidata can quickly insert its products into the larger knowledge ecosystem, extending HCA’s reach.
This PhD project aims at studying Wikidata as a platform for representing cell types.<br />
We review the literature on cell types, refining and formalizing concepts for cell type delimitation.
At the same time, integrate biomedical databases (e.g. PanglaoDB) into the Wikidata infrastructure and we enrich Wikidata with manually curated cell types.
To optimize the curation, we develop Wikidata Bib, a framework for literature management and organized note-taking system for dealing with the academic literature.
Finally, we are improving the interplay of Wikidata, the Cell Ontology and software used for single-cell RNA-seq data, inserting Wikidata <em>de facto</em> as a tool for the Human Cell Atlas community.</p>
<h2 class="page_break_before" id="preface">Preface</h2>
<p>In this preface we present an overview of the chapters in this document.</p>
<p>The introduction contains an overview of the Human Cell Atlas project and the current state of classifying cells into types.
Then, it introduces ontologies and knowledge graphs as tools for connecting what we know about cells.</p>
<p>The methodology section is an overview of the core methods used throughout the work.
Of note, some chapters in the results section also display methods when they are intimately linked with the results.</p>
<p>It is worth noticing that the different results shown were not developed chronologically in the order shown.
The different branches were developed in parallel, with overlapping periods of activity.
They are organized into separate chapters and part of different publications.</p>
<p>The discussion on the concept of cell type is presented first, as it is instrumental for the later steps.
It is followed by an account of how PanglaoDB, a database of cell markers, was integrated into Wikidata.
Then, we present Wikidata Bib, a framework for an organized reading and curation of cell types.
The framework, although used as a method throughout the PhD project, is also an intelectual product, thus presented in the results session.
To end the results, we discuss how the work integrates with the Cell Ontology, the leading system for organizing cell types.</p>
<p>After the results, an account of other academic aspects of the project is presented as part of the qualification requirements.
They present an overview of collaborations, participation in events and academic courses taken during the first part of the PhD project.</p>
<h2 class="page_break_before" id="background">Background</h2>
<!-- 1.4. The challenges of the Human Cell Atlas -->
<h2 id="the-human-cell-atlas-hca-project">The Human Cell Atlas (HCA) Project</h2>
<!-- 1.4.1. The Human Cell Atlas project and its scope -->
<!-- 1.4.1.1. Participants -->
<p>The Human Cell Atlas (HCA) project is, arguably, the biggest multinational biomedical project of the present time, running since 2017 to characterize every cell type in the human body <span class="citation" data-cites="1GmbExweg">[<a href="#ref-1GmbExweg" role="doc-biblioref">1</a>]</span>.
The HCA consortium gathers people from all over the world to tackle different parts of the project to have a diverse and equitable account of the cell type diversity. <span class="citation" data-cites="tjdjR2Xf">[<a href="#ref-tjdjR2Xf" role="doc-biblioref">2</a>]</span><br />
It intends to “ultimately describe at least 10 billion cells, covering all tissues, organs, and systems” and, to achieve such bold goals, it commits to “to open membership, to the open and immediate data release with no restrictions, and to open-source code for all computational approaches”. <span class="citation" data-cites="1GmbExweg">[<a href="#ref-1GmbExweg" role="doc-biblioref">1</a>]</span></p>
<!-- - 1.4.1.2. Overview of main analytical techniques  -->
<p>Building a complete atlas of human cells comes with multiple challenges.
The project includes the detection, in single cells, of RNA species (scRNA-Seq), chromatin accessibility (scATAC-Seq), and protein markers (primarily by CYTOF), as well as spatial information on cells with multiplexed <em>in situ</em> hybridization (such as MERFISH) and imaging mass cytometry <span class="citation" data-cites="1GmbExweg kkwRTArg">[<a href="#ref-1GmbExweg" role="doc-biblioref">1</a>,<a href="#ref-kkwRTArg" role="doc-biblioref">3</a>]</span>.
The HCA is set to revolutionize the biomedical sciences by creating tools and standards for basic research, allowing better characterization of disease, and improving diagnostics and therapy.
Its products (data, information, knowledge and wisdom) need to be FAIR: findable, accessible, interoperable and reusable.
Data stewardship and management are growing as core demands of the scientific community, ranging from data management plans <span class="citation" data-cites="1DSEIjFha">[<a href="#ref-1DSEIjFha" role="doc-biblioref">4</a>]</span> to specialized data personnel <span class="citation" data-cites="1DSEIjFha">[<a href="#ref-1DSEIjFha" role="doc-biblioref">4</a>]</span>.</p>
<!-- 1.4.3. Data availability -->
<!-- - 1.4.3.1. As coordinated by the Human Cell Atlas -->
<p>The Human Cell Atlas has a dedicated team for organizing data: the Data Coordination Platform (DCP) <span class="citation" data-cites="zDRzmIGu">[<a href="#ref-zDRzmIGu" role="doc-biblioref">5</a>]</span> <span class="citation" data-cites="kkwRTArg">[<a href="#ref-kkwRTArg" role="doc-biblioref">3</a>]</span>.
The DCP is responsible for tracing the plan for computational interoperability, from the data generators to the consumers.<span class="citation" data-cites="kkwRTArg">[<a href="#ref-kkwRTArg" role="doc-biblioref">3</a>]</span>.
The Human Cell Atlas has its portal for data <span class="citation" data-cites="kX6KnbUo">[<a href="#ref-kX6KnbUo" role="doc-biblioref">6</a>]</span>), which composes the data repository landscape with other resources, like the Broad Institute Single Cell Portal [https://singlecell.broadinstitute.org/single_cell&gt;) and the Chan-Zuckerberg Biohub Tabula Sapiens (<a href="https://tabula-sapiens-portal.ds.czbiohub.org/" class="uri">https://tabula-sapiens-portal.ds.czbiohub.org/</a>).
In addition to its core team, the HCA is poised to grow by community interaction.
As stated in its opening publication “As with the Human Genome Project, a robust plan will best emerge from wide-ranging scientific discussions and careful planning”.<span class="citation" data-cites="1GmbExweg">[<a href="#ref-1GmbExweg" role="doc-biblioref">1</a>]</span><br />
Thus, this project inserts itself among the wide-ranging scientific discussions to improve data - and knowledge - interoperability.</p>
<p>The highlight of “knowledge” in the last paragraph is meant to stress that raw data <em>per se</em> is not enough to turn the Atlas objectives into reality.
There is a long way from raw datasets to commonly-agreed scientific knowledge.
Currently, the gap between data and knowledge is attempted by writing and sharing scientific manuscripts, the <em>de facto</em> currency of exchange of claims about the natural world.
A Human Cell Atlas Publication Committee reviews and selects publications, <span class="citation" data-cites="2A7P3sTH">[<a href="#ref-2A7P3sTH" role="doc-biblioref">7</a>]</span> presenting the papers as one of the significant outputs of the whole endeavour.
As of december 2021, the list includes 96 different publications, which, arguably, expose only a fraction of the knowledge extractable from the underlying data.</p>
<p>The challenge that arises is one of managing a wealth of information and casting it into useful knowledge.
Ideally, we would like to understand, remember, and use every statement produced by the HCA.
As this goal is humanely impossible, we need to develop tools to make the knowledge interoperable with the aid of computers.
At that point, the challenges of the HCA enter in resonance with the challenges of biocuration, and with the goals of this project.</p>
<p>One core step for knowledge management is the minting of identifiers for the concepts of interest. <span class="citation" data-cites="tvnuubVW">[<a href="#ref-tvnuubVW" role="doc-biblioref">8</a>]</span>
The minting of identifiers, however, depend on clear, well defined entities.
The sheer notion of “cell type” is undefined <span class="citation" data-cites="1GmbExweg">[<a href="#ref-1GmbExweg" role="doc-biblioref">1</a>]</span>, a challenge for organizing both data and knowledge.
Accordingly, this PhD project concerns itself also with the theoretical basis of defining a “cell type”.
Thus, the next chapter will introduce the state-of-the-art of classifying cell into types, in a preparation for the chapter on knowledge modelling and the theoretical discussions of the results section.</p>
<h1 id="classification-of-cells-into-types">Classification of cells into types</h1>
<p>Given that a core goal of the Human Cell Atlas is to advance knowledge about <em>all</em> human cell types, <span class="citation" data-cites="1GmbExweg">[<a href="#ref-1GmbExweg" role="doc-biblioref">1</a>]</span> the definition of “cell type” becomes essential.
Although a number of views exist <span class="citation" data-cites="paKiKTRS sWv3okaH 19yYJWEkz BlxFp1v3 eOUB1Bmj TupzTvtA w0Woz7JE srfuRMe 1GXQByEaQ k0Pe9YOB pNGap1Du MoaYZAA0 WAeCfntu">[<a href="#ref-paKiKTRS" role="doc-biblioref">9</a>,<a href="#ref-sWv3okaH" role="doc-biblioref">10</a>,<a href="#ref-19yYJWEkz" role="doc-biblioref">11</a>,<a href="#ref-BlxFp1v3" role="doc-biblioref">12</a>,<a href="#ref-eOUB1Bmj" role="doc-biblioref">13</a>,<a href="#ref-TupzTvtA" role="doc-biblioref">14</a>,<a href="#ref-w0Woz7JE" role="doc-biblioref">15</a>,<a href="#ref-srfuRMe" role="doc-biblioref">16</a>,<a href="#ref-1GXQByEaQ" role="doc-biblioref">17</a>,<a href="#ref-k0Pe9YOB" role="doc-biblioref">18</a>,<a href="#ref-pNGap1Du" role="doc-biblioref">19</a>,<a href="#ref-MoaYZAA0" role="doc-biblioref">20</a>,<a href="#ref-WAeCfntu" role="doc-biblioref">21</a>]</span>, there is no formal, commonly agreed upon defintion of cell type.</p>
<p>A 2017 article on the Human Cell Atlas mentions<span class="citation" data-cites="BlxFp1v3">[<a href="#ref-BlxFp1v3" role="doc-biblioref">12</a>]</span>:
“Descriptors such as ‘cell type’ and ‘cell state’ can be difficult to define at the moment. An integrative, systematic effort by many teams of scientists working together and bringing different expertise to the problem could dramatically sharpen our terminology, and revolutionize the way we see our cells, tissues and organs. We invite you to join the effort.”</p>
<p>The article further highlights both the current gap in knowledge and the need for a community effort to work in finding definitions.
The magnitude of the challenge justifies the position of the HCA to avoid attempting to propose a precies definition of ‘cell type’. <span class="citation" data-cites="1GmbExweg">[<a href="#ref-1GmbExweg" role="doc-biblioref">1</a>]</span></p>
<p>One consequence of a lack of a definition is that there is no commonly agreed number of cell types, and not even on the order of magnitude.
As of November 2021, the leading answers in the Google Search Engine for the question “How many different cell types are found in the human body?” all point to around 200 different types <span class="citation" data-cites="V85zvNGa SKmDXJlO WBDY0Uoy">[<a href="#ref-V85zvNGa" role="doc-biblioref">22</a>,<a href="#ref-SKmDXJlO" role="doc-biblioref">23</a>,<a href="#ref-WBDY0Uoy" role="doc-biblioref">24</a>]</span>, an estimate that is agreed upon by Bionumbers, a database of useful biological numbers <span class="citation" data-cites="SprCx2w6">[<a href="#ref-SprCx2w6" role="doc-biblioref">25</a>]</span> <span class="citation" data-cites="jePJgR0H">[<a href="#ref-jePJgR0H" role="doc-biblioref">26</a>]</span>.
A list of cell types in the adult human body on Wikipedia also amounts to around a couple hundred cell types <span class="citation" data-cites="url">[<a href="#ref-url" role="doc-biblioref"><strong>url?</strong></a>]</span>.</p>
<p>If we look at other sources, however, the estimates increase considerably.
For example, the Cell Ontology has catalogued more than 2,311 cell types of interest for the Human Cell Atlas <span class="citation" data-cites="J5X2Xu6M">[<a href="#ref-J5X2Xu6M" role="doc-biblioref">27</a>]</span>, increasing the estimate by at least one order of magnitude.
Across the literature, we can find mentions of thousands to tens of thousands of cell types only in the brain. <span class="citation" data-cites="15efKvxi0 wvN4dtjQ">[<a href="#ref-15efKvxi0" role="doc-biblioref">28</a>,<a href="#ref-wvN4dtjQ" role="doc-biblioref">29</a>]</span>
Additionally, with an estimate of 37 trillion cells on average per human body <span class="citation" data-cites="xWhvkd3B">[<a href="#ref-xWhvkd3B" role="doc-biblioref">30</a>]</span> and an ever-increasing report of new cell types/clusters in single-cell transcriptomics <span class="citation" data-cites="HFiosSKK">[<a href="#ref-HFiosSKK" role="doc-biblioref">31</a>]</span>, it is reasonable to assume the number of relevant cell types might be much larger.
The Human Cell Atlas project itself does not commit to any estimates of numbers of cell types due to the sheer difficulty of estimating a number given current knowledge. (Aviv Regev; reply to a question in the HCA General Meeting conference <span class="citation" data-cites="url">[<a href="#ref-url" role="doc-biblioref"><strong>url?</strong></a>]</span>)</p>
<!-- Adapted from https://github.com/lubianat/technotype/blob/9f428cbcf8e8a8d2843faac684275fb0c37914ed/content/02.introduction.md BEGIN-->
<p>Even though there is no commonly agreed definition, different views on cell types are maturing, tailored to different research needs.
One core line of thought to define “cell type” is based on the cell type as an evolutionary unit.
That definition enables the drawing of parallels, from the evolution of other biological entities (such as genes, proteins, and species) to cell types’ evolution.
Models of how multicellular life works greatly benefit from concepts as sister types (sharing a single ancestor), cell type homology (sharing a common evolutionary origin), and cell type convergence (executing similar functions without direct evolutionary links) <span class="citation" data-cites="O95WN4x0 ogAGO9KH">[<a href="#ref-O95WN4x0" role="doc-biblioref">32</a>,<a href="#ref-ogAGO9KH" role="doc-biblioref">33</a>]</span></p>
<p>Another school of thought is based on attractors: regions of dynamical stability in a feature space. <span class="citation" data-cites="14J9lsoMd OjxQDZpx">[<a href="#ref-14J9lsoMd" role="doc-biblioref">34</a>,<a href="#ref-OjxQDZpx" role="doc-biblioref">35</a>]</span>
In this theory, “basins of attraction” direct cell phenotypes, providing points in, say, a gene expression space towards which different cells “move” their expression programs.
This dynamic view sees each cell type corresponding to “a self-stabilizing regulatory program, which acts to maintain and restore the cell type-specific program of gene expression.” <span class="citation" data-cites="19g5Iusf9">[<a href="#ref-19g5Iusf9" role="doc-biblioref">36</a>]</span>
It aligns itself with dynamic systems theory, and some authors go as far as to say that “Lacking the idea of attractors we have no clear idea of what a cell type is.” <span class="citation" data-cites="4TJnL1xT">[<a href="#ref-4TJnL1xT" role="doc-biblioref">37</a>]</span></p>
<p>As much as different species concepts coexist <span class="citation" data-cites="g74dZjOM">[<a href="#ref-g74dZjOM" role="doc-biblioref">38</a>]</span>, the quest to define cell types may take various forms.
The challenge of representing cell types in evolution is conceptually different from representing cell types for identifying different entities on biomedical data.
In that second direction, the groundwork of the Cell Ontology <span class="citation" data-cites="1CI8uS6ka gAysgnsS 17U3pGEjv">[<a href="#ref-1CI8uS6ka" role="doc-biblioref">39</a>,<a href="#ref-gAysgnsS" role="doc-biblioref">40</a>,<a href="#ref-17U3pGEjv" role="doc-biblioref">41</a>]</span> is notable, providing identifiers that are reused across a range of databases. <span class="citation" data-cites="qT8WxqjA">[<a href="#ref-qT8WxqjA" role="doc-biblioref">42</a>]</span>
Theoretical discussions on cell types is also the topic of specialized conferences, namely the International Workshop on Cells in Experimental Life Sciences conference series <span class="citation" data-cites="jX2ZmXcc rK8KD0us">[<a href="#ref-jX2ZmXcc" role="doc-biblioref">43</a>,<a href="#ref-rK8KD0us" role="doc-biblioref">44</a>]</span>.</p>
<!-- Text from https://github.com/lubianat/technotype/blob/9f428cbcf8e8a8d2843faac684275fb0c37914ed/content/02.introduction.md END-->
<h2 id="single-cell-transcriptomics">Single-cell transcriptomics</h2>
<p>Even though many sources of knowledge contribute to our understanding of cell types <span class="citation" data-cites="16phbXNJE">[<a href="#ref-16phbXNJE" role="doc-biblioref">45</a>]</span>, arguably single-cell transcriptomics is the workhorse for current efforts of the Human Cell Atlas. <span class="citation" data-cites="16phbXNJE">[<a href="#ref-16phbXNJE" role="doc-biblioref">45</a>]</span>
Current scRNA-seq data analyses often rely on unsupervised clustering followed by labeling clusters.
For the clustering, bioinformaticians select parameter sets to a target resolution, i.e., the level of detail used to detect cell identities. <span class="citation" data-cites="WKbly37M">[<a href="#ref-WKbly37M" role="doc-biblioref">46</a>]</span> <span class="citation" data-cites="1GmbExweg">[<a href="#ref-1GmbExweg" role="doc-biblioref">1</a>]</span>
When the clustering is finished, the groups of cells are annotated with class labels, representing the underlying biology in a language we can understand. <span class="citation" data-cites="ii1dVqoy">[<a href="#ref-ii1dVqoy" role="doc-biblioref">47</a>]</span></p>
<p>Instead of assigning expression gates, as done for flow-cytometry, single-cell RNA-seq analysis pipelines start from <em>de novo</em> clustering of cells followed by cluster annotation. <span class="citation" data-cites="WKbly37M">[<a href="#ref-WKbly37M" role="doc-biblioref">46</a>]</span>
While it is clear that clusters and cell types are different concepts <span class="citation" data-cites="WKbly37M">[<a href="#ref-WKbly37M" role="doc-biblioref">46</a>]</span>, often cluster labels are treated as cell types.
There are several ways to cluster cells to find groups of similarity, and arguably the current default is derived from the methodology proposed by PhenoGraph. <span class="citation" data-cites="pEJYKNFc">[<a href="#ref-pEJYKNFc" role="doc-biblioref">48</a>]</span>
The protocol is to calculate the distances between cells in a reduced PCA space (with the number of dimensions chosen by the experimenters), followed by constructing a k-nearest-neighbours network.
Each cell is a node connected by <em>k</em> (another parameter) edges to other cells.
Once the network is built, network modules (i.e. cell clusters) are commonly found using the Louvain algorithm, published in 2008 by researchers of the Université Catholique de Louvain in Belgium. <span class="citation" data-cites="JpCsfe9G">[<a href="#ref-JpCsfe9G" role="doc-biblioref">49</a>]</span>
The cell clusters found by the PhenoGraph (or any other) algorithm are then labelled by domain experts, often based on genes called “cluster markers”, differentially expressed on each cluster. <span class="citation" data-cites="WKbly37M">[<a href="#ref-WKbly37M" role="doc-biblioref">46</a>]</span></p>
<p>While it is possible to manually assign labels to clusters, automatic methods have been developed to aid in the task. <span class="citation" data-cites="ii1dVqoy">[<a href="#ref-ii1dVqoy" role="doc-biblioref">47</a>]</span>
Mmarker-based automatic annotation bases itself on crossing clusters markers in the analyzed dataset with previous knowledge from databases like PanglaoDB <span class="citation" data-cites="1GezH4XNn">[<a href="#ref-1GezH4XNn" role="doc-biblioref">50</a>]</span> and CellMarker <span class="citation" data-cites="chGii6yw">[<a href="#ref-chGii6yw" role="doc-biblioref">51</a>]</span> <span class="citation" data-cites="ii1dVqoy">[<a href="#ref-ii1dVqoy" role="doc-biblioref">47</a>]</span>.
Reference-based automatic cell annotation, on the other hand, relies on expert-annotated reference datasets to transfer labels to the dataset under study. <span class="citation" data-cites="ii1dVqoy">[<a href="#ref-ii1dVqoy" role="doc-biblioref">47</a>]</span>
Other methods bypass the clustering step and focus on labelling individual cells, which avoids lumping different cells together. <span class="citation" data-cites="ii1dVqoy">[<a href="#ref-ii1dVqoy" role="doc-biblioref">47</a>]</span>
Clarke et al.’s recent review and tutorial <span class="citation" data-cites="ii1dVqoy">[<a href="#ref-ii1dVqoy" role="doc-biblioref">47</a>]</span> provides an extensive account of current techniques for clustering and annotating cells.</p>
<p>Of note, even though a range of methods is available, most techniques and publications do not use standard identifiers for cell types.
That omission disregards the advantages of using standard identifiers, such as those provided by the Cell Ontology. <span class="citation" data-cites="ii1dVqoy">[<a href="#ref-ii1dVqoy" role="doc-biblioref">47</a>]</span> <span class="citation" data-cites="tvnuubVW">[<a href="#ref-tvnuubVW" role="doc-biblioref">8</a>]</span> <span class="citation" data-cites="J5X2Xu6M">[<a href="#ref-J5X2Xu6M" role="doc-biblioref">27</a>]</span> <span class="citation" data-cites="agnqfdk6">[<a href="#ref-agnqfdk6" role="doc-biblioref">52</a>]</span> <span class="citation" data-cites="SozKx0lF">[<a href="#ref-SozKx0lF" role="doc-biblioref">53</a>]</span> <span class="citation" data-cites="15xnotFsw">[<a href="#ref-15xnotFsw" role="doc-biblioref">54</a>]</span>.
Reassuringly, projects that use stable identifiers for single-cell RNA-seq data are appearing <span class="citation" data-cites="qT8WxqjA">[<a href="#ref-qT8WxqjA" role="doc-biblioref">42</a>]</span>, including python and R packages (e.g. Besca <span class="citation" data-cites="P1gDLuog">[<a href="#ref-P1gDLuog" role="doc-biblioref">55</a>]</span>, OnClass <span class="citation" data-cites="sW6aNZJB">[<a href="#ref-sW6aNZJB" role="doc-biblioref">56</a>]</span> and ontoProc<span class="citation" data-cites="15YmDXALp">[<a href="#ref-15YmDXALp" role="doc-biblioref">57</a>]</span>), data management projects and reference datasets, (e.g. Tabula Muris <span class="citation" data-cites="tdlWsvTd">[<a href="#ref-tdlWsvTd" role="doc-biblioref">58</a>]</span> and Tabula Sapiens <span class="citation" data-cites="FjUqVIOp">[<a href="#ref-FjUqVIOp" role="doc-biblioref">59</a>]</span> Azimuth map <span class="citation" data-cites="T7RxnUNw">[<a href="#ref-T7RxnUNw" role="doc-biblioref">60</a>]</span> and HubMap’s ASCT+B Tables <span class="citation" data-cites="1FghWzORJ">[<a href="#ref-1FghWzORJ" role="doc-biblioref">61</a>]</span>) and annotation platforms (e.g. the Cell Annotation Platform <span class="citation" data-cites="yMavNVkS">[<a href="#ref-yMavNVkS" role="doc-biblioref">62</a>]</span> and CellTypist <span class="citation" data-cites="123DiCgc8">[<a href="#ref-123DiCgc8" role="doc-biblioref">63</a>]</span>).</p>
<p>The Cell Ontology is currently the biggest provider of such standard identifiers.
As elegantly put by Meehan et al. <span class="citation" data-cites="UduMr7x2">[<a href="#ref-UduMr7x2" role="doc-biblioref">64</a>]</span> the Cell Ontology is a “manually constructed computer-readable resource that links cell types by different relationships”.
It was first described in 2005 <span class="citation" data-cites="1CI8uS6ka">[<a href="#ref-1CI8uS6ka" role="doc-biblioref">39</a>]</span> and was oriented both at creating a species-neutral classification of cells and
for researchers to “learn a considerable amount about that cell type and its relationships to other biological objects” <span class="citation" data-cites="1CI8uS6ka">[<a href="#ref-1CI8uS6ka" role="doc-biblioref">39</a>]</span>:</p>
<p>The collaborative project gradually evolved and changed its design and scope to fit new needs.
By 2011, a need for computable definitions led to an advance in the number and quality of immune cell types represented in CL. <span class="citation" data-cites="ecteO8vI">[<a href="#ref-ecteO8vI" role="doc-biblioref">65</a>]</span>
It also included the addition of species-specific cell types to handle better marker-based definitions, usually presented at the species level. <span class="citation" data-cites="ecteO8vI">[<a href="#ref-ecteO8vI" role="doc-biblioref">65</a>]</span>
Further developments over the years included both technical improvements and the addition of new cell types.
By the time of the last official CL publication, in 2016, it contained approximately 2,200 classes. <span class="citation" data-cites="agnqfdk6">[<a href="#ref-agnqfdk6" role="doc-biblioref">52</a>]</span><br />
Currently, the Cell Ontology is growing as a resource for the Human Cell Atlas and in providing identifiers for cell types <span class="citation" data-cites="qT8WxqjA">[<a href="#ref-qT8WxqjA" role="doc-biblioref">42</a>]</span> and presents over 2500 classes of cells (Figure <a href="#fig:cl_counts">1</a>)</p>
<div id="fig:cl_counts" class="fignos">
<figure>
<img src="https://github.com/lubianat/cell_ontology_count/blob/master/results/classes_per_release.png?raw=true" alt="Figure 1: Cell Ontology unique cell classes mentioned per release. Some releases around 2014 are not available anymore, thus yielding 0 counts." /><figcaption aria-hidden="true"><span>Figure 1:</span> Cell Ontology unique cell classes mentioned per release. Some releases around 2014 are not available anymore, thus yielding 0 counts.</figcaption>
</figure>
</div>
<p>The solution to the challenges of cell type classification, thus, pass via ontologies and other best practices of knowledge management. <span class="citation" data-cites="qT8WxqjA">[<a href="#ref-qT8WxqjA" role="doc-biblioref">42</a>]</span>
Consequently, on the following chapter, we present tools for computer-based knowledge processing.
We introduce the technical aspects of ontologies and knowledge graphs, present Wikidata, and discuss how such tools can can influence life-sciences research.</p>
<h2 class="page_break_before" id="ontologies">Ontologies</h2>
<!-- - 1.2. Formal representation of knowledge and - 1.3. Knowledge Representation in biology -->
<p>The classification of biological concepts is at the core of biology. At least since the Aristotelian endeavours to group classes of animals, a good part of the scientific work is to capture concepts into knowledge systems <span class="citation" data-cites="DLTMbf54">[<a href="#ref-DLTMbf54" role="doc-biblioref">66</a>]</span>.
Linnaeus’ binomial system for naming species and Mendeleev’s periodic table are likely the two most famous classification systems but are part of a much larger ecosystem of structuring scientific knowledge.</p>
<!-- - 1.2.1. Descriptional logic and its historical context -->
<p>On the 20th century, the development of the analytical philosophy of Russel and Wittgenstein and their search for formalizations <span class="citation" data-cites="cZEmYsqx">[<a href="#ref-cZEmYsqx" role="doc-biblioref">67</a>]</span> gradually layed the foundations for the the logic of scientific descriptions.
Karl Popper and his “The Logic of Scientific Discovery”<span class="citation" data-cites="to2onCDy">[<a href="#ref-to2onCDy" role="doc-biblioref">68</a>]</span> was heavily influenced by analytical philosophy.
Less known among life scientists, Tarski’s inquiries on what can be considered to be “true” <span class="citation" data-cites="RghcuiSS">[<a href="#ref-RghcuiSS" role="doc-biblioref">69</a>]</span> were also foundational for formal ontologies.
In the end of the 20th century, the tradition of analytic philosophy contributed to the rise of applied ontology, which provided the theoretical basis for the computational ontlogies from the early 21st centhury. <span class="citation" data-cites="35incsRu">[<a href="#ref-35incsRu" role="doc-biblioref">70</a>]</span></p>
<p>The whole movement for formalization of knowledge progressed on the computational end. At the late 20th century, the advent of computational ontologies and large-scale knowledge graphs were at the root of the functioning of the World Wide Web.
This chapter will provide an overview of ontologies and knowledge graphs and their use in today’s biomedical sciences, alongside its future prospects.</p>
<h3 id="the-obo-foundry-and-biomedical-ontologies">The OBO Foundry and biomedical ontologies</h3>
<p>An ontology, as used here, is a formal computational representation of reality, which tries to represent each concept (and their relations) as precisely as possible. <span class="citation" data-cites="DLTMbf54">[<a href="#ref-DLTMbf54" role="doc-biblioref">66</a>]</span><br />
Constructing an ontology is a process of selecting and defining terms of interest, selecting and defining relationships of interest and making statements about reality using terms and relationships.
The Gene Ontology is probably the most well known biomedical ontology; it describes (among other things) different classes of biological process related to each_other by “is_a” and “part_of relations. <span class="citation" data-cites="18wkLcGxT">[<a href="#ref-18wkLcGxT" role="doc-biblioref">71</a>]</span> <span class="citation" data-cites="2KmDs8gy">[<a href="#ref-2KmDs8gy" role="doc-biblioref">72</a>]</span>.</p>
<p>The Gene Ontology is part of a much larger effort to formalize concepts across biology: the Open Biomedical and Biological Ontologies (OBO) Foundry. <span class="citation" data-cites="j3tJyG7J">[<a href="#ref-j3tJyG7J" role="doc-biblioref">73</a>]</span>
Created in 2007, the OBO Foundry is a hub of biomedical ontologies that sets guidelines for designing and constructing high-quality ontologies.
Under a common framework towards interoperability, the initial OBO Foundry united several independent ontologies (like the Cell Ontology, the Disease Ontology and the Protein Ontology).
At the same time, the creation of the Relation Ontology (RO) provided a go-to point for relations in biology that different ontologies could reuse.</p>
<h3 id="owl-and-ontology-languages">OWL and ontology languages</h3>
<p>One of the OBO Principles for its ontologies is that they should be resolvable as a “syntactically valid OWL file using the RDF-XML syntax.” (http://www.obofoundry.org/principles/fp-002-format.html).
The OWL Web Ontology Language was introduced as a standard by the W3C consortium in 2004.
OWL is not a programming language, as it does not instruct computers to perform actions, but an ontology language, which allows computerizable descriptions of the world.
Furthermore, it is an umbrella ontology language that includes several languages with varying levels of expressivity.
Generally, more expressive languages can represent more complex ideas but make computations harder.</p>
<p>Regardless of ontology’s sublanguage, it must be resolvable to an RDF-XML file.
RDF stands for Resource Description Framework, another W3C standard built around a graph-based data model <span class="citation" data-cites="djVboz3w">[<a href="#ref-djVboz3w" role="doc-biblioref">74</a>]</span>.
Statements in RDF are triples consisting of 2 nodes (a subject and an object) and an edge (a predicate) connecting the nodes.
All nodes and edges are represented in RDFs by International Resource Identifiers (IRIs), and there are many ways to lay out those IRIs on a text file to represent triples.
One of those layouts is the RDF-XML syntax, inspired by the XML markup language.
Arguably, other syntaxes (interchangeable with RDF-XML) are easier to read for a human.
As an example of an RDF triple, here is how one would represent in the Turtle RDF Syntax, the notion that plasmacytoid dendritic cells are a type of dendritic cells:</p>
<pre><code>http://purl.obolibrary.org/obo/CL_0000784  http://www.w3.org/2000/01/rdf-schema#subClassOf  http://purl.obolibrary.org/obo/CL_0000451 .</code></pre>
<p>Where http://purl.obolibrary.org/obo/CL_0000784 and http://purl.obolibrary.org/obo/CL_0000451 are the unique IDs in the Cell Ontology for “plasmacytoid dendritic cells” and dendritic cells, respectively, and http://www.w3.org/2000/01/rdf-schema#subClassOf is the identifier for the “subclass of” relation as defined by the RDF schema.</p>
<p>A longer explanation of the details of OWL and RDF is outside the scope of this work.
This brief introduction has a dual goal of introducing the architecture of formal representations and demonstrating the system’s complexity.
There is a high energy barrier to acquiring the knowledge and the technical skills to engage in ontology building.
That complexity might be one of the reasons why a tiny fraction of the biomedical communities represent data with ontologies, and an even smaller fraction engages with ontology building.</p>
<h2 id="wikidata">Wikidata</h2>
<!-- - 1.2.3. Web of Data and Linked Open Data -->
<p>Even though the Semantic Web (which ontologies are a part of) spawned with promises of a revolution in the way knowledge is shared, it is still to be widely known outside semantic engineering. Two recent projects are playing a significant role in bringing the Semantic Web to a broader audience: the Google Knowledge Graph and Wikidata.</p>
<p>The Google Knowledge Graph introduced the Semantic Web <em>de facto</em> in the daily life of users of Google. <span class="citation" data-cites="16mCkZC1r">[<a href="#ref-16mCkZC1r" role="doc-biblioref">75</a>/]</span>. Its underlying structure is similar to the triples in an ontology, but it is less concerned with being logically coherent and does have strict semantics of a representation.
In that way, Google Knowledge Graphs can feed on a variety of sources and not crash if there is some data modelling that, rigorously, could be inconsistent.
Even though there is not a strict boundary between ontologies and knowledge graphs, one reasonable interpretation is that a knowledge graph may not be perfectly coherent, as long as it still can provide enough knowledge and reasoning for the approach of interest.
While the lack of formal semantics limits reasoning and inference, the knowledge graphs are arguably easier to use, edit and understand, and so provide a user-friendly alternative for computable information with a lower entry barrier.</p>
<p>While the Google Knowledge Graph is widely used as a source of knowledge, it does not allow independent users to contribute information.
On the other hand, Wikidata, the collaborative knowledge graph of the Wikimedia Foundation, allows users to contribute with classes and statements in the same spirit as Wikipedia and share its “epistemic virtues, like power, speed and availability. <span class="citation" data-cites="qMozvNth">[<a href="#ref-qMozvNth" role="doc-biblioref">76</a>]</span>
Its power is derived from its large community of contributors, closely linked to the hugely successful Wikipedia.
With a community of more than 20,000 active editors <span class="citation" data-cites="SGzjBOG0">[<a href="#ref-SGzjBOG0" role="doc-biblioref">77</a>]</span> and growing, it can cover a much wider number of concepts than any user individually.
It is fast because one does not need to install any software or ask for permissions to update it: any user can do it via a web interface.
That speed makes it easier for newcomers to join and contribute, in contrast to OBO Foundry ontologies, which require extensive training on semantics and knowledge of Git/GitHub for contributions.
Finally, the information on Wikidata is available via a user interface, via a SPARQL query service and as large, full-size database dumps, providing full extent reusability.
The Wikidata model has been so successful that Google decided to migrate its knowledge base, Freebase, fully into Wikidata.<span class="citation" data-cites="xLpRePoh">[<a href="#ref-xLpRePoh" role="doc-biblioref">78</a>]</span></p>
<h3 id="the-inner-workings-of-wikidata">The inner workings of Wikidata</h3>
<!-- Wikidata's technical infrastructure -->
<p>Wikidata uses the same framework (RDF) that powers ontologies, and its model represents statements about the world in triples containing a subject, a property and an object. <span class="citation" data-cites="Ea6hVYrD">[<a href="#ref-Ea6hVYrD" role="doc-biblioref">79</a>]</span>
Its data model is serialized both in JSON and RDF.
The data model contains 17 different data types, including, for example, “Item”, an entry on Wikidata that refers to “o a real-world object, concept, or event that is given an identifier in Wikidata” and “String”, a “sequence of freely chosen characters interpreted as text”. <span class="citation" data-cites="17Yt10yrY">[<a href="#ref-17Yt10yrY" role="doc-biblioref">80</a>]</span>.
Knowledge is stored on Wikidata upon basic triples containing a subject (of type “Item”), a property and a value (which can be of any of the 17 types).
As of November 2021, Wikidata contains more than 90 million data items <span class="citation" data-cites="SGzjBOG0">[<a href="#ref-SGzjBOG0" role="doc-biblioref">77</a>]</span> and more than 9000 properties that link them to values.
As values often are other items, the database acquires a network format with labelled edges.</p>
<p>As seen in the example in <a href="#fig:douglas_adams">2</a>, each of the items in the database contains an item identifier (Q followed by numbers).
They also contain a label, a description, and a list of aliases, which can be recorded in more than 200 hundred languages, making it a multilingual project. <span class="citation" data-cites="S1Lksw7i">[<a href="#ref-S1Lksw7i" role="doc-biblioref">81</a>]</span>
Each item is decorated with statements comprising property-value pairs.
These pairs can be further specified via qualifiers and references, which treats the full triple as the subject, adding metadata to it (a process called reification <span class="citation" data-cites="ZhP1mKyl">[<a href="#ref-ZhP1mKyl" role="doc-biblioref">82</a>/#reification]</span>).
Qualifiers provide ways to extend the information on the triple, while references provide provenance, enabling users to judge the validity of the claims in the database.</p>
<div id="fig:douglas_adams" class="fignos">
<figure>
<img src="https://upload.wikimedia.org/wikipedia/commons/a/ae/Datamodel_in_Wikidata.svg" alt="Figure 2: Wikidata’s model for describing an item. Image released in Public Domain by Charlie Kritschmar." /><figcaption aria-hidden="true"><span>Figure 2:</span> Wikidata’s model for describing an item. Image released in Public Domain by Charlie Kritschmar.</figcaption>
</figure>
</div>
<p>All the information is available on a user interface and programmatically.
Advanced users can download dumps in JSON and RDF dumps and acess the data via the MediaWiki API and a SPARQL endpoint. <span class="citation" data-cites="4wSLyAPr">[<a href="#ref-4wSLyAPr" role="doc-biblioref">83</a>]</span>
Several wrappers of such services are available in languages such as R <span class="citation" data-cites="1HghnypzG">[<a href="#ref-1HghnypzG" role="doc-biblioref">84</a>]</span> and python <span class="citation" data-cites="VKWfrtR1">[<a href="#ref-VKWfrtR1" role="doc-biblioref">85</a>]</span>.
The data scheme can be seen in ?? model, where each item is connected to a statement node via a property in the “p:” namespace, from which references and qualifiers are accessible.
To facilitate primary usage, the namespace “wdt:” connects items to values directly, simplifying, for example, the writing of SPARQL queries.</p>
<div id="fig:datamodel" class="fignos">
<figure>
<img src="https://upload.wikimedia.org/wikipedia/commons/6/6c/Rdf_mapping-vector.svg" alt="Figure 3: Wikidata’s data model, scheme released under the CC-BY 4.0 license by Michael F. Schönitzer. It outlines the primary representation of statements, qualifiers and values in the Wikidata database" /><figcaption aria-hidden="true"><span>Figure 3:</span> Wikidata’s data model, scheme released under the CC-BY 4.0 license by Michael F. Schönitzer. It outlines the primary representation of statements, qualifiers and values in the Wikidata database</figcaption>
</figure>
</div>
<p>Information on Wikidata is released under a CC0 license, which enables full reuse of the data. <span class="citation" data-cites="tQIJPG4k">[<a href="#ref-tQIJPG4k" role="doc-biblioref">86</a>]</span>
One of the major points of access and reuse of the information is the Wikidata Query Service <span class="citation" data-cites="2wDsXBwd">[<a href="#ref-2wDsXBwd" role="doc-biblioref">87</a>]</span>, a core resource of the community which enables live querying in the SPARQL language. <span class="citation" data-cites="dsmUPCk">[<a href="#ref-dsmUPCk" role="doc-biblioref">88</a>]</span>
A number of services make use of embedded queries from the Wikidata Query Service <span class="citation" data-cites="2wDsXBwd">[<a href="#ref-2wDsXBwd" role="doc-biblioref">87</a>]</span> to create interactive, live dashboards like Scholia <span class="citation" data-cites="17P4zruFZ">[<a href="#ref-17P4zruFZ" role="doc-biblioref">89</a>]</span> and the SARS-CoV-2 Query Book <span class="citation" data-cites="guMp3q1V">[<a href="#ref-guMp3q1V" role="doc-biblioref">90</a>]</span></p>
<p>Wikidata is accessible in many ways and writable in many ways.
It provides a user-friendly, point-and-click interface for modifying the database, providing low entry barriers for newcomers.
It is also possible to semi-automatically reconcile spreadsheets to Wikidata items and use batch tools such as Open Refine <span class="citation" data-cites="17rRjL1Xj">[<a href="#ref-17rRjL1Xj" role="doc-biblioref">91</a>]</span> and Quickstatements <span class="citation" data-cites="K6ggpwRg">[<a href="#ref-K6ggpwRg" role="doc-biblioref">92</a>]</span>, which enable batches on the magnitude of thousands of edits.
For larger amounts of edits, it is possible to ask for bot permissions <span class="citation" data-cites="sSH04fxa">[<a href="#ref-sSH04fxa" role="doc-biblioref">93</a>]</span> and deploy systems that integrate big data sources.
Bot edits are made via the Wikimedia API and are predominantly written via Python wrappers, such as Pywikibot <span class="citation" data-cites="5K5MfnVl">[<a href="#ref-5K5MfnVl" role="doc-biblioref">94</a>]</span> and the Wikidata Integrator. <span class="citation" data-cites="qDI8I4IJ">[<a href="#ref-qDI8I4IJ" role="doc-biblioref">95</a>]</span></p>
<h3 id="wikidata-as-a-knowledge-graph-for-the-life-sciences">Wikidata as a knowledge graph for the life sciences</h3>
<p>Due to its privileged position inside the linked data ecosystem and its ease of writing and query, Wikidata has been growing as a hub for interoperable data for the life sciences community. <span class="citation" data-cites="3GqlN9Dk">[<a href="#ref-3GqlN9Dk" role="doc-biblioref">96</a>]</span> <span class="citation" data-cites="Ym2HoRLl">[<a href="#ref-Ym2HoRLl" role="doc-biblioref">97</a>]</span>
Even though Wikidata was created in 2013, the demand for a community-cured life sciences knowledge graph is apparent at least since 2008 <span class="citation" data-cites="179mXrGIk">[<a href="#ref-179mXrGIk" role="doc-biblioref">98</a>]</span> <span class="citation" data-cites="TItj7EYf">[<a href="#ref-TItj7EYf" role="doc-biblioref">99</a>]</span>
The Wikidata-like project proposed was eventually discontinued, an example of the challenge of maintaining independent biomedical databases. <span class="citation" data-cites="P2QdZvGc">[<a href="#ref-P2QdZvGc" role="doc-biblioref">100</a>]</span>
As Wikidata has a very large community, has stable funding and is at the core of modern technologies, like the Google Knowledge Graph <span class="citation" data-cites="xLpRePoh">[<a href="#ref-xLpRePoh" role="doc-biblioref">78</a>]</span> and Amazon’s Alexa, <span class="citation" data-cites="bO7BekzO">[<a href="#ref-bO7BekzO" role="doc-biblioref">101</a>]</span> it is virtually guaranteed that data in Wikidata will remain accessible for a long time, regardless of local funding schemes.</p>
<p>The Gene Wiki project <span class="citation" data-cites="18UXATsxP">[<a href="#ref-18UXATsxP" role="doc-biblioref">102</a>]</span> was likely the first large scale biomedical project to rely directly on the Wikipedia infrastructure for community curation.
It provided a direction connection between the generalist community of Wikipedia and domain experts.
The interplay of both communities is a topic of discussion and the opportunities and challenges were already discussed in NAR in 2012. <span class="citation" data-cites="13LWXQF0E">[<a href="#ref-13LWXQF0E" role="doc-biblioref">103</a>]</span><br />
Notably, Wikidata appeared chronologically after those efforts.<br />
Notwithstanding, the Gene Wiki research group has embraced the Wikidata environment for community biocuration and data interoperability <span class="citation" data-cites="2ZhxC0dg">[<a href="#ref-2ZhxC0dg" role="doc-biblioref">104</a>]</span><span class="citation" data-cites="DaJdrXLk">[<a href="#ref-DaJdrXLk" role="doc-biblioref">105</a>]</span> <span class="citation" data-cites="3GqlN9Dk">[<a href="#ref-3GqlN9Dk" role="doc-biblioref">96</a>]</span> <span class="citation" data-cites="aQfGxPRH">[<a href="#ref-aQfGxPRH" role="doc-biblioref">106</a>]</span>.
The information on Wikidata is still integrated to Wikipedias across multiple languages, often a a source of information in Wikipedia’s infoboxes.</p>
<p>Other projects outside the Gene Wiki initiative also started using Wikidata as a platform for knowledge integration.
A list of several projects that use Wikidata as part of their service to their community is given in table ??.
There is a movement exploring how Wikidata can be employed to advance Computational Biology and how it can be integrated to the current publication status quo. <span class="citation" data-cites="1GijvCQtv">[<a href="#ref-1GijvCQtv" role="doc-biblioref">107</a>]</span>
In that direction, Wikidata is being developed as a platform for scholarly linked open data, mainly via the Scholia platform <span class="citation" data-cites="hxzL9pmm">[<a href="#ref-hxzL9pmm" role="doc-biblioref">108</a>]</span> <span class="citation" data-cites="PdG2G4Bp">[<a href="#ref-PdG2G4Bp" role="doc-biblioref">109</a>]</span>,(<a href="https://scholia.toolforge.org/" class="uri">https://scholia.toolforge.org/</a>) which provides profiles of pre-templated SPARQL queries for entities like particular authors and articles (e.g. Scholia profile on Prof. Helder Nakaya available at <a href="https://scholia.toolforge.org/author/Q42614737" class="uri">https://scholia.toolforge.org/author/Q42614737</a>).</p>
<div id="tbl:wikidata_databases" class="tablenos">
<table id="tbl:wikidata_databases">
<caption><span>Table 1:</span> Samples of databases reconciled partially or totally to Wikidata
</caption>
<colgroup>
<col style="width: 15%" />
<col style="width: 40%" />
<col style="width: 44%" />
</colgroup>
<thead>
<tr class="header">
<th>name</th>
<th>project website</th>
<th>citation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>LOTUS</td>
<td><a href="https://lotus.naturalproducts.net/" class="uri">https://lotus.naturalproducts.net/</a></td>
<td><span class="citation" data-cites="P1EnNa7Q">[<a href="#ref-P1EnNa7Q" role="doc-biblioref">110</a>]</span></td>
</tr>
<tr class="even">
<td>GeneDB</td>
<td><a href="https://www.genedb.org/" class="uri">https://www.genedb.org/</a></td>
<td><span class="citation" data-cites="gqY9YdmH">[<a href="#ref-gqY9YdmH" role="doc-biblioref">111</a>]</span></td>
</tr>
<tr class="odd">
<td>Cellosaurus</td>
<td><a href="https://web.expasy.org/cellosaurus/" class="uri">https://web.expasy.org/cellosaurus/</a></td>
<td><span class="citation" data-cites="HUr1iigG">[<a href="#ref-HUr1iigG" role="doc-biblioref">112</a>]</span></td>
</tr>
<tr class="even">
<td>Complex Portal</td>
<td><a href="https://www.ebi.ac.uk/complexportal/" class="uri">https://www.ebi.ac.uk/complexportal/</a></td>
<td><span class="citation" data-cites="k4BJEMHE">[<a href="#ref-k4BJEMHE" role="doc-biblioref">113</a>]</span></td>
</tr>
<tr class="odd">
<td>WikiPathways</td>
<td><a href="https://www.wikipathways.org/" class="uri">https://www.wikipathways.org/</a></td>
<td><span class="citation" data-cites="pzgOjFLZ">[<a href="#ref-pzgOjFLZ" role="doc-biblioref">114</a>]</span></td>
</tr>
<tr class="even">
<td>Reactome</td>
<td><a href="https://reactome.org/" class="uri">https://reactome.org/</a></td>
<td><span class="citation" data-cites="1Fqp7ppqH">[<a href="#ref-1Fqp7ppqH" role="doc-biblioref">115</a>]</span></td>
</tr>
<tr class="odd">
<td>CIViC</td>
<td><a href="http://www.civicdb.org" class="uri">http://www.civicdb.org</a></td>
<td><span class="citation" data-cites="HUodcdz1">[<a href="#ref-HUodcdz1" role="doc-biblioref">116</a>]</span></td>
</tr>
<tr class="even">
<td>PubChem</td>
<td><a href="https://pubchem.ncbi.nlm.nih.gov" class="uri">https://pubchem.ncbi.nlm.nih.gov</a></td>
<td><span class="citation" data-cites="HUodcdz1">[<a href="#ref-HUodcdz1" role="doc-biblioref">116</a>]</span></td>
</tr>
<tr class="odd">
<td>Human Disease Ontology</td>
<td><a href="https://www.ebi.ac.uk/ols/ontologies/doid" class="uri">https://www.ebi.ac.uk/ols/ontologies/doid</a></td>
<td><span class="citation" data-cites="HUodcdz1">[<a href="#ref-HUodcdz1" role="doc-biblioref">116</a>]</span></td>
</tr>
</tbody>
</table>
</div>
<p>During the COVID-19 pandemic, Wikidata has spawned as a hotspot for modelling information about the virus and the pandemic in real-time. <span class="citation" data-cites="zH3KDqhp">[<a href="#ref-zH3KDqhp" role="doc-biblioref">117</a>]</span> <span class="citation" data-cites="wikidata:99196713">[<a href="#ref-wikidata:99196713" role="doc-biblioref"><strong>wikidata:99196713?</strong></a>]</span>
The general scope of the database allowed representation in a shared system of molecular, epidemiologic and socio-economic aspects of the pandemic. <span class="citation" data-cites="zH3KDqhp">[<a href="#ref-zH3KDqhp" role="doc-biblioref">117</a>]</span><span class="citation" data-cites="mPoPwN77">[<a href="#ref-mPoPwN77" role="doc-biblioref">118</a>]</span>
Information curated in Wikidata was immediately available, feeding live dashboards and other applications based on SPARQL queries. <span class="citation" data-cites="3YCzr3jU">[<a href="#ref-3YCzr3jU" role="doc-biblioref">119</a>]</span> <span class="citation" data-cites="SkOeuFRd">[<a href="#ref-SkOeuFRd" role="doc-biblioref">120</a>]</span> <span class="citation" data-cites="AOVCR8mW">[<a href="#ref-AOVCR8mW" role="doc-biblioref">121</a>]</span>
Additionally, as the information presented on Wikidata is multilingual and collaboratively edited, it presented itself as a resource for constructing structured vocabularies in non-English languages. <span class="citation" data-cites="19wiU1LCA">[<a href="#ref-19wiU1LCA" role="doc-biblioref">122</a>]</span></p>
<p>In addition to its value as a structured database, Wikidata is tightly connected to Wikipedia.
The gene identifiers in the context of Gene Wiki <span class="citation" data-cites="2ZhxC0dg">[<a href="#ref-2ZhxC0dg" role="doc-biblioref">104</a>]</span> are now fed to Wikipedias across languages, benefitting users directly.
Additionally, gene expression information from the Bgee database <span class="citation" data-cites="10EWNsZCz">[<a href="#ref-10EWNsZCz" role="doc-biblioref">123</a>]</span> was added to Wikidata and connected to Wikipedia, which led to a sizeable increase of access to the Bgee database.
Currently, Wikipedia is one of the top 3 sources from which people access Bgee (personal communication with Tarcisio Farias), thus leading to direct recognition for integrated bases.
More generally, the connections of Wikidata and Wikipedia make it unique in the power of flowing knowledge back to human-accessed interfaces.
In the words of Matthias Samwald <span class="citation" data-cites="864Pcm4C">[<a href="#ref-864Pcm4C" role="doc-biblioref">124</a>]</span> and colleagues, “Wikidata could emerge as a community-backed and highly visible structured knowledge base of medical and biological information, bringing concepts and methodologies such as controlled taxonomies, Semantic Web / semantic technologies and ontologies into mainstream use.”</p>
<p>In conclusion, Wikidata’s unique position, robustness and guarantee of long term stability prompts the need for works exploring new ways of integrating it into current knowledge management systems.
In light of the speed and breadth of the Human Cell Atlas and the challenges of knowledge representation on cells, this PhD works on addressing how Wikidata can play a role in organizing the discoveries about all human cell types.</p>
<h2 class="page_break_before" id="objectives">Objectives</h2>
<ul>
<li><p>Study and refine theories of classes of cells within the constraints of ontologies and knowledge bases</p></li>
<li><p>Catalog a comprehensive list of currently described cell types</p>
<ul>
<li>Develop a biocuration framework connected to Wikidata</li>
<li>Catalog as many cell types as possible</li>
</ul></li>
<li><p>Devise ways to connect life-sciences resources to Wikidata:</p>
<ul>
<li>Design and implement relations for making cell-type-related assertions</li>
<li>Connect Wikidata to the Cell Ontology and other authoritative sources of identifiers</li>
<li>Write bots and scripts to reconcile data sources to Wikidata</li>
</ul></li>
<li><p>Provide proofs-of-concept of how Wikidata integration can benefit the advancement of the Human Cell Atlas Project</p></li>
</ul>
<h2 class="page_break_before" id="methodology">Methodology</h2>
<p>This project’s methodology resembles practical research-action practices <span class="citation" data-cites="11UV98PHT">[<a href="#ref-11UV98PHT" role="doc-biblioref">125</a>]</span>.
The “action” facet is done by contributing to projects in the Human Cell Atlas and knowledge management context.</p>
<p>Research in 3 forms:
- Philosophical investigation on knowledge representations of cell types, both in formal logic and in academic literature
- Applied investigations of database integration and data quality in the context of Wikidata and biomedical ontologies
- Data-driven biomedical research targeted at hypothesis generation and literature-based discovery using knowledge at the level of cell-type</p>
<p>All research branches are linked to the improvement of knowledge management in biomedical sciences, focusing on the Human Cell Atlas.
The methods included the development and application of a framework for an organized reading of the scientific literature, providing contact with the different facets of biocuration and Human Cell Atlas-related research.</p>
<h2 id="organized-reading">Organized reading</h2>
<p>As much of the project is based directly on published research, we developed a reading framework, described in detail in the results section.
The framework is based on GitHub and includes Python scripts, a file organizing the reading list, and another documenting the reading history in RDF.
Notes and additional information are saved in a GitHub repository, and the structured information powers a live website with analytics on the users recent readings.
The source code for Wikidata Bib is available at <a href="https://github.com/lubianat/wikidata_bib/tree/template" class="uri">https://github.com/lubianat/wikidata_bib/tree/template</a> and notes on my readings can currently be accessed at <a href="https://lubianat.github.io/wikidata_bib/" class="uri">https://lubianat.github.io/wikidata_bib/</a>.</p>
<p>Additionally, the methodology included a discipline of reading that entails the daily task of reading 2 articles, one about “cell types” and another about “biocuration”.
The articles are obtained by a mixed manual and automatic approach, including a la carte selection of articles to read alongside Wikidata queries for Cell, Nature, Science and eLife papers about single-cell transcriptomics (query: <a href="https://w.wiki/4LHr" class="uri">https://w.wiki/4LHr</a>) and papers on biocuration (query: <a href="https://w.wiki/4LHi" class="uri">https://w.wiki/4LHi</a>).</p>
<h3 id="biocuration-of-cell-classes-for-wikidata">Biocuration of cell classes for Wikidata</h3>
<p>For each article about cell types read, cell types previously absent on Wikidata are added via a combination of curation in a Google Spreadsheet and a custom Python script (https://github.com/lubianat/wikidata_markers/tree/master/curation_of_classes).</p>
<h2 id="wikidata-updates">Wikidata updates</h2>
<p>We have updated the Wikidata database with new entities, relations and triples.
Creation of new entities was done in the Graphical User Interface (<a href="https://www.wikidata.org/wiki/Special:NewItem" class="uri">https://www.wikidata.org/wiki/Special:NewItem</a>), via the batch Quickstatements tool (<a href="https://quickstatements.toolforge.org/#/" class="uri">https://quickstatements.toolforge.org/#/</a>) and the Wikidata Integrator python library (<a href="https://github.com/SuLab/WikidataIntegrator" class="uri">https://github.com/SuLab/WikidataIntegrator</a>).</p>
<p>Properties, which link items to values need community approval. Under the scope of this PhD project, we have gotten the community approval for the properties :
Cell Ontology ID (<a href="https://www.wikidata.org/wiki/Property:P7963" class="uri">https://www.wikidata.org/wiki/Property:P7963</a>) used to link cell types to their IDs in the Cell Ontology and has marker (<a href="https://www.wikidata.org/wiki/Property:P8872" class="uri">https://www.wikidata.org/wiki/Property:P8872</a>) used to link cell types to genes and proteins considered their markers.</p>
<p>The property acceptance cycle takes at least one week and is open for opinions by any Wikidata user.
All the information regarding property proposals is available at <a href="https://www.wikidata.org/wiki/Wikidata:Property_proposal" class="uri">https://www.wikidata.org/wiki/Wikidata:Property_proposal</a>.</p>
<h2 id="cell-ontology-participation">Cell Ontology participation</h2>
<p>As part of the research-action process, I have joined the Cell Ontology working group.
I participate in the monthly meetings and sporadic workshops, learning and contributing to the discussions.
Additionally, I contribute to the ontology development, actively engaging in the Cell Ontology GitHub repository (<a href="https://github.com/obophenotype/cell-ontology" class="uri">https://github.com/obophenotype/cell-ontology</a>) and contributing with new terms and assertions.
I edit the ontology with the software for ontology editing Protégé v. 5.5.0 <span class="citation" data-cites="CHsAoEvr">[<a href="#ref-CHsAoEvr" role="doc-biblioref">126</a>]</span>.</p>
<h3 id="status-of-cell-type-info-on-wikidata-and-the-cell-ontology">Status of cell type info on Wikidata and the Cell Ontology</h3>
<p>Status of cell type information on Wikidata was accessed via SPARQL queries combined with processing in python and is available at <a href="https://colab.research.google.com/drive/1GvQXOs51_U8icdGMtKXMeLOXKM8pXWet#scrollTo=szvBWI9zr_AA" class="uri">https://colab.research.google.com/drive/1GvQXOs51_U8icdGMtKXMeLOXKM8pXWet#scrollTo=szvBWI9zr_AA</a>.</p>
<p>Counts of cell classes in the Cell Ontology were performed via regex matching on Cell Ontology releases following the code available at <a href="https://github.com/lubianat/cell_ontology_count" class="uri">https://github.com/lubianat/cell_ontology_count</a>.</p>
<h2 class="page_break_before" id="preliminary-results">Preliminary results</h2>
<!-- Maybe include annotation of HCA articles ? -->
<!-- Maybe include Scholia and Bioinformatics ? -->
<h2 class="page_break_before" id="concept-of-cell-types">Concept of cell types</h2>
<h3 id="general-work-on-the-concept-of-cell-type">General work on the concept of cell type</h3>
<p>As an initial step of this PhD project, we investigatet the definition of “cell type” for knowledge management on Wikidata.
The definition of “cell type” is a topic of avid debate. <span class="citation" data-cites="paKiKTRS sWv3okaH 19yYJWEkz BlxFp1v3 eOUB1Bmj TupzTvtA w0Woz7JE srfuRMe 1GXQByEaQ k0Pe9YOB pNGap1Du MoaYZAA0">[<a href="#ref-paKiKTRS" role="doc-biblioref">9</a>,<a href="#ref-sWv3okaH" role="doc-biblioref">10</a>,<a href="#ref-19yYJWEkz" role="doc-biblioref">11</a>,<a href="#ref-BlxFp1v3" role="doc-biblioref">12</a>,<a href="#ref-eOUB1Bmj" role="doc-biblioref">13</a>,<a href="#ref-TupzTvtA" role="doc-biblioref">14</a>,<a href="#ref-w0Woz7JE" role="doc-biblioref">15</a>,<a href="#ref-srfuRMe" role="doc-biblioref">16</a>,<a href="#ref-1GXQByEaQ" role="doc-biblioref">17</a>,<a href="#ref-k0Pe9YOB" role="doc-biblioref">18</a>,<a href="#ref-pNGap1Du" role="doc-biblioref">19</a>,<a href="#ref-MoaYZAA0" role="doc-biblioref">20</a>]</span>.
Before we handled data in a large scale, we dedicated time for solidifying our theoretical basis of “cell type”.</p>
<p>In a preprint derived from this PhD project <span class="citation" data-cites="uCE45Jh8">[<a href="#ref-uCE45Jh8" role="doc-biblioref">127</a>]</span>, we proposed naming conventions for different cell types classes.
Much of the literature uses the same names for single species (e.g., when dealing with a cell type as an evolutionary unit) or multiple species (e.g., in the Cell Ontology).
Fe find it helpful to distil these different cases into different categories.
Given the importance of the species’ concept in biological classification <span class="citation" data-cites="AHGRV6KQ">[<a href="#ref-AHGRV6KQ" role="doc-biblioref">128</a>]</span>, we derive a species-centric view on naming classes of cell types.
Using the notion of “taxonomic scope” as the taxonomic range in which instances of the class are expected to materialize, we propose 4 classes (Figure <a href="#fig:classes">4</a>):</p>
<ul>
<li>archetypes, for when the taxonomic scope of the type is beyond the level of species; for example, “mammal neutrophils.” Note that this also includes species-neutral cell types, which are assumed to display instances in multiple species.</li>
<li><em>sensu stricto</em> cell types, for when the taxonomic scope of the type corresponds to a single species; for example, <em>Mus musculus</em> neutrophils.”</li>
<li>infratypes, for when the taxonomic scope is below the species level; for example, considering the mouse strain “C57BL/6J”, “neutrophils from C57BL/6J mice”.</li>
<li>technotypes, for specific, experimentally defined cell types that harbor in their definition the precise conditions of the cells sampled; “2-month-old male C57BL/6J, Ly-6G<sup>+</sup> CD11b<sup>+</sup> M-CSF R<sup>-</sup> CD244<sup>-</sup> neutrophils”.</li>
</ul>
<div id="fig:classes" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/lubianat/fapesp_report_1/main/content/images/archetypes_hn.jpg" alt="Figure 4: Names for classes of cell types." /><figcaption aria-hidden="true"><span>Figure 4:</span> Names for classes of cell types.</figcaption>
</figure>
</div>
<p>By specifying 4 different categories of cell types, we can hone in our organization.
While resources often do not make the taxonomic scope explicit, it might be inferred.
For example, the vast majority of biomedical articles present experiments with a single strain or a single species.
While two articles using different species might call some cells by the same name, they have important intrinsic differences.
Even if functionally equivalent, a mouse fibroblast can never become a human fibroblast, no matter the protocol.<br />
While implicitly obvious, for computer-based, large scale knowledge management, it becomes necessary to state the obvious.</p>
<p>The division between archetypes and <em>sensu stricto</em> cell types is of particular importance for biocuration and data annotation.
Usually, names and identifiers for genes and proteins are standardized for single species.<span class="citation" data-cites="baJRsCMu">[<a href="#ref-baJRsCMu" role="doc-biblioref">129</a>]</span>
Thus, if we want to annotate marker genes, we would do better associating them to a species-specific cell type (a <em>sensu stricto</em> cell type) instead of the more vague association to a species-neutral type.
Of note, current scRNAseq reference datasets and databases still use species-neutral cell type IDs(e.g. in the reference HuBMAP app; <a href="https://azimuth.hubmapconsortium.org/references/" class="uri">https://azimuth.hubmapconsortium.org/references/</a>).</p>
<p>Our theoretical discussion on the notion of “cell type” extends the current state-of-the-art and introduces new ways to organize our knowledge about cells.
The technotype and the infratype are currently pure theoretical constructs, and almost no resources deal with cell types at the level of strains or below.
While we reason that this level of granularity would provide a more precise description of research projects, they are still far from being applicable in the present, and are present as tools for the future.
The division of archetypes and <em>sensu stricto</em> cell types, on the other hand, was immediate value.
As an example, it was instrumental for the integration of the Panglao database of cell markers to Wikidata, described in the following session of the results.</p>
<h3 id="a-simplified-definition">A simplified definition</h3>
<p>While theoretical discussion on the notion of “cell type” are important, a reasonable consensus is likely to take longer than the duration of this PhD project.
Here we adopt a liberal view of cell type, defining, for our purposes, a cell type as any class of cells described by a domain expert with evidence of the reality of its instances.
The requirement of evidence of existence in the material world is based on the principle of instantiation of ontological realism <span class="citation" data-cites="5ZFRyTy5">[<a href="#ref-5ZFRyTy5" role="doc-biblioref">130</a>]</span>. Barry Smith and Werner Ceuster state “A term should be included in a reference ontology only if there is experimental evidence that instances to which that term refers exist in reality. (‘Exists’ here should be understood in a tenseless sense in order to accommodate, for example, universals pertaining to extinct species as well as universals such as swarm or hurricane which are instantiated only intermittently.)”
Following their advice, our minimum requirement for a cell type is public evidence for materializations of its instances.</p>
<p>By “type”, or “class”, we mean an abstract entity in a sense intended by the multilevel theory (MLT) of conceptual modelling <span class="citation" data-cites="eP47FILn">[<a href="#ref-eP47FILn" role="doc-biblioref">131</a>]</span>
Figure <a href="#fig:multilevel">5</a> displays a simplified version of MLT adopted throughout this project.
In this framework, real-world entities are materializations of <em>individuals</em>.
<em>Individuals</em> are theoretical constructs that are (1) thought to exist or have existed, as per the principle of instantiation, and (2) refer to only one material entity at any point in time.
For example, Wikidata’s entries for “Helder Nakaya (<a href="https://www.wikidata.org/wiki/Q42614737">Q42614737</a>)” and “Charles Darwin (<a href="https://www.wikidata.org/wiki/Q42614737">Q1035</a>)” are considered <em>individuals</em> by Multi-Level Theory.
Other examples of <em>individuals</em> include “Albert Einstein’s brain (<a href="https://www.wikidata.org/wiki/Q2464312">Q2464312</a>)” and the “Christ the Redeemer statue (<a href="https://www.wikidata.org/wiki/Q79961">Q79961</a>)”.</p>
<div id="fig:multilevel" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/lubianat/multilevel_ontology_drawings/master/combination_human_cell.png" style="width:85.0%" alt="Figure 5: A multilevel theory (MLT) can divide the theoretical realm into different kingdoms. A) A representation of people in the MLT framework as adopted in this work. The theoretical-realm entity “Mahatma Gandhi” is materialized by the material-realm Mahatma Gandhi. The theoretical individual is considered an instance of multiple classes such as “Indian lawyer” and “Homo sapiens”, which are related to each other via subclass relations. The classes themselves are instances of metaclasses, like “taxon”, a first-order metaclass. B) An analogous representation of the MLT framework, but applied to cells and cell types." /><figcaption aria-hidden="true"><span>Figure 5:</span> A multilevel theory (MLT) can divide the theoretical realm into different kingdoms. A) A representation of people in the MLT framework as adopted in this work. The theoretical-realm entity “Mahatma Gandhi” is materialized by the material-realm Mahatma Gandhi. The theoretical <em>individual</em> is considered an instance of multiple <em>classes</em> such as “Indian lawyer” and “<em>Homo sapiens</em>”, which are related to each other via subclass relations. The classes themselves are instances of <em>metaclasses</em>, like “taxon”, a first-order metaclass. B) An analogous representation of the MLT framework, but applied to cells and cell types.</figcaption>
</figure>
</div>
<p>In MLT, <em>individuals</em> are instances of some <em>classes</em>.
For example, both “Helder Nakaya (<a href="https://www.wikidata.org/wiki/Q42614737">Q42614737</a>)” and “Charles Darwin (<a href="https://www.wikidata.org/wiki/Q42614737">Q1035</a>)” could be represented as instances of the class “<em>Homo sapiens</em> (<a href="https://www.wikidata.org/wiki/Q15978631">Q15978631</a>)”.
“<em>Homo sapiens</em> (Q15978631)” is only one of the classes that those individuals belong to.
Another one is “animal (<a href="https://www.wikidata.org/wiki/Q729">Q729</a>)”.
As all instances of “<em>Homo sapiens</em> (Q15978631)” are also instances of “animal (Q729)”, “<em>Homo sapiens</em> (Q15978631)” is a subclass of “animal (Q729)”.
It is possible to continue the hierarchy of subclasses: “animal (Q729)” is a subclass of “organism (Q7239)”, and so on until the root case “entity (<a href="https://www.wikidata.org/wiki/Q35120">Q35120</a>)”</p>
<p>Classes can also behave as individuals in some aspects.
For example, both “<em>Homo sapiens</em> (Q15978631)” and “animal (Q729)” are instances of “taxon (<a href="https://www.wikidata.org/wiki/Q16521">Q16521</a>)”.
“Taxon (Q16521)”, thus, is a <em>metaclass</em>, or, more precisely, a <em>1st-order metaclass</em>.
Other examples of metaclasses are “<em>species</em> (<a href="https://www.wikidata.org/wiki/Q7432">Q7432</a>)” and “phylum (<a href="https://www.wikidata.org/wiki/Q38348">Q38348</a>)”.
These, in turn, are instances of “taxonomic rank (<a href="https://www.wikidata.org/wiki/Q427626">Q427626</a>)”, a <em>2nd-order metaclass</em>.</p>
<p>In the Figure <a href="#fig:multilevel">5</a> B there is a proposal of this version of MLT for cell types.
As individual cells are rarely named, for the sake of example, we can consider the “zygote of Mahatman Gandhi” as an <em>individual</em> in the theoretical system, an instance of the <em>class</em> “zygote (<a href="https://www.wikidata.org/wiki/Q170145">Q170145</a>)”.
“Zygote(Q170145)” is an instance of the <em>metaclass</em> “cell type (<a href="https://www.wikidata.org/wiki/Q189118">Q189118</a>)”
A more concrete example of from RNA-sequencing datasets, where there are barcodes for each cell in a particular sample.
Each barcode labels an <em>individual</em>.
Thus, labelling single-cells is a process of identification, where each <em>individual</em> is connected to a <em>class</em> of interest.</p>
<!-- From  https://raw.githubusercontent.com/lubianat/technotype/9f428cbcf8e8a8d2843faac684275fb0c37914ed/content/05.logical.md BEGIN-->
<p>We avoid the dissection of the differences between persistent classes of cells (often called “cell types”) or the transient, fugacious classes of cells (often called “cell states”) (see “Definition of cell identity” section in <span class="citation" data-cites="pFijpXkl">[<a href="#ref-pFijpXkl" role="doc-biblioref">132</a>]</span> for an example). We also consider only the cell as it was observed in an experiment, not necessarily the future conditions of any cell (i.e., the “cell fate”). <span class="citation" data-cites="OjxQDZpx">[<a href="#ref-OjxQDZpx" role="doc-biblioref">35</a>]</span><br />
Even though such a distinction is an important topic for theoretical research, it is outside the initial scope of this work.</p>
<p>Another consequence of is that “subtype” becomes redundant with the idea of cell type and differ only stylistically.
The notion of subtype, only will make sense in context of superclasses, but all in the same kingdom (i.e. the kingdom of classes).</p>
<!-- From  https://raw.githubusercontent.com/lubianat/technotype/9f428cbcf8e8a8d2843faac684275fb0c37914ed/content/05.logical.md END-->
<p>We opted to frame our work around the term “cell type” due to its historical usage and familiarity for the life sciences community.
The term “cell class” is also used in the literature and is a suitable synonym for our notion of cell type.
Other related terms present semantic ambiguities, and were avoided whenever possible.
For example, terms as “cell set,” “cell population,” and “cell cluster,” can reminisce of a specific, countable group of cells, frequently from the same experiment.
The term “cell identity” has also been suggested for avoiding the cell type/cell state dilemma <span class="citation" data-cites="WKbly37M">[<a href="#ref-WKbly37M" role="doc-biblioref">46</a>]</span>, but we avoid it to emphasize a nominalistic perspective (in the Popperian sense<span class="citation" data-cites="1C6LI68h6">[<a href="#ref-1C6LI68h6" role="doc-biblioref">133</a>]</span>).
In doing so, we reinforce the intent to represent the cell types <em>reported to exist</em> instead of stating bluntly which cell types <em>exist</em> or are <em>essential</em> for human beings.</p>
<p>The employment of MLT and species-specific cell type are instrumental for the next chapters of this work.
In the chapter about the PanglaoDB integration, we describe how we applied the theory to add marker information to Wikidata and cleaned up conceptual disarrays throughout the platform.
Later, on the chapter about Wikidata Bib, we describe how we performed a large-scale curation of the biomedical literature for new cell types, using the theories discussed here as a starting principle.</p>
<h2 class="page_break_before" id="panglaodb-integration-to-wikidata">PanglaoDB integration to Wikidata</h2>
<h3 id="introduction">Introduction</h3>
<p>Biomedical databases gather structured information for end users.
They are present in different states of maintenance, and reconciling cell-oriented databases to Wikidata has the potential to increase interoperability, and multiply the value of previous biocuration efforts.
PanglaoDB <span class="citation" data-cites="M0lqRYJb">[<a href="#ref-M0lqRYJb" role="doc-biblioref">134</a>]</span> <span class="citation" data-cites="T3PlZ3Vh">[<a href="#ref-T3PlZ3Vh" role="doc-biblioref">135</a>]</span> is a publically-available database that contains data and metadata on hundreds of single-cell RNA sequencing experiments.
It provides extensive information on cell types, genes, tissues, and cell type markers, obtained via automatic and manual methods.
It also displays a rich web user interface for easy data acquisition, including database dumps for bulk downloads.</p>
<p>As of 8 December June 2021, the article describing PanglaoDB had been cited 230 times.
Despite its use by the community, the database is on a 3-star category for Linked Open Data <span class="citation" data-cites="OGXYtGZ8">[<a href="#ref-OGXYtGZ8" role="doc-biblioref">136</a>]</span> as it does not use the open semantic standards from W3C (RDF and SPARQL) needed for a 4-star rank, neither the links to external data via standard identifiers that make datasets 5-star.
Improving the data format is a valuable step in making biological knowledge FAIR (Findable, Accessible, Interoperable, and Reusable).
Thus, we provide a case study of making PandlaoDB available in a 5-star Linked Open Data Format on Wikidata.</p>
<p>As of August 2020, Wikidata had 264 items categorized as a “cell type”, considerably less than the Cell Ontology, which counts over two thousand cell types <span class="citation" data-cites="4AEy2xhQ agnqfdk6">[<a href="#ref-agnqfdk6" role="doc-biblioref">52</a>,<a href="#ref-4AEy2xhQ" role="doc-biblioref">137</a>]</span>.
Strikingly, there were also 23 items categorized as instances of “cell (Q7868)”.
This classification is imprecise, as an instance of cell would be an individual named cell from a single named individual, an example of conceptual disarray that often occurs on Wikidata. <span class="citation" data-cites="imSZTi52">[<a href="#ref-imSZTi52" role="doc-biblioref">138</a>]</span></p>
<p>Wikidata editors often mix 1st-order classes such as “cells” and “organs” with metaclasses like “cell types” and “organ types”.
As mentioned in the chapter on the concept of cell type and Multi Level Theory, individuals, like the “Dolly sheep zygote” and the “brain of Albert Einstein” are instances of classes like “zygote” and “brain”, respectively.
Classes, like “zygote” and “brain” are instances of <em>metaclasses</em>, like “cell type” or “organ type”.</p>
<p>We diligently fixed and improved the conceptual consistency of cell type entries on Wikidata.
As of 8 December 2021, the Wikidata database contains 2834 instances of “cell type” (see current status at <a href="https://w.wiki/b2t" class="uri">https://w.wiki/b2t</a>) and 0 instances of “cell” (<a href="https://w.wiki/4XAg" class="uri">https://w.wiki/4XAg</a>) highlighting the improvements in both quantity and quality.
This increase stems from the PanglaoDB integration (around 430 new types) and the Wikidata Bib curation described later.</p>
<h3 id="methodology-for-panglaodb-integration">Methodology for PanglaoDB integration</h3>
<p>After obtaining approval from the database owners, we matched genes and cell types to Wikidata and performed Wikidata queries to demonstrate the value of the approach.
An overview of the process is shown in ??.
<img src="https://github.com/lubianat/fapesp_report_1/blob/main/content/images/poster_ISCB_intro.png?raw=true" id="fig:iscb_intro" style="width:85.0%" alt="Overview of how data from a database is integrated into Wikidata" /></p>
<h3 id="class-creation-on-wikidata">Class creation on Wikidata</h3>
<p>Classes corresponding to species-neutral classes were curated from Wikidata using Wikidata’s Graphic User Interface.
A manually-curated dictionary matching terms in PanglaoDB to Wikidata identifiers was assembled and used for integration.
Cell types that were not represented on Wikidata were added to the database via the graphical user interface (<a href="https://www.wikidata.org/wiki/Special:NewItem" class="uri">https://www.wikidata.org/wiki/Special:NewItem</a>) and logged in the reference table.</p>
<p>Species-specific cell types for human and mouse cell types were created for every entry in the reference table and connected to the species-neutral concept via a “<a href="http://www.wikidata.org/entity/P279">subclass of</a>” property (e.g. every single “<a href="http://www.wikidata.org/entity/Q101405102">human neutrophil</a>” is a also “<a href="http://www.wikidata.org/entity/Q188417">neutrophil</a>”).
Our approach was analogous to the one taken by the CELDA ontology to create species-specific cell types. <span class="citation" data-cites="RLHsA1U8">[<a href="#ref-RLHsA1U8" role="doc-biblioref">139</a>]</span></p>
<h4 id="integration-of-panglaodb-to-wikidata">Integration of PanglaoDB to Wikidata</h4>
<p>After receiving authorization by e-mail from the PanglaoDB developer, Oscar Franzen, the PanglaoDB markers dataset was downloaded manually from PanglaoDB’s website (<a href="https://panglaodb.se/markers/PanglaoDB_markers_27_Mar_2020.tsv.gz" class="uri">https://panglaodb.se/markers/PanglaoDB_markers_27_Mar_2020.tsv.gz</a>) for integration. It contains 15 columns and 8256 rows. Only the columns <code>species</code>, <code>official gene symbol</code>, and <code>cell type</code> were used for the reconciliation.
The reconciled dataset was uploaded to Wikidata via the WikidataIntegrator Python package <span class="citation" data-cites="qDI8I4IJ">[<a href="#ref-qDI8I4IJ" role="doc-biblioref">95</a>]</span>, a wrapper for the Wikidata Application Programming Interface.</p>
<h4 id="sparql-queries">SPARQL queries</h4>
<p>Besides the Wikidata Dumps, Wikidata provides a SPARQL endpoint with a Graphical User Interface (<a href="https://query.wikidata.org/" class="uri">https://query.wikidata.org/</a>).
Updated data was immediately accessible via this endpoint, enabling integrative queries integrated with other database statements.</p>
<h3 id="overview-of-integrated-information-on-wikidata">Overview of integrated information on Wikidata</h3>
<h4 id="cell-marker-information-on-wikidata">Cell Marker information on Wikidata</h4>
<p>Adding marker information on Wikidata was not possible before this study and became possible after we proposed and got community approval of the property “has marker” (P8872).
Figure <a href="#fig:chat_marker">6</a> shows 2 of the current markers of “human colinergic neuron”(<a href="http://www.wikidata.org/entity/Q101405051">Q101405051</a>), <a href="http://www.wikidata.org/entity/Q14863671">CHAT</a> and <a href="http://www.wikidata.org/entity/Q407983">ACHE</a>, as they are seen on Wikidata.
The PanglaoDB is referenced both via URL to the website (<a href="https://panglaodb.se/markers.html" class="uri">https://panglaodb.se/markers.html</a>) and a pointer to the PanglaoDB item on Wikidata, <a href="http://www.wikidata.org/entity/Q99936939">Q99936939</a>.</p>
<div id="fig:chat_marker" class="fignos">
<figure>
<img src="https://github.com/lubianat/fapesp_report_1/blob/main/content/images/chat_marker.png?raw=true" style="width:85.0%" alt="Figure 6: Subset of the marker genes for item Q101405051 (human cholinergic neuron)" /><figcaption aria-hidden="true"><span>Figure 6:</span> Subset of the marker genes for item Q101405051 (human cholinergic neuron)</figcaption>
</figure>
</div>
<p>Now that we re-formatted the markers on PanglaoDB as Linked Open Data, we can make queries that were not possible before, including federated queries with other biological databases, such as Uniprot <span class="citation" data-cites="HUJ9NzAb">[<a href="#ref-HUJ9NzAb" role="doc-biblioref">140</a>]</span>
and Wikipathways <span class="citation" data-cites="AgOOCR24">[<a href="#ref-AgOOCR24" role="doc-biblioref">141</a>]</span>.
Due to previous similar reconciliation projects, Wikidata already contains information about genes, including their relations to Gene Ontology (GO) terms.</p>
<p>PanglaoDB’s integration to the Wikidata ecosystem allows us to ask various questions (figure <a href="#fig:iscb_results">7</a>).</p>
<div id="fig:iscb_results" class="fignos">
<figure>
<img src="https://github.com/lubianat/fapesp_report_1/blob/main/content/images/poster_ISCB_results.png?raw=true" style="width:85.0%" alt="Figure 7: SPARQL queries in Wikidata now harness information from Panglao DB. Queries with the above design were run on Wikidata. Results might change in real-time with Wikidata updates by contributors A-C) Graphical representation of feasible SPARQL queries (https://w.wiki/yQ6, https://w.wiki/yQD and https://w.wiki/3HjX,). D) Sample result from query C." /><figcaption aria-hidden="true"><span>Figure 7:</span> SPARQL queries in Wikidata now harness information from Panglao DB. Queries with the above design were run on Wikidata. Results might change in real-time with Wikidata updates by contributors A-C) Graphical representation of feasible SPARQL queries (https://w.wiki/yQ6, https://w.wiki/yQD and https://w.wiki/3HjX,). D) Sample result from query C.</figcaption>
</figure>
</div>
<h3 id="which-human-cell-types-are-related-to-neurogenesis-via-their-markers">“Which human cell types are related to neurogenesis via their markers?”</h3>
<p>As expected, the query below retrieved a series of neuron types, such as “<a href="https://www.wikidata.org/wiki/Q101404913">human Purkinje neuron</a>” and “<a href="https://www.wikidata.org/wiki/Q101405091">human Cajal-retzius cell</a>.” It also retrieved non-neural cell types such as the “<a href="https://www.wikidata.org/wiki/Q101405109">human loop of Henle cell</a>, a kidney cell type, and”<a href="https://www.wikidata.org/wiki/Q101404928">human osteoclast</a>.
These seemingly unrelated cell types markedly express genes involved in neurogenesis, but that does not mean that they are involved with this process.
The unexpected results reinforce that one needs to be careful when using curated pathways to analyze gene sets, as false positives abound.</p>
<p>The molecular process that gene products take part depends on the cell type.
SPARQL allows us to seamlessly compare Gene Ontology processes with cell marker data, providing a sandbox to generate hypotheses and explore the biomedical knowledge landscape.</p>
<div id="tbl:neuro" class="tablenos">
<table id="tbl:neuro">
<caption><span>Table 2:</span> Sample of 10 cell types related to neurogenesis via markers (07/02/2020, full query on <a href="https://w.wiki/yQ6" class="uri">https://w.wiki/yQ6</a>).
</caption>
<thead>
<tr class="header">
<th style="text-align: left;">geneLabel</th>
<th style="text-align: left;">cellTypeLabel</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">OMP</td>
<td style="text-align: left;">human purkinje neuron</td>
</tr>
<tr class="even">
<td style="text-align: left;">OMP</td>
<td style="text-align: left;">human olfactory epithelial cell</td>
</tr>
<tr class="odd">
<td style="text-align: left;">OMP</td>
<td style="text-align: left;">human neuron</td>
</tr>
<tr class="even">
<td style="text-align: left;">EPHB1</td>
<td style="text-align: left;">human oligodendrocyte</td>
</tr>
<tr class="odd">
<td style="text-align: left;">EPHB1</td>
<td style="text-align: left;">human osteoclast</td>
</tr>
<tr class="even">
<td style="text-align: left;">PCSK9</td>
<td style="text-align: left;">human delta cell</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PCSK9</td>
<td style="text-align: left;">human loop of Henle cell</td>
</tr>
<tr class="even">
<td style="text-align: left;">CXCR4</td>
<td style="text-align: left;">human b cell</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CXCR4</td>
<td style="text-align: left;">human T cell</td>
</tr>
<tr class="even">
<td style="text-align: left;">CXCR4</td>
<td style="text-align: left;">human NK cell</td>
</tr>
</tbody>
</table>
</div>
<h3 id="which-cell-types-express-markers-associated-with-parkinsons-disease">“Which cell types express markers associated with Parkinson’s disease?”</h3>
<p>Besides integration with Gene Ontology, Wikidata reconciliation connects the marker gene info on PanglaoDB with disease markers.</p>
<p>Disease genes are often compiled from Genomic Wide Association Studies, which look for sequence variation in the DNA.
These studies are commonly blind to the cell types related to the pathophysiology of the disease.
In the query below, we can see cell types marked by genes genetically associated with Parkinson’s disease.
Even considering the false positives, the overview can aid domain experts in coming up with novel hypotheses.</p>
<div id="tbl:parkinson" class="tablenos">
<table id="tbl:parkinson">
<caption><span>Table 3:</span> Sample of 5 cell types related to Parkinson’s disease via markers (07/02/2020, full query on <a href="https://w.wiki/yQD" class="uri">https://w.wiki/yQD</a>).
</caption>
<thead>
<tr class="header">
<th style="text-align: left;">geneLabel</th>
<th style="text-align: left;">diseaseLabel</th>
<th style="text-align: left;">cellTypeLabel</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BST1</td>
<td style="text-align: left;">Parkinson’s disease</td>
<td style="text-align: left;">human b cell</td>
</tr>
<tr class="even">
<td style="text-align: left;">BST1</td>
<td style="text-align: left;">Parkinson’s disease</td>
<td style="text-align: left;">human neutrophil</td>
</tr>
<tr class="odd">
<td style="text-align: left;">RIT2</td>
<td style="text-align: left;">Parkinson’s disease</td>
<td style="text-align: left;">human neuron</td>
</tr>
<tr class="even">
<td style="text-align: left;">SH3GL2</td>
<td style="text-align: left;">Parkinson’s disease</td>
<td style="text-align: left;">human alpha cell</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SH3GL2</td>
<td style="text-align: left;">Parkinson’s disease</td>
<td style="text-align: left;">human beta-cell</td>
</tr>
</tbody>
</table>
</div>
<h2 id="discussion-and-conclusion">Discussion and conclusion</h2>
<p>In this part of the PhD project, we re-released the knowledge curated in PanglaoDB on Wikidata, connecting it to the semantic web.
Each cell-type/marker statement was added to Wikidata with a pointer to PanglaoDB and a citation of the article, providing proper provenance.
Based on the theoretical considerations on the concept of cell type, we added species-specific terms to Wikidata for cell types of <em>Homo sapiens</em> and <em>Mus musculus</em> described in the PanglaoDB database.</p>
<p>This work exemplifies the power of releasing Linked Open Data via Wikidata, and provides the biomedical community with the first semantically accessible, 5-star LOD dataset of cell markers, easily reachable from Wikidata’s SPARQL Query Service (<a href="https://query.wikidata.org/" class="uri">https://query.wikidata.org/</a>).
Alongside other case studies of biomedical data integration to Wikidata (see <span class="citation" data-cites="mPoPwN77">[<a href="#ref-mPoPwN77" role="doc-biblioref">118</a>]</span>, it contributes with tools and practices to serve as basis for contributors.)
The work also paves the way for reconciling of other databases for cell-type markers, such as CellMarker <span class="citation" data-cites="chGii6yw">[<a href="#ref-chGii6yw" role="doc-biblioref">51</a>]</span>, labome <span class="citation" data-cites="rhRRCtlA">[<a href="#ref-rhRRCtlA" role="doc-biblioref">142</a>]</span>, CellFinder <span class="citation" data-cites="4AEy2xhQ">[<a href="#ref-4AEy2xhQ" role="doc-biblioref">137</a>]</span> and SHOGoiN/CELLPEDIA <span class="citation" data-cites="6uWWsiSq">[<a href="#ref-6uWWsiSq" role="doc-biblioref">143</a>]</span>) (if the owners give proper authorization).
The approach we took here can be applied to any knowledge set of public interest, providing a low-cost and low-barrier platform for sharing biocurated knowledge in gold-standard format.</p>
<h2 class="page_break_before" id="wikidata-bib-and-a-professional-system-for-biocuration">Wikidata Bib and a professional system for biocuration</h2>
<h3 id="introduction-1">Introduction</h3>
<p>Reading scientific articles is an integral part of the routine of modern scientists.
Although several literature-management software are available <span class="citation" data-cites="SdbIYKUa">[<a href="#ref-SdbIYKUa" role="doc-biblioref">144</a>]</span>, the process of reading is mainly artisanal.
There are no standard guidelines on how to probe the literature organize notes for biomedical researchers.
Thus, while reading and studying is a core activity, there are few (if any) protocols for the efficient screening of scientific articles.</p>
<p>Other professional traditions have dealt with similar issues in the past.
Notetaking is vital to keep track of financial balances and avoid costly problems in accounting.
Double-entry bookkeeping was developed in the 13th century as a professional solution for notetaking in accounting where “every entry to an account requires a corresponding and opposite entry to a different account.” <span class="citation" data-cites="uYuz0opI">[<a href="#ref-uYuz0opI" role="doc-biblioref">145</a>, =Double-entry_bookkeeping&amp;oldid=1055066428]</span>
In software development, Test-Driven Development (TDD) is a popular methodology where tests for code snippets are written before the code itself, therefore ensuring that written software passes minimum quality standards.
The similarities of Double-entry bookkeeping and TDD are diverse <span class="citation" data-cites="KNGv5MPp">[<a href="#ref-KNGv5MPp" role="doc-biblioref">146</a>]</span>, but for our purpose, here suffices to see both as professionalized systems that promote better quality and accountability of works.</p>
<p>In the humanities, there is a well-established practice of annotations of readings.
The annotation skills are part of standard academic training in the humanities <span class="citation" data-cites="rPKBwmYh">[<a href="#ref-rPKBwmYh" role="doc-biblioref">147</a>]</span><span class="citation" data-cites="url">[<a href="#ref-url" role="doc-biblioref"><strong>url?</strong></a>]</span>.
An influential work in presenting methods for academic reading in the humanities is Umberto Eco’s book “How to Write a Thesis” <span class="citation" data-cites="1HBVPtZGp">[<a href="#ref-1HBVPtZGp" role="doc-biblioref">148</a>]</span>, which outlines not only <em>how</em> to annotate the literature that basis an academic thesis, but also <em>why</em> to do so.
The book, written originally in 1977, is still influential today.
Still, its theoretical scope (roughly the humanities) and its date preceding the digital era limits the extent to which it applies to the biomedical sciences.</p>
<p>Notably, the need for an organized reading system for biocuration studies stems from a difference in methodology.
In humanities, the main (if not sole) research material is the written text, the books and articles from which research stems—<span class="citation" data-cites="url">[<a href="#ref-url" role="doc-biblioref"><strong>url?</strong></a>]</span>.
In the biomedical sciences, including a large part of bioinformatics, the object of study is the natural world, observed via experimentation.
Thus, naturally, scientific training focuses on experimentation and data analysis’s theoretical and practical basis.
With the boom of scientific articles, however, the scientific literature (and accompanying public datasets) already provide a strong material for sculpting scientific projects.
Thus, developing a methodology for academic reading tailored to the digital environment is a need.</p>
<p>This chapter concerns itself with presenting Wikidata Bib, a framework for large scale reading of scientific articles.
It is presented in three parts, each with a technical overview alongside the theoretical foundations.
First, Wikidata Bib is presented as a reading system for managing references and notes using a GitHub repository and plain text notes.
Then, we present how the system ensures accountability, allowing users to get personalized analytics on their reading patterns.
Finally, we demonstrate how Wikidata Bib fits an active curation environment, connecting the framework with the larger goal of this project of curating information about cell types on Wikidata.</p>
<h2 id="wikidata-bib-as-a-reading-system">Wikidata Bib as a reading system</h2>
<p>The reading framework of Wikidata bib is built upon a git repository integrated with GitHub, Python 3 scripts and SPARQL queries.
It has a standard file structure, summarized as the following:</p>
<ul>
<li><code>docs/</code>
<ul>
<li><code>index.html</code></li>
</ul></li>
<li><code>downloads/</code>
<ul>
<li><code>10.7554_ELIFE.52614.pdf</code></li>
</ul></li>
<li><code>notes/</code>
<ul>
<li><code>Q87830400.md</code></li>
</ul></li>
<li><code>src/</code>
<ul>
<li><code>get_pdf.py</code></li>
<li><code>helper.py</code></li>
<li><code>read_paper.py</code></li>
<li><code>update_dashboard.py</code></li>
</ul></li>
<li><code>index.md</code></li>
<li><code>toread.md</code></li>
<li><code>config.yaml</code></li>
<li><code>pop</code></li>
<li><code>wadd</code></li>
<li><code>wadd_all</code></li>
<li><code>wread</code></li>
<li><code>wlog</code></li>
</ul>
<p>The <code>docs/</code> directory contains the live dashboard from the readings, which will be discussed in the following sessions.
The <code>downloads/</code> directory hosts the pdfs of the articles read with the system.
These are not committed to the repository and are only stored locally.
The <code>notes/</code> directory contains markdown files, one for each article read.
The <code>src/</code> directory contains the python code with the system’s mechanics.
They contain helper functions for the command line commands discussed below:
- <code>wread</code> which receives a Wikidata QID for an article and outputs (1) a notes document, (2) a pdf for the paper obtained from Unpaywall <span class="citation" data-cites="15luL9zZC">[<a href="#ref-15luL9zZC" role="doc-biblioref">149</a>]</span> and (3) an updated version of the dashboard HTML files in the <code>docs/</code> directory.
- <code>pop</code>, which “pops” an article from <code>toread.md</code> and runs <code>wread</code> for it
- <code>wadd</code>, which takes an URL for a Wikidata SPARQL query and adds new QIDs to <code>toread.md</code>
- <code>wadd_all</code>, which parses <code>config.yaml</code> for recurrent SPARQL queries and runs <code>wadd</code> for each
- <code>wlog</code>, which adds, commits and pushes recent readings and dashboard updates to GitHub</p>
<p>All the structures described so far are commonly shared by any user of Wikidata Bib.
To personalize the use of the system, the user edits three plain text files.
<code>toread.md</code> hosts plain text QIDs of the articles that will be read.
These can be added either manually or via wadd.
While the <code>pop</code> command only sees QIDs, articles titles or other identifiers can temporarily be added to <code>toread.md</code> without breaking the system.
<code>index.md</code> hosts a numbered list of topics of interest.
This file plays the role of Umberto Eco’s work plan, with the topics of interest for the academic. <span class="citation" data-cites="1HBVPtZGp">[<a href="#ref-1HBVPtZGp" role="doc-biblioref">148</a>]</span>
These are used to tag articles for retrieval in a later step.
<code>config.yaml</code> contains shortcuts for different reading lists.
This is better explained by example.
In my <code>to read.md</code> file there are two reading lists, one following a <code># Cell types</code> header and another following a <code># Biocuration</code> header.
My <code>config.yaml</code> contains the following snippet:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode yaml"><code class="sourceCode yaml"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lists</span><span class="kw">:</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># - shortcut: Title of header in toread.md  </span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">ct</span><span class="kw">:</span><span class="at"> Cell types</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="at">  </span><span class="fu">bioc</span><span class="kw">:</span><span class="at"> Biocuration</span></span></code></pre></div>
<p>The <code>config.yaml</code> shortcuts are used as arguments by the <code>pop</code> command, where <code>$ ./pop ct</code> retrieves an article from the “Cell types” list, while <code>$ ./pop bioc</code> retrieves an article from the “Biocuration” list.</p>
<p>The Wikidata bib framework is coupled with a discipline of daily reading.
The discipline is inspired by Robert Cecil Martin’s description of Test Driven Development in the book “Clean Code”, which includes not only a technical description but a <em>school of thought</em> of how software development might be approached. <span class="citation" data-cites="13HqB23xH">[<a href="#ref-13HqB23xH" role="doc-biblioref">150</a>]</span>
Every day, I read one article of each list, using the notetaking station displayed in Figure <span class="citation" data-cites="fig">[<a href="#ref-fig" role="doc-biblioref"><strong>fig?</strong></a>]</span>: notetaking.
The constancy of reading allows steady coverage of the relevant literature.
While the discipline has worked for this research project, it is not required to use the Wikidata Bib system.</p>
<p>The notetaking station of Wikidata Bib, opened in Virtual Studio Code, is depicted on Figure <span class="citation" data-cites="fig">[<a href="#ref-fig" role="doc-biblioref"><strong>fig?</strong></a>]</span>: notetaking A.
The title and publication dates are displayed, and the reading process entails copying snippets from the text to the “Highlights” session.
Copying the highlights into plain text makes the sections of interest searchable via command line using <code>grep</code> (https://en.wikipedia.org/w/index.php?title=Grep&amp;oldid=1039541979).
Comments can be added either in the comment section or inline, alongside the highlights, using <code>--&gt; Comment goes here</code> to differentiate from highlights.
Also searchable by <code>grep</code> are the tags, copied and pasted from <code>index.md</code> in the <code>## Tags</code> session or alongside the main article.</p>
<p>The discipline also includes, whenever possible, an improvement of the metadata about the article on Wikidata.
In <span class="citation" data-cites="fig">[<a href="#ref-fig" role="doc-biblioref"><strong>fig?</strong></a>]</span>: notetaking B are shown the links included in the dashboard.
A link to a Scholia <span class="citation" data-cites="hxzL9pmm">[<a href="#ref-hxzL9pmm" role="doc-biblioref">108</a>]</span> profile allows identification of related articles from a series of pre-made SPARQL queries probing bibliography data on Wikidata.
While Scholia provides an overview of a given article, it does not allow direct curation of the metadata.
For that, two links are provided, one to Wikidata and one to Author Disambiguator <span class="citation" data-cites="1A9RvszKC">[<a href="#ref-1A9RvszKC" role="doc-biblioref">151</a>]</span>.
By accessing the Wikidata page for the entity, one can add new triples, for example, curating authors and topics of the article, which are then used by Scholia and by Wikidata Bib’s dashboard.
Author Disambiguator is a wrapper of an Wikimedia API that facilitates disambiguating author names to unique identifiers on Wikidata, thus feeding the public knowledge graph of publication and authors.<br />
Finally, a link to the article’s DOI or full-text URL is provided and serves as a fallback when the automatic download fails.
Of note, while the metadata curation has a technical benefit to Wikidata and the dashboard, it also plays a theoretical role.
By curating metadata on authors, the user of Wikidata Bib can better understand the people they read, and expand their metascientific perspective on their domain of interest.</p>
<div id="fig:notetaking" class="fignos">
<figure>
<img src="images/note_taking_station_annotated_with_links.png" alt="Figure 8: Wikidata Bib’s platform for note taking" /><figcaption aria-hidden="true"><span>Figure 8:</span> Wikidata Bib’s platform for note taking</figcaption>
</figure>
</div>
<p>The source code for Wikidata Bib is available at <a href="https://github.com/lubianat/wikidata_bib" class="uri">https://github.com/lubianat/wikidata_bib</a>.</p>
<h2 id="wikidata-bib-as-a-dashboard">Wikidata Bib as a dashboard</h2>
<p>The Wikidata Bib system also enables the reader to get statistics on their readings.
Two simple databases are stored on the GitHub repository:
* <code>read.ttl</code> - An RDF document recording the dates in which each article was read.
* <code>read.csv</code> - An simple, human-readable index connecting QIDs with article titles.
The CSV file is only stored for accountability and as a quick way to glance at the titles read.
The .ttl file, on the other hand, is processed by the <code>update_dashboard.py</code> script to render 4 different HTML files under the <code>docs/</code> folder:
- <code>index.html</code>
- <code>last_day.html</code>
- <code>past_week.html</code>
- <code>past_month.html</code>
All files are displayed in a GitHub pages.
In the case of this work, they are displayed at <a href="https://lubianat.github.io/wikidata_bib/" class="uri">https://lubianat.github.io/wikidata_bib/</a>.</p>
<p>To organize the code for rendering the dashboard, we created a python package, wbib, and deposited it in PyPi, making it available via <code>pip</code>. <span class="citation" data-cites="6chnW6cc">[<a href="#ref-6chnW6cc" role="doc-biblioref">152</a>]</span>.
The package implements the logic for rendering complex Wikidata-based academic dashboards and is available in GitHub at <a href="https://github.com/lubianat/wbib" class="uri">https://github.com/lubianat/wbib</a>.
It allows the user to build dashboards based on Wikidata records of information such as gender of authors, the region of author’s institutions, topics of articles and similar metascientific information.
The dashboard is composed of SPARQL queries written for the Wikidata Query Service <span class="citation" data-cites="2wDsXBwd">[<a href="#ref-2wDsXBwd" role="doc-biblioref">87</a>]</span>
It also allows users to feed an arbitrary list of articles and obtain a custom dashboard.
Wikidata Bib obtains the HTML dashboards after feeding wbib the lists of articles read in total (<code>index.html</code>) or in pre-determined time spans (<code>last_day.html</code>, <code>past_week.html</code> and <code>past_month.html</code> )</p>
<div id="fig:dashboard" class="fignos">
<figure>
<img src="images/wikidata_bib_display.png" style="width:85.0%" alt="Figure 9: Wikidata Bib queries for institutions of authors and most-read venues" /><figcaption aria-hidden="true"><span>Figure 9:</span> Wikidata Bib queries for institutions of authors and most-read venues</figcaption>
</figure>
</div>
<p>The dashboard includes not only a basic list of read articles, but also statistics on most read authors and most-read venues.
It also displays an interactive map of the institutions of articles read, permitting a glance at geographic biases in activities.
An example of queries is shown in <a href="#fig:dashboard">9</a>.
As the queries are rendered live, they evolve in quality with the growth of Wikidata.
Finally, the clean 5-star-open data format enables users to adapt the queries to include different aspects of Wikidata.
For example, table <a href="#tbl:articles_read_hca">4</a> showcases 10 articles that (1) I have read in the past year and (2) were authored by a speaker of the 1st Human Cell Atlas Latin America Single Cell RNA-seqData Analysis Workshop <span class="citation" data-cites="1hag8XE6">[<a href="#ref-1hag8XE6" role="doc-biblioref">153</a>]</span>.
One practical application that the dashboard enables, thus, is to identify people in an event, institution or location that the user has read before, therefore catalyzing the possibility of collaborations.
Anecdotally, this strategy was tested successfully at Biohackathon Europe 2021 <span class="citation" data-cites="kHL3NVxk">[<a href="#ref-kHL3NVxk" role="doc-biblioref">154</a>]</span>, where I used the system both to identify possible collaborators and as a conversation starter.</p>
<div id="tbl:articles_read_hca" class="tablenos">
<table id="tbl:articles_read_hca">
<caption><span>Table 4:</span> Articles read by Tiago Lubiana before 8 December 2021 in which an author was a speaker at HCA Latin America
 |</caption>
<colgroup>
<col style="width: 56%" />
<col style="width: 43%" />
</colgroup>
<thead>
<tr class="header">
<th>workLabel</th>
<th>authors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>A promoter-level mammalian expression atlas</td>
<td>Jay W Shin</td>
</tr>
<tr class="even">
<td>Single-cell RNA-seq reveals new types of human blood dendritic cells, monocytes, and progenitors.</td>
<td>Muzlifah Haniffa</td>
</tr>
<tr class="odd">
<td>The Human Cell Atlas.</td>
<td>Musa Mhlanga, Jay W Shin, Muzlifah Haniffa, Menna R Clatworthy, Dana Pe’er</td>
</tr>
<tr class="even">
<td>The Human Cell Atlas: Technical approaches and challenges.</td>
<td>Jay W Shin</td>
</tr>
<tr class="odd">
<td>Innate Immune Landscape in Early Lung Adenocarcinoma by Paired Single-Cell Analyses.</td>
<td>Dana Pe’er</td>
</tr>
<tr class="even">
<td>Single cell RNA sequencing of human liver reveals distinct intrahepatic macrophage populations</td>
<td>Sonya A MacParland</td>
</tr>
<tr class="odd">
<td>Single-cell reconstruction of the early maternal-fetal interface in humans</td>
<td>Muzlifah Haniffa</td>
</tr>
<tr class="even">
<td>Distinct microbial and immune niches of the human colon</td>
<td>Rasa Elmentaite, Menna R Clatworthy</td>
</tr>
<tr class="odd">
<td>A cell atlas of human thymic development defines T cell repertoire formation</td>
<td>Muzlifah Haniffa, Menna R Clatworthy</td>
</tr>
<tr class="even">
<td>Decoding human fetal liver haematopoiesis</td>
<td>Muzlifah Haniffa</td>
</tr>
</tbody>
</table>
</div>
<h2 id="wikidata-bib-for-curation-of-cells-to-wikidata">Wikidata Bib for curation of cells to Wikidata</h2>
<p>The Wikidata Bib system was devised originally to allow an overview of the fields of cell classification and biocuration.
However, during the process, it was also repurposed for biocuration of new cell classes in Wikidata.
By fast-tracking the reading of new articles, Wikidata Bib enables an efficient parsing of the literature and, thus, the identification of previously uncatalogued cell types.</p>
<p>Articles read with Wikidata Bib were screened to mention cell types absent from Wikidata.
As discussed in the chapter about the concept of cell type, we considered a “cell type” as any class of cells described by a domain expert with evidence of the reality of its instances.
When a mention of such a class appears in an article, I first verify Wikidata for the existence of a related class.
If it is absent from the platform, I enter a class name, alongside a superclass, and a QID in a Google Spreadsheet, as shown in Figure <a href="#fig:biocuration_of_cells">10</a>.</p>
<p>The information from the spreadsheet is pulled by a python script and processed locally with a series of dictionaries that match common terms to Wikidata IDs.
In the example shown in Figure <a href="#fig:biocuration_of_cells">10</a>, the string “endothelial cell” was matched against a manually curated dictionary to the Wikidata entry <a href="https://www.wikidata.org/wiki/Q11394395">Q11394395</a>, the representation of that concept on Wikidata.
After reconciling the data, the script uses the Wikidata Integrator python package <span class="citation" data-cites="qDI8I4IJ">[<a href="#ref-qDI8I4IJ" role="doc-biblioref">95</a>]</span> to insert the new entries on the Wikidata database.
The code for integrating a Google Spreadsheet to Wikidata is available at <a href="https://github.com/lubianat/wikidata_cell_curation" class="uri">https://github.com/lubianat/wikidata_cell_curation</a>.</p>
<div id="fig:biocuration_of_cells" class="fignos">
<figure>
<img src="images/biocuration_of_cells.png" style="width:85.0%" alt="Figure 10: Wikidata Bib was coupled with a biocuration framework for cell types" /><figcaption aria-hidden="true"><span>Figure 10:</span> Wikidata Bib was coupled with a biocuration framework for cell types</figcaption>
</figure>
</div>
<p>Wikidata contains 2940 subclasses of “cell (<a href="https://www.wikidata.org/wiki/Q7868">Q7868</a>)” as of 8 December 2021.
From those, 550 cell classes are specific for humans, and 318 are specific for mice.<br />
As a comparison, as of 8 December 2021, Wikidata has more cell classes than the Cell Ontology, which lists 2577 classes.
It is worth noticing that classes on the Cell Ontology are added after careful consideration by ontologists and domain experts and should be considered of higher quality than the ones on Wikidata.</p>
<p>From the 2940 cell classes on Wikidata, 2812 (95.6%) have been edited somehow by User:TiagoLubiana, and 1668 (56.7%) have been created by User:TiagoLubiana.
Edits made to the cells were often connecting a dangling term, created automatically from an Wikipedia page to the cell subclass hierarchy, and included adding identifiers, images, markers, and other pieces of information.
From the 1668 entities created, approximately 63 species-neutral cell types, 188 human and 188 mouse cell types were added based on PanglaoDB entries (total of 439).
The remaining 1229 entries were created either via Wikidata’s web interface or via the curation workflow described in this chapter.
These statistics are a simple demonstration of how the curation system efficiently contributes to the status of cell type information on Wikidata.</p>
<p>As mentioned by Aviv Regev in the Human Cell Atlas General Meeting 2021, after a shoutou to ontologies: “It’s everyone’s collective
responsibility to participate in the annotation efforts, because that relies on domain expertise. To really tease apart things and give
them names. Until we have names, people will have really a hard time working with things in biology.”<span class="citation" data-cites="url">[<a href="#ref-url" role="doc-biblioref"><strong>url?</strong></a>]</span>”
We hope that by developing simplified curation tools we will engage more domain experts into the curation efforts.</p>
<h2 id="wikidata-and-the-cell-ontology-interplay">Wikidata and the Cell Ontology interplay</h2>
<p>The contributions to cell types on Wikidata will be of most value if they are integrated to the current state-of-art of knowledge representation.
Arguably, the Cell Ontology is the main source of cell type identifiers in the context of the Human Cell Atlas project.<span class="citation" data-cites="qT8WxqjA">[<a href="#ref-qT8WxqjA" role="doc-biblioref">42</a>]</span>
Thus, data about cell types on Wikidata must be connected to the Cell Ontology.</p>
<p>To start the improvement in the interplay of both databases, we proposed and got the approval of a specific Wikidata identifier for the Cell Ontology, the “Cell Ontology ID” (<a href="https://www.wikidata.org/wiki/Property:P7963" class="uri">https://www.wikidata.org/wiki/Property:P7963</a>).
IDs can be added to Wikidata entities and connected them to external databases enabling integrative SPARQL queries.
Besides using the common Wikidata interface, one can crowd-curate identifiers via a 3rd-party service, Mix’N’Match, which provides a user-friendly framework for connecting identifier catalogues to Wikidata. <span class="citation" data-cites="JgiKEEdq">[<a href="#ref-JgiKEEdq" role="doc-biblioref">155</a>/?p=114]</span>, as seen in Figure ??.
Logically, we created a Mix’N’Match catalogue for harmonizing Cell Ontology IDs to Wikidata (<a href="https://mix-n-match.toolforge.org/#/catalog/4719" class="uri">https://mix-n-match.toolforge.org/#/catalog/4719</a>), harnessing the community support for the task.</p>
<div id="fig:mixnmatch_cl" class="fignos">
<figure>
<img src="https://pointstodots.files.wordpress.com/2021/09/image-17.png" style="width:85.0%" alt="Figure 11: Mix’N’Match curation system" /><figcaption aria-hidden="true"><span>Figure 11:</span> Mix’N’Match curation system</figcaption>
</figure>
</div>
<p>As of early December 2021, more than 700 Cell Ontology IDs have been manually matched to Wikidata.
The integration already enables queries that harness the previously existing information on Wikidata for Cell Ontology-based applications.
For example, one can query Wikidata items that have (1) a crossref to a CL ID (2) a picture in Wikimedia Commons (<a href="https://w.wiki/4F6e" class="uri">https://w.wiki/4F6e</a>, Figure <a href="#fig:cl_images">12</a>).
The different possibilities of mutual benefit between the Cell Ontology and Wikidata will continue to be explored in the following years of this PhD project.</p>
<div id="fig:cl_images" class="fignos">
<figure>
<img src="https://user-images.githubusercontent.com/7917951/137942026-7645f368-d62a-4434-be05-083555cf0757.png" style="width:85.0%" alt="Figure 12: Entries on Wikidata with a depicting image and a Cell Ontology ID" /><figcaption aria-hidden="true"><span>Figure 12:</span> Entries on Wikidata with a depicting image and a Cell Ontology ID</figcaption>
</figure>
</div>
<h1 id="final-considerations-and-next-steps">Final considerations and next steps</h1>
<p>To sum up, this PhD research project aims at improving knowledge representation in the context of the Human Cell Atlas.
It is composed of a mixture of theoretical studies on conceptual modelling, practical contributions to knowledge organization projects (mainly the Cell Ontology and Wikidata), explorations of the data to generate biomedical insights, and a technical framework for organized reading.
By approaching the object of study from a new perspective, we hope to make sizeable contributions and promote discussion and fruitful conflation of approaches.</p>
<p>The next years of study will be devoted to improving the projects presented here into mature, useful objects.
We hope to improve the interplay of Wikidata and Cell Ontology, developing frameworks to combine community- and expert-based curation of knowledge on cell types.
Furthermore, we plan to integrate Wikidata to current single-cell RNA-sequencing pipelines by adapting R packages to use Wikidata (e.g. the ontology-based packages OnClass <span class="citation" data-cites="sW6aNZJB">[<a href="#ref-sW6aNZJB" role="doc-biblioref">56</a>]</span> and ontoProc<span class="citation" data-cites="15YmDXALp">[<a href="#ref-15YmDXALp" role="doc-biblioref">57</a>]</span>).
Finally, we aim at moving the Wikidata Bib system to a well documented, user-friendly mature system, testing usability with other academics and distributing it as a durable open-source project.</p>
<h1 class="page_break_before" id="additional-work">Additional Work</h1>
<h2 id="collaborations-and-manuscripts">Collaborations and manuscripts</h2>
<h3 id="fcoex">fcoex</h3>
<p>During the initial course of this PhD work, we also completed the development and reporting of <em>fcoex</em>, an R package for investigating cellular phenotypes using co-expression networks. <span class="citation" data-cites="MxIeSJYt">[<a href="#ref-MxIeSJYt" role="doc-biblioref">156</a>]</span> The software was maintained to withstand new releases of dependencies and new R version and was published as a preprint on biorxiv. <span class="citation" data-cites="doi:10.1101/2021.12.07.471603v1">[<a href="#ref-doi:10.1101/2021.12.07.471603v1" role="doc-biblioref"><strong>doi:10.1101/2021.12.07.471603v1?</strong></a>]</span></p>
<h3 id="wikidata-bots">Wikidata Bots</h3>
<p>Alongside the editing of cell-type information on Wikidata, I have joined different efforts to improve biological information on Wikidata.
I have collaborated with the ComplexPortal curators as part of the Virtual Elixir BioHackathon 2020 (https://github.com/virtual-biohackathons/covid-19-bh20/wiki) and for the following year, to build a Wikidata Bot to integrate information on protein complexes to Wikidata. An overview of the Wikidata integration is in Figure <a href="#fig:complexportal">13</a>, presented in an article published in Nucleic Acid Research (re-use of the image and legend possible under the CC-BY license of the article). <span class="citation" data-cites="CQRJ53gu">[<a href="#ref-CQRJ53gu" role="doc-biblioref">157</a>]</span></p>
<div id="fig:complexportal" class="fignos">
<figure>
<img src="https://oup.silverchair-cdn.com/oup/backfile/Content_public/Journal/nar/PAP/10.1093_nar_gkab991/2/gkab991fig3.jpeg?Expires=1641821957&amp;Signature=RK-es18S~Qh6vGQE~61i6u4prMuij8kVTbrjN6WUJLfYHOAhUhx9qQorBxROohjLLxbHvZ2YK9e7EwlI9HjVeNoGZ2PJs0Pv78Y31MdZLY8FeLYI2E4azwrqRyv9q0AH8QL3RorWZV1AhOb9bl-44Mr97Q~9MWzeTDnQQbxpCnGLG~YoG49kocD5KE~dmTSQdkXBU7kZnuGM1NPqMHo5ZDUoCRFwmTbLvd4kXoH~6CTyqx4ruQRIO-ks4Q0FUjYnxGX18gCFvoO2LMPZUPrQjTgD35RrVAkLs91Ur6BQHUuZ-ZsE3Iag3mBDu27vT9Jr93bOj0i9wz~surTtEGATgw__&amp;Key-Pair-Id=APKAIE5G5CRDK6RD3PGA" alt="Figure 13: Complex Portal and Wikidata. (A) example of an entry assertion in Wikidata with provenance pointing to Complex Portal (Q104836061). (B) The number of protein complexes in Wikidata per taxon (https://w.wiki/3ggX). (C) Subset of Wikidata connected to the SARS-CoV-2 polymerase complex (https://w.wiki/3eta)." /><figcaption aria-hidden="true"><span>Figure 13:</span> Complex Portal and Wikidata. (A) example of an entry assertion in Wikidata with provenance pointing to Complex Portal (Q104836061). (B) The number of protein complexes in Wikidata per taxon (<a href="https://w.wiki/3ggX" class="uri">https://w.wiki/3ggX</a>). (C) Subset of Wikidata connected to the SARS-CoV-2 polymerase complex (<a href="https://w.wiki/3eta" class="uri">https://w.wiki/3eta</a>).</figcaption>
</figure>
</div>
<p>I have also collaborated with the Cellosaurus database <span class="citation" data-cites="1DguATd9G">[<a href="#ref-1DguATd9G" role="doc-biblioref">158</a>]</span> to revive the CellosaurusBot <span class="citation" data-cites="lMQxhx3q">[<a href="#ref-lMQxhx3q" role="doc-biblioref">159</a>]</span>, responsible for updating the metadata on more than 100,000 cell lines on Wikidata. The bot code, written in Python, was refactored entirely and runs semi-automatically after the Cellosaurus database was released. A write-up of the integration is in progress and is planned for release/submission in the first semester of 2022.</p>
<h3 id="systematic-reviews-and-publishing-of-intermediary-tables">Systematic Reviews and publishing of intermediary tables</h3>
<p>Finally, in collaboration with Olavo Amaral and Kleber Neves, from the Brazilian Reproducibility Initiative <span class="citation" data-cites="F2mYjDJ0">[<a href="#ref-F2mYjDJ0" role="doc-biblioref">160</a>]</span> I wrote a commentary on the value of publishing intermediate datasets as citable products. <span class="citation" data-cites="oXMFSv5v">[<a href="#ref-oXMFSv5v" role="doc-biblioref">161</a>]</span>
The pieces discuss the value of small curations done both in systematic reviews and by experimentalists in the course of their research projects. Published curation tables can serve as a source for improving the ecosystem of open knowledge, not less by reconciliation to Wikidata (thereby bridging the commentary with this project)</p>
<h3 id="wisecube---enterprise-biomedical-question-and-answering">WiseCube - enterprise biomedical question and answering</h3>
<p>During a part of this project, I have worked part-time as a consultant for the Wisecube company, based in Seattle, United States. <span class="citation" data-cites="gdYsBE7d">[<a href="#ref-gdYsBE7d" role="doc-biblioref">162</a>]</span>
The job was approved by FAPESP and consisted mainly in writing SPARQL queries that probe Wikidata for answers to the questions posed by the BioASQ competition. <span class="citation" data-cites="rkXotO9x">[<a href="#ref-rkXotO9x" role="doc-biblioref">163</a>]</span>
It also entails on-demand curation of biomedical topics on Wikidata based on requests by pharmaceutical companies as well as the development of dashboards targeted at providing insights to customers.</p>
<h2 id="awards-and-participation-in-events">Awards and participation in events</h2>
<p>During the initial course of this PhD project, I have participated in several events:</p>
<ul>
<li>(Feb-2021) Presented an open talk at the “Semana da Bioinformática” event about modelling of biological systems (1020 views as of December 2021) <span class="citation" data-cites="url">[<a href="#ref-url" role="doc-biblioref"><strong>url?</strong></a>“https://www.youtube.com/watch?v=VDvCxskIGEI”]</span></li>
<li>(Jun-Aug 2021) Helped to organize the No-Budget-Science HackWeek virtual hackathon <span class="citation" data-cites="12LOzmXRs">[<a href="#ref-12LOzmXRs" role="doc-biblioref">164</a>]</span></li>
<li>(Jul - 2021) Presented the work “Wikidata for 5-star Linked Open Databases: A case study of PanglaoDB” at the Bio-Ontologies section of the Annual International Conference on Intelligent Systems for Molecular Biology. <span class="citation" data-cites="doi:10.5281/zenodo.5747849">[<a href="#ref-doi:10.5281/zenodo.5747849" role="doc-biblioref"><strong>doi:10.5281/zenodo.5747849?</strong></a>]</span>. The presentation was awarded the best</li>
<li>(Jul - 2021) Awarded the 2nd place in the International Society for Computational Biology (ISCB) Wikipedia Competition for the contributions to the Wikipedia page on Biocuration (<a href="https://en.wikipedia.org/wiki/Biocuration" class="uri">https://en.wikipedia.org/wiki/Biocuration</a>) <span class="citation" data-cites="IJG65hFm">[<a href="#ref-IJG65hFm" role="doc-biblioref">165</a>]</span></li>
<li>(Nov - 2021) Managed a project during BioHackathon Europe 2021, in Barcelona, Spain, on the representation of ELIXIR information on Wikidata. <span class="citation" data-cites="14Wi842eZ">[<a href="#ref-14Wi842eZ" role="doc-biblioref">166</a>]</span></li>
</ul>
<h2 id="course-work">Course work</h2>
<p>During the first year of the PhD program, I took four different classes, acquiring 36 academic credits.
Figure <a href="#fig:courses_taken">14</a> displays the disciplines taken, available only in Portuguese.</p>
<div id="fig:courses_taken" class="fignos">
<figure>
<img src="images/janus_courses_taken.png" alt="Figure 14: Courses taken during this PhD project" /><figcaption aria-hidden="true"><span>Figure 14:</span> Courses taken during this PhD project</figcaption>
</figure>
</div>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-1GmbExweg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>The Human Cell Atlas.</strong> <div class="csl-block">Aviv Regev, Sarah Teichmann, Eric Lander, Amir Giladi, Christophe Benoist, Ewan Birney, Bernd Bodenmiller, Peter Campbell, Piero Carninci, Menna R Clatworthy, … Human Cell Atlas Meeting Participants</div> <em>eLife</em> (2017-12-05) <a href="https://www.wikidata.org/wiki/Q46368626">https://www.wikidata.org/wiki/Q46368626</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7554/elife.27041">10.7554/elife.27041</a></div></div>
</div>
<div id="ref-tjdjR2Xf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>The Human Cell Atlas and equity: lessons learned</strong> <div class="csl-block">Partha P Majumder, Musa M Mhlanga, Alex K Shalek</div> <em>Nature Medicine</em> (2020-10-01) <a href="https://www.wikidata.org/wiki/Q100491106">https://www.wikidata.org/wiki/Q100491106</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41591-020-1100-4">10.1038/s41591-020-1100-4</a></div></div>
</div>
<div id="ref-kkwRTArg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>The Human Cell Atlas White Paper</strong> <div class="csl-block">Aviv Regev, Sarah Teichmann, Orit Rozenblatt-Rosen, Michael JT Stubbington, Kristin Ardlie, Amir Giladi, Paola Arlotta, Gary D Bader, Christophe Benoist, Moshe Biton, … Human Cell Atlas Organizing Committee</div> (2018-10-11) <a href="https://www.wikidata.org/wiki/Q104450645">https://www.wikidata.org/wiki/Q104450645</a></div>
</div>
<div id="ref-1DSEIjFha" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>Everyone needs a data-management plan</strong> <div class="csl-block">Nature</div> (2018-03-15) <a href="https://www.wikidata.org/wiki/Q56524391">https://www.wikidata.org/wiki/Q56524391</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/d41586-018-03065-z">10.1038/d41586-018-03065-z</a></div></div>
</div>
<div id="ref-zDRzmIGu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>About the Data Coordination Platform</strong> <div class="csl-block">HCA Data Portal</div> <a href="https://data.humancellatlas.org/about/">https://data.humancellatlas.org/about/</a></div>
</div>
<div id="ref-kX6KnbUo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>Mapping the Human Body at the Cellular Level</strong> <div class="csl-block">HCA Data Portal</div> <a href="https://data.humancellatlas.org/">https://data.humancellatlas.org/</a></div>
</div>
<div id="ref-2A7P3sTH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Publications</strong> <a href="https://www.humancellatlas.org/publications/">https://www.humancellatlas.org/publications/</a></div>
</div>
<div id="ref-tvnuubVW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>Identifiers for the 21st century: How to design, provision, and reuse persistent identifiers to maximize utility and impact of life science data</strong> <div class="csl-block">Julie A McMurry, Nick Juty, Niklas Blomberg, Tony Burdett, Tom Conlin, Nathalie Conte, Melanie Courtot, John Deck, Michel Dumontier, Donal K Fellows, … Helen Parkinson</div> <em>PLOS Biology</em> (2017-06-29) <a href="https://www.wikidata.org/wiki/Q33037209">https://www.wikidata.org/wiki/Q33037209</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pbio.2001414">10.1371/journal.pbio.2001414</a></div></div>
</div>
<div id="ref-paKiKTRS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>What Is Your Conceptual Definition of "Cell Type" in the Context of a Mature Organism?</strong> <div class="csl-block">Paul Blainey, Hans Clevers, Cole Trapnell, Ed Lein, Emma Lundberg, Alfonso Martinez Arias, Joshua R Sanes, Jay Shendure, James Eberwine, Junhyong Kim, … Mathias Uhlén</div> <em>Cell systems</em> (2017-03-01) <a href="https://www.wikidata.org/wiki/Q87649649">https://www.wikidata.org/wiki/Q87649649</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cels.2017.03.006">10.1016/j.cels.2017.03.006</a></div></div>
</div>
<div id="ref-sWv3okaH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>A periodic table of cell types</strong> <div class="csl-block">Bo Xia, Itai Yanai</div> <em>Development</em> (2019-06-15) <a href="https://doi.org/ggctwf">https://doi.org/ggctwf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1242/dev.169854">10.1242/dev.169854</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31249003">31249003</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6602355">PMC6602355</a></div></div>
</div>
<div id="ref-19yYJWEkz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Exciting times to study the identity and evolution of cell types</strong> <div class="csl-block">Maria Sachkova, Pawel Burkhardt</div> <em>Development</em> (2019-09-15) <a href="https://doi.org/ghdb9v">https://doi.org/ghdb9v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1242/dev.178996">10.1242/dev.178996</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31537583">31537583</a></div></div>
</div>
<div id="ref-BlxFp1v3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>The Human Cell Atlas: from vision to reality.</strong> <div class="csl-block">Orit Rozenblatt-Rosen, Michael JT Stubbington, Aviv Regev, Sarah Teichmann</div> <em>Nature</em> (2017-10-01) <a href="https://www.wikidata.org/wiki/Q47565008">https://www.wikidata.org/wiki/Q47565008</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/550451a">10.1038/550451a</a></div></div>
</div>
<div id="ref-eOUB1Bmj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>Human Cell Atlas and cell-type authentication for regenerative medicine</strong> <div class="csl-block">Yulia Panina, Peter Karagiannis, Andreas Kurtz, Glyn N Stacey, Wataru Fujibuchi</div> <em>Experimental and Molecular Medicine</em> (2020-09-15) <a href="https://www.wikidata.org/wiki/Q99418657">https://www.wikidata.org/wiki/Q99418657</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s12276-020-0421-1">10.1038/s12276-020-0421-1</a></div></div>
</div>
<div id="ref-TupzTvtA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>A community-based transcriptomics classification and nomenclature of neocortical cell types</strong> <div class="csl-block">Rafael Yuste, Michael J Hawrylycz, Nadia Aalling, Argel Aguilar-Valles, Detlev Arendt, Rubén Armañanzas, Giorgio A Ascoli, Concha Bielza, Vahid Bokharaie, Tobias B Bergmann, … Ed S Lein</div> <em>Nature Neuroscience</em> (2020-08-24) <a href="https://www.wikidata.org/wiki/Q98665291">https://www.wikidata.org/wiki/Q98665291</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41593-020-0685-8">10.1038/s41593-020-0685-8</a></div></div>
</div>
<div id="ref-w0Woz7JE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>The evolving concept of cell identity in the single cell era</strong> <div class="csl-block">Samantha A Morris</div> <em>Development</em> (2019-06-27) <a href="https://www.wikidata.org/wiki/Q93086971">https://www.wikidata.org/wiki/Q93086971</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1242/dev.169748">10.1242/dev.169748</a></div></div>
</div>
<div id="ref-srfuRMe" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Implications of Epigenetic Variability within a Cell Population for "Cell Type" Classification</strong> <div class="csl-block">Inna Tabansky, Joel Stern, Donald W Pfaff</div> <em>Frontiers in Behavioral Neuroscience</em> (2015-12-16) <a href="https://www.wikidata.org/wiki/Q26770736">https://www.wikidata.org/wiki/Q26770736</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fnbeh.2015.00342">10.3389/fnbeh.2015.00342</a></div></div>
</div>
<div id="ref-1GXQByEaQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>Geometry of the Gene Expression Space of Individual Cells</strong> <div class="csl-block">Yael Korem, Pablo Szekely, Yuval Hart, Hila Sheftel, Jean Hausser, Avi Mayo, Michael E Rothenberg, Tomer Kalisky, Uri Alon</div> <em>PLOS Computational Biology</em> (2015-07-10) <a href="https://www.wikidata.org/wiki/Q35688096">https://www.wikidata.org/wiki/Q35688096</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1004224">10.1371/journal.pcbi.1004224</a></div></div>
</div>
<div id="ref-k0Pe9YOB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Evolution of Cellular Differentiation: From Hypotheses to Models</strong> <div class="csl-block">Pedro Márquez-Zacarías, Rozenn M Pineau, Marcella Gomez, Alan Veliz-Cuba, David Murrugarra, William C Ratcliff, Karl J Niklas</div> <em>Trends in Ecology &amp; Evolution</em> (2020-08-20) <a href="https://www.wikidata.org/wiki/Q98633613">https://www.wikidata.org/wiki/Q98633613</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.tree.2020.07.013">10.1016/j.tree.2020.07.013</a></div></div>
</div>
<div id="ref-pNGap1Du" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>An era of single-cell genomics consortia</strong> <div class="csl-block">Yoshinari Ando, Andrew T Kwon, Jay W Shin</div> <em>Experimental and Molecular Medicine</em> (2020-09-15) <a href="https://www.wikidata.org/wiki/Q99418649">https://www.wikidata.org/wiki/Q99418649</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s12276-020-0409-x">10.1038/s12276-020-0409-x</a></div></div>
</div>
<div id="ref-MoaYZAA0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>Inferring cell type innovations by phylogenetic methods-concepts, methods, and limitations</strong> <div class="csl-block">Koryu Kin, Koryu Kin</div> <em>Journal of Experimental Zoology. Part B: Molecular and Developmental Evolution</em> (2015-10-14) <a href="https://www.wikidata.org/wiki/Q40436539">https://www.wikidata.org/wiki/Q40436539</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/jez.b.22657">10.1002/jez.b.22657</a></div></div>
</div>
<div id="ref-WAeCfntu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>Towards a pragmatic definition of cell type</strong> <div class="csl-block">Tiago Lubiana, Helder Nakaya</div> (2021-01-04) <a href="https://www.wikidata.org/wiki/Q108723646">https://www.wikidata.org/wiki/Q108723646</a> <div class="csl-block">DOI: <a href="https://doi.org/10.22541/au.160979530.02627436/v1">10.22541/au.160979530.02627436/v1</a></div></div>
</div>
<div id="ref-V85zvNGa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>How Many Types of Cells Are in the Human Body?</strong> <div class="csl-block">ibswit</div> (2017-05-17) <a href="https://askabiologist.asu.edu/questions/human-cell-types">https://askabiologist.asu.edu/questions/human-cell-types</a></div>
</div>
<div id="ref-SKmDXJlO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>How many cell types in a human body? How about the number of cell cycles in each species?</strong> <div class="csl-block">ResearchGate</div> <a href="https://www.researchgate.net/post/How-many-cell-types-in-a-human-body-How-about-the-number-of-cell-cycles-in-each-species">https://www.researchgate.net/post/How-many-cell-types-in-a-human-body-How-about-the-number-of-cell-cycles-in-each-species</a></div>
</div>
<div id="ref-WBDY0Uoy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>Types of cells in the human body</strong> <div class="csl-block">Kenhub</div> <a href="https://www.kenhub.com/en/library/anatomy/types-of-cells-in-the-human-body">https://www.kenhub.com/en/library/anatomy/types-of-cells-in-the-human-body</a></div>
</div>
<div id="ref-SprCx2w6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>BioNumbers--the database of key numbers in molecular and cell biology</strong> <div class="csl-block">Ron Milo, Paul Jorgensen, Uri Moran, Griffin M Weber, Michael Springer</div> <em>Nucleic Acids Research</em> (2010-01-01) <a href="https://www.wikidata.org/wiki/Q24643881">https://www.wikidata.org/wiki/Q24643881</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkp889">10.1093/nar/gkp889</a></div></div>
</div>
<div id="ref-jePJgR0H" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>Search BioNumbers - The Database of Useful Biological Numbers</strong> <a href="https://bionumbers.hms.harvard.edu/search.aspx">https://bionumbers.hms.harvard.edu/search.aspx</a></div>
</div>
<div id="ref-J5X2Xu6M" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>Cell types and ontologies of the Human Cell Atlas</strong> <div class="csl-block">David Osumi-Sutherland, Chuan Xu, Maria C Keays, Peter V Kharchenko, Aviv Regev, Ed S Lein, Sarah Teichmann</div> (2021-06-28) <a href="https://www.wikidata.org/wiki/Q107373831">https://www.wikidata.org/wiki/Q107373831</a></div>
</div>
<div id="ref-15efKvxi0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>The continued need for animals to advance brain research</strong> <div class="csl-block">Judith R Homberg, Roger AH Adan, Natalia Alenina, Antonis Asiminas, Michael Bader, Tom Beckers, Denovan P Begg, Arjan Blokland, Marilise E Burger, Gertjan van Dijk, … Lisa Genzel</div> <em>Neuron</em> (2021-08-01) <a href="https://www.wikidata.org/wiki/Q110088775">https://www.wikidata.org/wiki/Q110088775</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.neuron.2021.07.015">10.1016/j.neuron.2021.07.015</a></div></div>
</div>
<div id="ref-wvN4dtjQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Neuronal diversity: too many cell types for comfort?</strong> <div class="csl-block">Stevens CF</div> <em>Current Biology</em> (1998-10-01) <a href="https://www.wikidata.org/wiki/Q48373178">https://www.wikidata.org/wiki/Q48373178</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/s0960-9822(98)70454-3">10.1016/s0960-9822(98)70454-3</a></div></div>
</div>
<div id="ref-xWhvkd3B" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>An estimation of the number of cells in the human body</strong> <div class="csl-block">Eva Bianconi, Allison Piovesan, Federica Facchin, Alina Beraudi, Raffaella Casadei, Flavia Frabetti, Lorenza Vitale, Maria Chiara Pelleri, Simone Tassani, Francesco Piva, … Silvia Canaider</div> <em>Annals of Human Biology</em> (2013-07-05) <a href="https://www.wikidata.org/wiki/Q34037445">https://www.wikidata.org/wiki/Q34037445</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3109/03014460.2013.807878">10.3109/03014460.2013.807878</a></div></div>
</div>
<div id="ref-HFiosSKK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>A curated database reveals trends in single-cell transcriptomics</strong> <div class="csl-block">Valentine Svensson, Eduardo da Veiga Beltrame, Lior Pachter</div> <em>Database</em> (2020-11-01) <a href="https://www.wikidata.org/wiki/Q103034964">https://www.wikidata.org/wiki/Q103034964</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/database/baaa073">10.1093/database/baaa073</a></div></div>
</div>
<div id="ref-O95WN4x0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>The evolution of cell types in animals: emerging principles from molecular studies.</strong> <div class="csl-block">Detlev Arendt</div> <em>Nature reviews. Genetics</em> (2008-11) <a href="https://www.ncbi.nlm.nih.gov/pubmed/18927580">https://www.ncbi.nlm.nih.gov/pubmed/18927580</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nrg2416">10.1038/nrg2416</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18927580">18927580</a></div></div>
</div>
<div id="ref-ogAGO9KH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>The origin and evolution of cell types</strong> <div class="csl-block">Detlev Arendt, Jacob M Musser, Clare VH Baker, Aviv Bergman, Connie Cepko, Douglas H Erwin, Mihaela Pavlicev, Gerhard Schlosser, Stefanie Widder, Manfred D Laubichler, Günter P Wagner</div> <em>Nature Reviews Genetics</em> (2016-12) <a href="https://doi.org/f9b62x">https://doi.org/f9b62x</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nrg.2016.127">10.1038/nrg.2016.127</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27818507">27818507</a></div></div>
</div>
<div id="ref-14J9lsoMd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>Stem cell states, fates, and the rules of attraction.</strong> <div class="csl-block">Tariq Enver, Martin Pera, Carsten Peterson, Peter W Andrews</div> <em>Cell Stem Cell</em> (2009-05-01) <a href="https://www.wikidata.org/wiki/Q37475461">https://www.wikidata.org/wiki/Q37475461</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.stem.2009.04.011">10.1016/j.stem.2009.04.011</a></div></div>
</div>
<div id="ref-OjxQDZpx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>Theory of cell fate</strong> <div class="csl-block">Michael J Casey, Patrick S Stumpf, Ben D MacArthur</div> <em>Wiley interdisciplinary reviews. Systems biology and medicine</em> (2019-12-12) <a href="https://www.wikidata.org/wiki/Q91908361">https://www.wikidata.org/wiki/Q91908361</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/wsbm.1471">10.1002/wsbm.1471</a></div></div>
</div>
<div id="ref-19g5Iusf9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>Perspectives on defining cell types in the brain</strong> <div class="csl-block">Eran A Mukamel, John Ngai</div> <em>Current Opinion in Neurobiology</em> (2018-12-06) <a href="https://www.wikidata.org/wiki/Q90361677">https://www.wikidata.org/wiki/Q90361677</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.conb.2018.11.007">10.1016/j.conb.2018.11.007</a></div></div>
</div>
<div id="ref-4TJnL1xT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>Ensembles, dynamics, and cell types: Revisiting the statistical mechanics perspective on cellular regulation</strong> <div class="csl-block">Stefan Bornholdt, Stuart Kauffman</div> <em>Journal of Theoretical Biology</em> (2019-01-31) <a href="https://www.wikidata.org/wiki/Q91316993">https://www.wikidata.org/wiki/Q91316993</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.jtbi.2019.01.036">10.1016/j.jtbi.2019.01.036</a></div></div>
</div>
<div id="ref-g74dZjOM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><strong>Species Concepts and Species Delimitation</strong> <div class="csl-block">Kevin De Queiroz</div> <em>Systematic Biology</em> (2007-12-01) <a href="https://doi.org/c34kzf">https://doi.org/c34kzf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1080/10635150701701083">10.1080/10635150701701083</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18027281">18027281</a></div></div>
</div>
<div id="ref-1CI8uS6ka" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>An ontology for cell types</strong> <div class="csl-block">Jonathan Bard, Sue Rhee, Michael Ashburner</div> <em>Genome Biology</em> (2005-01-01) <a href="https://www.wikidata.org/wiki/Q21184168">https://www.wikidata.org/wiki/Q21184168</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/gb-2005-6-2-r21">10.1186/gb-2005-6-2-r21</a></div></div>
</div>
<div id="ref-gAysgnsS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>Logical Development of the Cell Ontology</strong> <div class="csl-block">Terrence F Meehan, Anna Maria Masci, Amina Abdulla, Lindsay G Cowell, Judith A Blake, Christopher J Mungall, Alexander D Diehl</div> <em>BMC Bioinformatics</em> (2011-12) <a href="https://doi.org/c7kw6x">https://doi.org/c7kw6x</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-12-6">10.1186/1471-2105-12-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21208450">21208450</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3024222">PMC3024222</a></div></div>
</div>
<div id="ref-17U3pGEjv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>The Cell Ontology 2016: enhanced content, modularization, and ontology interoperability</strong> <div class="csl-block">Alexander D Diehl, Terrence F Meehan, Yvonne M Bradford, Matthew H Brush, Wasila M Dahdul, David S Dougall, Yongqun He, David Osumi-Sutherland, Alan Ruttenberg, Sirarat Sarntivijai, … Christopher J Mungall</div> <em>Journal of Biomedical Semantics</em> (2016-12) <a href="https://doi.org/gg99b9">https://doi.org/gg99b9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13326-016-0088-7">10.1186/s13326-016-0088-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27377652">27377652</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4932724">PMC4932724</a></div></div>
</div>
<div id="ref-qT8WxqjA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>Cell type ontologies of the Human Cell Atlas</strong> <div class="csl-block">David Osumi-Sutherland, Chuan Xu, Maria Keays, Adam P Levine, Peter V Kharchenko, Aviv Regev, Ed Lein, Sarah Teichmann</div> <em>Nature Cell Biology</em> (2021-11-01) <a href="https://www.wikidata.org/wiki/Q109755180">https://www.wikidata.org/wiki/Q109755180</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41556-021-00787-7">10.1038/s41556-021-00787-7</a></div></div>
</div>
<div id="ref-jX2ZmXcc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>Cells in experimental life sciences - challenges and solution to the rapid evolution of knowledge</strong> <div class="csl-block">Sirarat Sarntivijai, Alexander D Diehl, Yongqun He</div> <em>BMC Bioinformatics</em> (2017-12) <a href="https://doi.org/gg99b7">https://doi.org/gg99b7</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-017-1976-2">10.1186/s12859-017-1976-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29322916">29322916</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5763506">PMC5763506</a></div></div>
</div>
<div id="ref-rK8KD0us" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline"><strong>Cells in ExperimentaL Life Sciences (CELLS-2018): capturing the knowledge of normal and diseased cells with ontologies</strong> <div class="csl-block">Sirarat Sarntivijai, Yongqun He, Alexander D Diehl</div> <em>BMC Bioinformatics</em> (2019-04) <a href="https://doi.org/gg99b8">https://doi.org/gg99b8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-019-2721-9">10.1186/s12859-019-2721-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31272374">31272374</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6509796">PMC6509796</a></div></div>
</div>
<div id="ref-16phbXNJE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline"><strong>Scaled, high fidelity electrophysiological, morphological, and transcriptomic cell characterization</strong> <div class="csl-block">Brian R Lee, Agata Budzillo, Kristen Hadley, Jeremy A Miller, Tim Jarsky, Katherine Baker, DiJon Hill, Lisa Kim, Rusty Mann, Lindsay Ng, … Jim Berg</div> <em>eLife</em> (2021-08-13) <a href="https://www.wikidata.org/wiki/Q109717199">https://www.wikidata.org/wiki/Q109717199</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7554/elife.65482">10.7554/elife.65482</a></div></div>
</div>
<div id="ref-WKbly37M" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline"><strong>Current best practices in single-cell RNA-seq analysis: a tutorial</strong> <div class="csl-block">Malte D Luecken, Fabian J Theis</div> <em>Molecular Systems Biology</em> (2019-06-19) <a href="https://www.wikidata.org/wiki/Q64974172">https://www.wikidata.org/wiki/Q64974172</a> <div class="csl-block">DOI: <a href="https://doi.org/10.15252/msb.20188746">10.15252/msb.20188746</a></div></div>
</div>
<div id="ref-ii1dVqoy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline"><strong>Tutorial: guidelines for annotating single-cell transcriptomic maps using automated and manual methods</strong> <div class="csl-block">Zoe A Clarke, Tallulah Andrews, Jawairia Atif, Delaram Pouyabahar, Brendan T Innes, Sonya A MacParland, Gary D Bader</div> <em>Nature Protocols</em> (2021-05-24) <a href="https://www.wikidata.org/wiki/Q107158224">https://www.wikidata.org/wiki/Q107158224</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41596-021-00534-0">10.1038/s41596-021-00534-0</a></div></div>
</div>
<div id="ref-pEJYKNFc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline"><strong>Data-Driven Phenotypic Dissection of AML Reveals Progenitor-like Cells that Correlate with Prognosis</strong> <div class="csl-block">Jacob H Levine, Erin F Simonds, Sean C Bendall, Kara L Davis, El-ad D Amir, Michelle D Tadmor, Oren Litvin, Harris G Fienberg, Astraea Jager, Eli R Zunder, … Garry P Nolan</div> <em>Cell</em> (2015-06-18) <a href="https://www.wikidata.org/wiki/Q30975629">https://www.wikidata.org/wiki/Q30975629</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2015.05.047">10.1016/j.cell.2015.05.047</a></div></div>
</div>
<div id="ref-JpCsfe9G" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline"><strong>Fast unfolding of communities in large networks</strong> <div class="csl-block">Vincent Blondel, Jean-Loup Guillaume, Renaud Lambiotte, Etienne Lefebvre</div> <em>Journal of Statistical Mechanics: Theory and Experiment</em> (2008-10-09) <a href="https://www.wikidata.org/wiki/Q29305711">https://www.wikidata.org/wiki/Q29305711</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1088/1742-5468/2008/10/p10008">10.1088/1742-5468/2008/10/p10008</a></div></div>
</div>
<div id="ref-1GezH4XNn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline"><strong>PanglaoDB: a web server for exploration of mouse and human single-cell RNA sequencing data</strong> <div class="csl-block">Oscar Franzén, Li-Ming Gan, Johan LM Bjorkegren</div> <em>Database</em> (2019-01-01) <a href="https://www.wikidata.org/wiki/Q63664483">https://www.wikidata.org/wiki/Q63664483</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/database/baz046">10.1093/database/baz046</a></div></div>
</div>
<div id="ref-chGii6yw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline"><strong>CellMarker: a manually curated resource of cell markers in human and mouse</strong> <div class="csl-block">Xinxin Zhang, Yujia Lan, Jinyuan Xu, Fei Quan, Erjie Zhao, Chunyu Deng, Tao Luo, Liwen Xu, Gaoming Liao, Min Yan, … Yun Xiao</div> <em>Nucleic Acids Research</em> (2019-01-01) <a href="https://www.wikidata.org/wiki/Q56984510">https://www.wikidata.org/wiki/Q56984510</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gky900">10.1093/nar/gky900</a></div></div>
</div>
<div id="ref-agnqfdk6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline"><strong>The Cell Ontology 2016: enhanced content, modularization, and ontology interoperability.</strong> <div class="csl-block">Alexander D Diehl, Terrence F Meehan, Yvonne M Bradford, Matthew H Brush, Wasila M Dahdul, David S Dougall, Yongqun He, David Osumi-Sutherland, Alan Ruttenberg, Sirarat Sarntivijai, … Christopher J Mungall</div> <em>Journal of Biomedical Semantics</em> (2016-07-04) <a href="https://www.wikidata.org/wiki/Q36067763">https://www.wikidata.org/wiki/Q36067763</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13326-016-0088-7">10.1186/s13326-016-0088-7</a></div></div>
</div>
<div id="ref-SozKx0lF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline"><strong>Cell type discovery using single-cell transcriptomics: implications for ontological representation</strong> <div class="csl-block">Brian D Aevermann, Mark Novotny, Trygve E Bakken, Jeremy A Miller, Alexander D Diehl, David Osumi-Sutherland, Roger S Lasken, Ed S Lein, Richard H Scheuermann</div> <em>Human Molecular Genetics</em> (2018-05-01) <a href="https://www.wikidata.org/wiki/Q52625486">https://www.wikidata.org/wiki/Q52625486</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/hmg/ddy100">10.1093/hmg/ddy100</a></div></div>
</div>
<div id="ref-15xnotFsw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline"><strong>Cell ontology in an age of data-driven cell classification.</strong> <div class="csl-block">David Osumi-Sutherland, David Osumi-Sutherland</div> <em>BMC Bioinformatics</em> (2017-12-21) <a href="https://www.wikidata.org/wiki/Q49192555">https://www.wikidata.org/wiki/Q49192555</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-017-1980-6">10.1186/s12859-017-1980-6</a></div></div>
</div>
<div id="ref-P1gDLuog" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline"><strong>Besca, a single-cell transcriptomics analysis toolkit to accelerate translational research</strong> <div class="csl-block">Sophia Clara Mädler, Alice Julien-Laferriere, Luis Wyss, Miroslav Phan, Albert SW Kang, Eric Ulrich, Roland Schmucki, Jitao David Zhang, Martin Ebeling, Laura Badi, … Klas Hatje</div> <em>bioRxiv</em> (2020-08-12) <a href="https://www.wikidata.org/wiki/Q104450593">https://www.wikidata.org/wiki/Q104450593</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2020.08.11.245795">10.1101/2020.08.11.245795</a></div></div>
</div>
<div id="ref-sW6aNZJB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline"><strong>Leveraging the Cell Ontology to classify unseen cell types</strong> <div class="csl-block">Sheng Wang, Angela Oliveira Pisco, Aaron McGeever, Maria Brbić, Marinka Žitnik, Spyros Darmanis, Jure Leskovec, Jim Karkanias, Russ Altman</div> <em>Nature Communications</em> (2021-09-21) <a href="https://www.wikidata.org/wiki/Q108929315">https://www.wikidata.org/wiki/Q108929315</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-021-25725-x">10.1038/s41467-021-25725-x</a></div></div>
</div>
<div id="ref-15YmDXALp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline"><strong>ontoProc: processing of ontologies of anatomy, cell lines, and so on</strong> <a href="https://www.wikidata.org/wiki/Q101074371">https://www.wikidata.org/wiki/Q101074371</a></div>
</div>
<div id="ref-tdlWsvTd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline"><strong>Tabula Muris</strong> <a href="https://tabula-muris.ds.czbiohub.org/">https://tabula-muris.ds.czbiohub.org/</a></div>
</div>
<div id="ref-FjUqVIOp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline"><strong>Tabula Sapiens</strong> <a href="https://tabula-sapiens-portal.ds.czbiohub.org/celltypes">https://tabula-sapiens-portal.ds.czbiohub.org/celltypes</a></div>
</div>
<div id="ref-T7RxnUNw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline"><strong>Azimuth</strong> <a href="https://azimuth.hubmapconsortium.org/">https://azimuth.hubmapconsortium.org/</a></div>
</div>
<div id="ref-1FghWzORJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline"><strong>Construction and Usage of a Human Body Common Coordinate Framework Comprising Clinical, Semantic, and Spatial Ontologies</strong> <div class="csl-block">Katy Börner, Ellen Quardokus, Bruce WHerr II, Leonard E Cross, Elizabeth G Record, Yingnan Ju, Andreas D Bueckle, James P Sluka, Jonathan C Silverstein, Kristen M Browne, … Griffin M Weber</div> (2020-07-28) <a href="https://www.wikidata.org/wiki/Q109755184">https://www.wikidata.org/wiki/Q109755184</a></div>
</div>
<div id="ref-yMavNVkS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline"><strong>Cell Annotation Platform | Coming Soon</strong> <a href="http://celltype.info/">http://celltype.info/</a></div>
</div>
<div id="ref-123DiCgc8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline"><strong>Cross-tissue immune cell analysis reveals tissue-specific adaptations and clonal architecture across the human body</strong> <div class="csl-block">Conde C Domínguez, Tomás Gomes, Lorna B Jarvis, C Xu, SK Howlett, DB Rainbow, Ondrej Suchanek, Hamish W King, Lira Mamanova, Krzysztof Polański, … Sarah Teichmann</div> (2021-04-28) <a href="https://www.wikidata.org/wiki/Q107363182">https://www.wikidata.org/wiki/Q107363182</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.04.28.441762">10.1101/2021.04.28.441762</a></div></div>
</div>
<div id="ref-UduMr7x2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline"><strong>Ontology based molecular signatures for immune cell types via gene expression analysis</strong> <div class="csl-block">Terrence F Meehan, Nicole Vasilevsky, Christopher J Mungall, David S Dougall, Melissa Haendel, Judith A Blake, Alexander D Diehl</div> <em>BMC Bioinformatics</em> (2013-08-30) <a href="https://www.wikidata.org/wiki/Q34978215">https://www.wikidata.org/wiki/Q34978215</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-14-263">10.1186/1471-2105-14-263</a></div></div>
</div>
<div id="ref-ecteO8vI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline"><strong>Logical development of the cell ontology</strong> <div class="csl-block">Terrence F Meehan, Anna Maria Masci, Amina Abdulla, Lindsay G Cowell, Judith A Blake, Christopher J Mungall, Alexander D Diehl</div> <em>BMC Bioinformatics</em> (2011-01-05) <a href="https://www.wikidata.org/wiki/Q33786317">https://www.wikidata.org/wiki/Q33786317</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-12-6">10.1186/1471-2105-12-6</a></div></div>
</div>
<div id="ref-DLTMbf54" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline"><strong>Ontologies for the life sciences</strong> <div class="csl-block">Steffen Schulze-Kremer, Barry Smith</div> (2005-11-15) <a href="https://www.wikidata.org/wiki/Q105870680">https://www.wikidata.org/wiki/Q105870680</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/047001153x.g408213">10.1002/047001153x.g408213</a></div></div>
</div>
<div id="ref-cZEmYsqx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline"><strong>The Philosophy of Logical Atomism, Lecture 1: Facts and Propositions</strong> <a href="https://www.wikidata.org/wiki/Q105105637">https://www.wikidata.org/wiki/Q105105637</a></div>
</div>
<div id="ref-to2onCDy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline"><strong>Logik der Forschung</strong> <div class="csl-block">Karl Popper</div> (1934-01-01) <a href="https://www.wikidata.org/wiki/Q1868040">https://www.wikidata.org/wiki/Q1868040</a></div>
</div>
<div id="ref-RghcuiSS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline"><strong>The semantic conception of truth: and the foundations of semantics</strong> <div class="csl-block">Alfred Tarski</div> <em>Philosophy and Phenomenological Research</em> (1944-03-01) <a href="https://www.wikidata.org/wiki/Q106090790">https://www.wikidata.org/wiki/Q106090790</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2307/2102968">10.2307/2102968</a></div></div>
</div>
<div id="ref-35incsRu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline"><strong>Applied Ontology - An Introduction</strong> <a href="https://www.wikidata.org/wiki/Q110015932">https://www.wikidata.org/wiki/Q110015932</a></div>
</div>
<div id="ref-18wkLcGxT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline"><strong>The Gene Ontology resource: enriching a GOld mine</strong> <div class="csl-block">Gene Ontology Consortium</div> <em>Nucleic Acids Research</em> (2020-12-08) <a href="https://www.wikidata.org/wiki/Q104130127">https://www.wikidata.org/wiki/Q104130127</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkaa1113">10.1093/nar/gkaa1113</a></div></div>
</div>
<div id="ref-2KmDs8gy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">72. </div><div class="csl-right-inline"><strong>Gene ontology: tool for the unification of biology. The Gene Ontology Consortium</strong> <div class="csl-block">M Ashburner, CA Ball, Judith A Blake, David Botstein, H Butler, JMichael Cherry, AP Davis, K Dolinski, Selina S Dwight, JT Eppig, … Gavin Sherlock</div> <em>Nature Genetics</em> (2000-05-01) <a href="https://www.wikidata.org/wiki/Q23781406">https://www.wikidata.org/wiki/Q23781406</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/75556">10.1038/75556</a></div></div>
</div>
<div id="ref-j3tJyG7J" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline"><strong>The OBO Foundry: coordinated evolution of ontologies to support biomedical data integration</strong> <div class="csl-block">Barry Smith, Michael Ashburner, Cornelius Rosse, Jonathan Bard, William Bug, Werner Ceusters, Louis J Goldberg, Karen Eilbeck, Amelia Ireland, Christopher J Mungall, … Suzanna Lewis</div> <em>Nature Biotechnology</em> (2007-11-01) <a href="https://www.wikidata.org/wiki/Q19671692">https://www.wikidata.org/wiki/Q19671692</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nbt1346">10.1038/nbt1346</a></div></div>
</div>
<div id="ref-djVboz3w" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline"><strong>RDF 1.1 Concepts and Abstract Syntax</strong> <a href="https://www.w3.org/TR/rdf11-concepts/">https://www.w3.org/TR/rdf11-concepts/</a></div>
</div>
<div id="ref-16mCkZC1r" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline"><strong>Introducing the Knowledge Graph: things, not strings</strong> <div class="csl-block">Google</div> (2012-05-16) <a href="https://blog.google/products/search/introducing-knowledge-graph-things-not/">https://blog.google/products/search/introducing-knowledge-graph-things-not/</a></div>
</div>
<div id="ref-qMozvNth" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">76. </div><div class="csl-right-inline"><strong>Toward an epistemology of Wikipedia</strong> <div class="csl-block">Don Fallis</div> <em>Journal of the Association for Information Science and Technology</em> (2008-08-01) <a href="https://www.wikidata.org/wiki/Q101955295">https://www.wikidata.org/wiki/Q101955295</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/asi.20870">10.1002/asi.20870</a></div></div>
</div>
<div id="ref-SGzjBOG0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">77. </div><div class="csl-right-inline"><strong>Wikidata:Statistics - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Wikidata:Statistics">https://www.wikidata.org/wiki/Wikidata:Statistics</a></div>
</div>
<div id="ref-xLpRePoh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">78. </div><div class="csl-right-inline"><strong>From Freebase to Wikidata: The Great Migration</strong> <div class="csl-block">Thomas Pellissier Tanon, Denny Vrandečić, Sebastian Schaffert, Thomas Steiner, Lydia Pintscher</div> <em>Proceedings of the 25th International Conference on World Wide Web</em> (2016-01-01) <a href="https://www.wikidata.org/wiki/Q24074986">https://www.wikidata.org/wiki/Q24074986</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1145/2872427.2874809">10.1145/2872427.2874809</a></div></div>
</div>
<div id="ref-Ea6hVYrD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">79. </div><div class="csl-right-inline"><strong>Wikibase/DataModel - MediaWiki</strong> <a href="https://www.mediawiki.org/wiki/Wikibase/DataModel">https://www.mediawiki.org/wiki/Wikibase/DataModel</a></div>
</div>
<div id="ref-17Yt10yrY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">80. </div><div class="csl-right-inline"><strong>Help:Data type - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Help:Data_type">https://www.wikidata.org/wiki/Help:Data_type</a></div>
</div>
<div id="ref-S1Lksw7i" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">81. </div><div class="csl-right-inline"><strong>Help:Multilingual - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Help:Multilingual">https://www.wikidata.org/wiki/Help:Multilingual</a></div>
</div>
<div id="ref-ZhP1mKyl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">82. </div><div class="csl-right-inline"><strong>RDF 1.1 Semantics</strong> <a href="https://www.w3.org/TR/rdf11-mt/">https://www.w3.org/TR/rdf11-mt/</a></div>
</div>
<div id="ref-4wSLyAPr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">83. </div><div class="csl-right-inline"><strong>Wikidata:Data access - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Wikidata:Data_access">https://www.wikidata.org/wiki/Wikidata:Data_access</a></div>
</div>
<div id="ref-1HghnypzG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">84. </div><div class="csl-right-inline"><strong>WikidataR package - RDocumentation</strong> <a href="https://www.rdocumentation.org/packages/WikidataR/versions/2.2.0">https://www.rdocumentation.org/packages/WikidataR/versions/2.2.0</a></div>
</div>
<div id="ref-VKWfrtR1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">85. </div><div class="csl-right-inline"><strong>wikidata2df: Utility package for easily turning a SPARQL query into a dataframe</strong> <div class="csl-block">João Vitor F Cavalcante</div> <a href="https://github.com/jvfe/wikidata2df">https://github.com/jvfe/wikidata2df</a></div>
</div>
<div id="ref-tQIJPG4k" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">86. </div><div class="csl-right-inline"><strong>Wikidata:Licensing - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Wikidata:Licensing">https://www.wikidata.org/wiki/Wikidata:Licensing</a></div>
</div>
<div id="ref-2wDsXBwd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">87. </div><div class="csl-right-inline"><a href="https://query.wikidata.org/">https://query.wikidata.org/</a></div>
</div>
<div id="ref-dsmUPCk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">88. </div><div class="csl-right-inline"><strong>Getting the Most out of Wikidata: Semantic Technology Usage in Wikipedia’s Knowledge Graph, Getting the Most Out of Wikidata: Semantic Technology Usage in Wikipedia’s Knowledge Graph</strong> <div class="csl-block">Stanislav Malyshev, Markus Krötzsch, Larry González, Julius Gonsior, Adrian Bielefeldt</div> <em>The Semantic Web – ISWC 2018: 17th International Semantic Web Conference, Monterey, CA, USA, October 8–12, 2018, Proceedings, Part II</em> (2018-01-01) <a href="https://www.wikidata.org/wiki/Q56010228">https://www.wikidata.org/wiki/Q56010228</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-3-030-00668-6_23">10.1007/978-3-030-00668-6_23</a></div></div>
</div>
<div id="ref-17P4zruFZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">89. </div><div class="csl-right-inline"><strong>Scholia</strong> <div class="csl-block">Scholia</div> <a href="https://scholia.toolforge.org/">https://scholia.toolforge.org/</a></div>
</div>
<div id="ref-guMp3q1V" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">90. </div><div class="csl-right-inline"><strong>SARS-CoV-2-Queries</strong> <div class="csl-block">SARS-CoV-2-Queries</div> <a href="https://egonw.github.io/SARS-CoV-2-Queries/">https://egonw.github.io/SARS-CoV-2-Queries/</a></div>
</div>
<div id="ref-17rRjL1Xj" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">91. </div><div class="csl-right-inline"><strong>Wikidata:Tools/OpenRefine - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Wikidata:Tools/OpenRefine">https://www.wikidata.org/wiki/Wikidata:Tools/OpenRefine</a></div>
</div>
<div id="ref-K6ggpwRg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">92. </div><div class="csl-right-inline"><strong>Help:QuickStatements - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Help:QuickStatements">https://www.wikidata.org/wiki/Help:QuickStatements</a></div>
</div>
<div id="ref-sSH04fxa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">93. </div><div class="csl-right-inline"><strong>Wikidata:Bots - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Wikidata:Bots">https://www.wikidata.org/wiki/Wikidata:Bots</a></div>
</div>
<div id="ref-5K5MfnVl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">94. </div><div class="csl-right-inline"><strong>Wikidata:Pywikibot - Python 3 Tutorial - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Wikidata:Pywikibot_-_Python_3_Tutorial">https://www.wikidata.org/wiki/Wikidata:Pywikibot_-_Python_3_Tutorial</a></div>
</div>
<div id="ref-qDI8I4IJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">95. </div><div class="csl-right-inline"><strong>GitHub - SuLab/WikidataIntegrator: A Wikidata Python module integrating the MediaWiki API and the Wikidata SPARQL endpoint</strong> <div class="csl-block">GitHub</div> <a href="https://github.com/SuLab/WikidataIntegrator">https://github.com/SuLab/WikidataIntegrator</a></div>
</div>
<div id="ref-3GqlN9Dk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">96. </div><div class="csl-right-inline"><strong>Wikidata as a knowledge graph for the life sciences</strong> <div class="csl-block">Andra Waagmeester, Gregory Stupp, Sebastian Burgstaller-Muehlbacher, Benjamin M Good, Malachi Griffith, Obi Griffith, Kristina Hanspers, Henning Hermjakob, Toby Hudson, Kevin Hybiske, … Andrew I Su</div> <em>eLife</em> (2020-03-17) <a href="https://www.wikidata.org/wiki/Q87830400">https://www.wikidata.org/wiki/Q87830400</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7554/elife.52614">10.7554/elife.52614</a></div></div>
</div>
<div id="ref-Ym2HoRLl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">97. </div><div class="csl-right-inline"><strong>Wikidata: A large-scale collaborative ontological medical database</strong> <div class="csl-block">Houcemeddine Turki, Thomas Shafee, Mohamed Ali Hadj Taieb, Mohamed Ben Aouicha, Denny Vrandečić, Diptanshu Das, Helmi Hamdi</div> <em>Journal of Biomedical Informatics</em> (2019-09-23) <a href="https://www.wikidata.org/wiki/Q68471881">https://www.wikidata.org/wiki/Q68471881</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.jbi.2019.103292">10.1016/j.jbi.2019.103292</a></div></div>
</div>
<div id="ref-179mXrGIk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">98. </div><div class="csl-right-inline"><strong>Big data: Wikiomics</strong> <div class="csl-block">Mitch Waldrop</div> <em>Nature</em> (2008-09-04) <a href="https://www.wikidata.org/wiki/Q28292893">https://www.wikidata.org/wiki/Q28292893</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/455022a">10.1038/455022a</a></div></div>
</div>
<div id="ref-TItj7EYf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">99. </div><div class="csl-right-inline"><strong>Calling on a million minds for community annotation in WikiProteins</strong> <div class="csl-block">Barend Mons, Michael Ashburner, Christine Chichester, Erik M van Mulligen, Marc Weeber, Johan den Dunnen, Gert-Jan van Ommen, Mark A Musen, Matt Cockerill, Henning Hermjakob, … Amos Bairoch</div> <em>Genome Biology</em> (2008-01-01) <a href="https://www.wikidata.org/wiki/Q21183907">https://www.wikidata.org/wiki/Q21183907</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/gb-2008-9-5-r89">10.1186/gb-2008-9-5-r89</a></div></div>
</div>
<div id="ref-P2QdZvGc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">100. </div><div class="csl-right-inline"><strong>Ten Simple Rules for Developing Public Biological Databases</strong> <div class="csl-block">Mohamed Helmy, Alexander Crits-Christoph, Gary D Bader</div> <em>PLOS Computational Biology</em> (2016-11-01) <a href="https://www.wikidata.org/wiki/Q28595967">https://www.wikidata.org/wiki/Q28595967</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1005128">10.1371/journal.pcbi.1005128</a></div></div>
</div>
<div id="ref-bO7BekzO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">101. </div><div class="csl-right-inline"><strong>Inside the Alexa-Friendly World of Wikidata</strong> <div class="csl-block">Tom Simonite</div> <em>Wired</em> <a href="https://www.wired.com/story/inside-the-alexa-friendly-world-of-wikidata/">https://www.wired.com/story/inside-the-alexa-friendly-world-of-wikidata/</a></div>
</div>
<div id="ref-18UXATsxP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">102. </div><div class="csl-right-inline"><strong>A gene wiki for community annotation of gene function</strong> <div class="csl-block">Jon W Huss, Camilo Orozco, James Goodale, Chunlei Wu, Serge Batalov, Tim J Vickers, Faramarz Valafar, Andrew I Su</div> <em>PLOS Biology</em> (2008-07-08) <a href="https://www.wikidata.org/wiki/Q21092744">https://www.wikidata.org/wiki/Q21092744</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pbio.0060175">10.1371/journal.pbio.0060175</a></div></div>
</div>
<div id="ref-13LWXQF0E" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">103. </div><div class="csl-right-inline"><strong>Making your database available through Wikipedia: the pros and cons</strong> <div class="csl-block">Robert D Finn, Paul P Gardner, Alex Bateman</div> <em>Nucleic Acids Research</em> (2012-01-01) <a href="https://www.wikidata.org/wiki/Q28254676">https://www.wikidata.org/wiki/Q28254676</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkr1195">10.1093/nar/gkr1195</a></div></div>
</div>
<div id="ref-2ZhxC0dg" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">104. </div><div class="csl-right-inline"><strong>Wikidata as a semantic framework for the Gene Wiki initiative</strong> <div class="csl-block">Sebastian Burgstaller-Muehlbacher, Andra Waagmeester, Elvira Mitraka, Julia Turner, Timothy Elliott Putman, Justin Leong, Chinmay Naik, Paul Pavlidis, Lynn Schriml, Benjamin M Good, Andrew I Su</div> <em>Database</em> (2016-01-01) <a href="https://www.wikidata.org/wiki/Q23712646">https://www.wikidata.org/wiki/Q23712646</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/database/baw015">10.1093/database/baw015</a></div></div>
</div>
<div id="ref-DaJdrXLk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">105. </div><div class="csl-right-inline"><strong>WikiGenomes: an open web application for community consumption and curation of gene annotation data in Wikidata</strong> <div class="csl-block">Timothy Elliott Putman, Sebastien Lelong, Sebastian Burgstaller-Muehlbacher, Andra Waagmeester, Colin Diesh, Nathan Dunn, Monica Munoz-Torres, Gregory Stupp, Chunlei Wu, Andrew I Su, Benjamin M Good</div> <em>Database</em> (2017-03-08) <a href="https://www.wikidata.org/wiki/Q28529449">https://www.wikidata.org/wiki/Q28529449</a></div>
</div>
<div id="ref-aQfGxPRH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">106. </div><div class="csl-right-inline"><strong>ChlamBase: a curated model organism database for the Chlamydia research community</strong> <div class="csl-block">Timothy Elliott Putman, Kevin Hybiske, Derek Jow, Cyrus Afrasiabi, Sebastien Lelong, Marco Alvarado Cano, Chunlei Wu, Andrew I Su</div> <em>Database</em> (2019-01-01) <a href="https://www.wikidata.org/wiki/Q63286185">https://www.wikidata.org/wiki/Q63286185</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/database/baz041">10.1093/database/baz041</a></div></div>
</div>
<div id="ref-1GijvCQtv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">107. </div><div class="csl-right-inline"><strong>Submit a Topic Page to PLOS Computational Biology and Wikipedia</strong> <div class="csl-block">Daniel Mietchen, Shoshana Wodak, Szymon Wasik, Natalia Szostak, Christophe Dessimoz</div> <em>PLOS Computational Biology</em> (2018-05-31) <a href="https://www.wikidata.org/wiki/Q54655231">https://www.wikidata.org/wiki/Q54655231</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1006137">10.1371/journal.pcbi.1006137</a></div></div>
</div>
<div id="ref-hxzL9pmm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">108. </div><div class="csl-right-inline"><strong>Scholia, Scientometrics and Wikidata</strong> <div class="csl-block">Finn Årup Nielsen, Daniel Mietchen, Egon Willighagen</div> <em>The Semantic Web: ESWC 2017 Satellite Events</em> (2017-10-01) <a href="https://www.wikidata.org/wiki/Q41799194">https://www.wikidata.org/wiki/Q41799194</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-3-319-70407-4_36">10.1007/978-3-319-70407-4_36</a></div></div>
</div>
<div id="ref-PdG2G4Bp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">109. </div><div class="csl-right-inline"><strong>Robustifying Scholia: paving the way for knowledge discovery and research assessment through Wikidata</strong> <div class="csl-block">Lane Rasberry, Egon Willighagen, Finn Årup Nielsen, Daniel Mietchen</div> <em>Research Ideas and Outcomes</em> (2019-05-02) <a href="https://www.wikidata.org/wiki/Q63433973">https://www.wikidata.org/wiki/Q63433973</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3897/rio.5.e35820">10.3897/rio.5.e35820</a></div></div>
</div>
<div id="ref-P1EnNa7Q" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">110. </div><div class="csl-right-inline"><strong>The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata</strong> <div class="csl-block">Adriano Rutz, Maria Sorokina, Jakub Galgonek, Daniel Mietchen, Egon Willighagen, Arnaud Gaudry, James G Graham, Ralf Stephan, Roderic Page, Jiří Vondrášek, … Pierre-Marie Allard</div> <em>Bioinformatics</em> (2021-03-01) <a href="https://doi.org/gh6dqk">https://doi.org/gh6dqk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.02.28.433265">10.1101/2021.02.28.433265</a></div></div>
</div>
<div id="ref-gqY9YdmH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">111. </div><div class="csl-right-inline"><strong>GeneDB and Wikidata</strong> <div class="csl-block">Magnus Manske, Ulrike Böhme, Christoph Püthe, Matt Berriman</div> <em>Wellcome Open Research</em> (2019-10-14) <a href="https://doi.org/gnq75m">https://doi.org/gnq75m</a> <div class="csl-block">DOI: <a href="https://doi.org/10.12688/wellcomeopenres.15355.2">10.12688/wellcomeopenres.15355.2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31723674">31723674</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6823904">PMC6823904</a></div></div>
</div>
<div id="ref-HUr1iigG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">112. </div><div class="csl-right-inline"><strong>The Cellosaurus, a Cell-Line Knowledge Resource</strong> <div class="csl-block">Amos Bairoch</div> <em>Journal of Biomolecular Techniques : JBT</em> (2018-07) <a href="https://doi.org/gf75bq">https://doi.org/gf75bq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7171/jbt.18-2902-002">10.7171/jbt.18-2902-002</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29805321">29805321</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5945021">PMC5945021</a></div></div>
</div>
<div id="ref-k4BJEMHE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">113. </div><div class="csl-right-inline"><strong>Complex Portal 2022: new curation frontiers</strong> <div class="csl-block">Birgit HM Meldal, Livia Perfetto, Colin Combe, Tiago Lubiana, João Vitor Ferreira Cavalcante, Hema Bye-A-Jee, Andra Waagmeester, Noemi del-Toro, Anjali Shrivastava, Elisabeth Barrera, … Sandra Orchard</div> <em>Nucleic Acids Research</em> (2021-10-29) <a href="https://doi.org/gnq75j">https://doi.org/gnq75j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkab991">10.1093/nar/gkab991</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34718729">34718729</a></div></div>
</div>
<div id="ref-pzgOjFLZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">114. </div><div class="csl-right-inline"><strong>WikiPathways: connecting communities</strong> <div class="csl-block">Marvin Martens, Ammar Ammar, Anders Riutta, Andra Waagmeester, Denise N Slenter, Kristina Hanspers, Ryan A. Miller, Daniela Digles, Elisson N Lopes, Friederike Ehrhart, … Martina Kutmon</div> <em>Nucleic Acids Research</em> (2021-01-08) <a href="https://doi.org/gh6dq2">https://doi.org/gh6dq2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkaa1024">10.1093/nar/gkaa1024</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33211851">33211851</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7779061">PMC7779061</a></div></div>
</div>
<div id="ref-1Fqp7ppqH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">115. </div><div class="csl-right-inline"><strong>Complex Portal 2018: extended content and enhanced visualization tools for macromolecular complexes</strong> <div class="csl-block">Birgit H M Meldal, Hema Bye-A-Jee, Lukáš Gajdoš, Zuzana Hammerová, Aneta Horáčková, Filip Melicher, Livia Perfetto, Daniel Pokorný, Milagros Rodriguez Lopez, Alžběta Türková, … Sandra Orchard</div> <em>Nucleic Acids Research</em> (2019-01-08) <a href="https://doi.org/gnq75k">https://doi.org/gnq75k</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gky1001">10.1093/nar/gky1001</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30357405">30357405</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6323931">PMC6323931</a></div></div>
</div>
<div id="ref-HUodcdz1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">116. </div><div class="csl-right-inline"><strong>Wikidata as a knowledge graph for the life sciences</strong> <div class="csl-block">Andra Waagmeester, Gregory Stupp, Sebastian Burgstaller-Muehlbacher, Benjamin M Good, Malachi Griffith, Obi L Griffith, Kristina Hanspers, Henning Hermjakob, Toby S Hudson, Kevin Hybiske, … Andrew I Su</div> <em>eLife</em> (2020-03-17) <a href="https://doi.org/ggqqc6">https://doi.org/ggqqc6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7554/elife.52614">10.7554/elife.52614</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32180547">32180547</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7077981">PMC7077981</a></div></div>
</div>
<div id="ref-zH3KDqhp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">117. </div><div class="csl-right-inline"><strong>Representing COVID-19 information in collaborative knowledge graphs: The case of Wikidata</strong> <div class="csl-block">Houcemeddine Turki, Mohamed Ali Hadj Taieb, Thomas Shafee, Tiago Lubiana, Dariusz Jemielniak, Mohamed Ben Aouicha, José Emilio Labra Gayo, Eric Youngstrom, Mossab Banat, Diptanshu Das, … WikiProject COVID-19</div> <em>Semantic Web: Interoperability, Usability, Applicability</em> (2021-09-28) <a href="https://www.wikidata.org/wiki/Q108766311">https://www.wikidata.org/wiki/Q108766311</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3233/sw-210444">10.3233/sw-210444</a></div></div>
</div>
<div id="ref-mPoPwN77" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">118. </div><div class="csl-right-inline"><strong>A protocol for adding knowledge to Wikidata: aligning resources on human coronaviruses</strong> <div class="csl-block">Andra Waagmeester, Egon Willighagen, Andrew I Su, Martina Summer-Kutmon, José Emilio Labra Gayo, Daniel Fernández-Álvarez, Quentin J Groom, Peter J Schaap, Lisa M Verhagen, Jasper Koehorst</div> <em>BMC Biology</em> (2021-01-22) <a href="https://www.wikidata.org/wiki/Q105037759">https://www.wikidata.org/wiki/Q105037759</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12915-020-00940-y">10.1186/s12915-020-00940-y</a></div></div>
</div>
<div id="ref-3YCzr3jU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">119. </div><div class="csl-right-inline"><strong>Wikidata Queries around the SARS-CoV-2 virus and pandemic</strong> <a href="https://www.wikidata.org/wiki/Q88647643">https://www.wikidata.org/wiki/Q88647643</a></div>
</div>
<div id="ref-SkOeuFRd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">120. </div><div class="csl-right-inline"><strong>COVIWD: COVID-19 Wikidata Dashboard</strong> <div class="csl-block">Fariz Darari</div> <em>Jurnal Ilmu Komputer dan Informasi</em> (2021-03-01) <a href="https://www.wikidata.org/wiki/Q105833381">https://www.wikidata.org/wiki/Q105833381</a> <div class="csl-block">DOI: <a href="https://doi.org/10.21609/jiki.v14i1.941">10.21609/jiki.v14i1.941</a></div></div>
</div>
<div id="ref-AOVCR8mW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">121. </div><div class="csl-right-inline"><strong>Painel de informação sobre a COVID-19: consultas SPARQL na Wikidata</strong> <div class="csl-block">Ana Carolina Simionato Arakaki, Fabiano Ferreira de Castro, Felipe Augusto Arakaki</div> <em>AtoZ: Novas Práticas em Informação e Conhecimento</em> (2020-12-03) <a href="https://www.wikidata.org/wiki/Q106249454">https://www.wikidata.org/wiki/Q106249454</a> <div class="csl-block">DOI: <a href="https://doi.org/10.5380/atoz.v9i2.76684">10.5380/atoz.v9i2.76684</a></div></div>
</div>
<div id="ref-19wiU1LCA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">122. </div><div class="csl-right-inline"><strong>Uso de Wikidata y Wikipedia para la generación asistida de un vocabulario estructurado multilingüe sobre la pandemia de Covid-19</strong> <div class="csl-block">Tomás Saorín, Juan-Antonio Pastor-Sánchez, María-José Baños-Moreno</div> <em>Profesional de la Informacion</em> (2020-09-13) <a href="https://www.wikidata.org/wiki/Q107377131">https://www.wikidata.org/wiki/Q107377131</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3145/epi.2020.sep.09">10.3145/epi.2020.sep.09</a></div></div>
</div>
<div id="ref-10EWNsZCz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">123. </div><div class="csl-right-inline"><strong>The Bgee suite: integrated curated expression atlas and comparative transcriptomics in animals</strong> <div class="csl-block">Frederic B Bastian, Julien Roux, Anne Niknejad, Aurélie Comte, Sara SFonseca Costa, Tarcisio M Farias, Sébastien Moretti, Gilles Parmentier, Valentine Rech de Laval, Marta Rosikiewicz, … Marc Robinson-Rechavi</div> <em>Nucleic Acids Research</em> (2020-10-10) <a href="https://www.wikidata.org/wiki/Q100513179">https://www.wikidata.org/wiki/Q100513179</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkaa793">10.1093/nar/gkaa793</a></div></div>
</div>
<div id="ref-864Pcm4C" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">124. </div><div class="csl-right-inline"><strong>Utilizing the Wikidata system to improve the quality of medical content in Wikipedia in diverse languages: a pilot study</strong> <div class="csl-block">Alexander Pfundner, Tobias Schönberg, John Horn, Richard David Boyce, Matthias Samwald</div> <em>Journal of Medical Internet Research</em> (2015-05-05) <a href="https://www.wikidata.org/wiki/Q21503276">https://www.wikidata.org/wiki/Q21503276</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2196/jmir.4163">10.2196/jmir.4163</a></div></div>
</div>
<div id="ref-11UV98PHT" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">125. </div><div class="csl-right-inline"><strong>Pesquisa-ação: uma introdução metodológica</strong> <div class="csl-block">David Tripp</div> <em>Educação e Pesquisa</em> (2005-12-01) <a href="https://www.wikidata.org/wiki/Q108479295">https://www.wikidata.org/wiki/Q108479295</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1590/s1517-97022005000300009">10.1590/s1517-97022005000300009</a></div></div>
</div>
<div id="ref-CHsAoEvr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">126. </div><div class="csl-right-inline"><strong>protégé</strong> <a href="https://protege.stanford.edu/">https://protege.stanford.edu/</a></div>
</div>
<div id="ref-uCE45Jh8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">127. </div><div class="csl-right-inline"><strong>Towards a pragmatic definition of cell type</strong> <div class="csl-block">Tiago Lubiana, Helder I Nakaya</div> <em>Preprints</em> (2021-01-04) <a href="https://doi.org/ghrxwf">https://doi.org/ghrxwf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.22541/au.160979530.02627436/v1">10.22541/au.160979530.02627436/v1</a></div></div>
</div>
<div id="ref-AHGRV6KQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">128. </div><div class="csl-right-inline"><strong>PhyloCode</strong> <a href="https://www.wikidata.org/wiki/Q1189395">https://www.wikidata.org/wiki/Q1189395</a></div>
</div>
<div id="ref-baJRsCMu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">129. </div><div class="csl-right-inline"><strong>Highlights of the 'gene nomenclature across species' meeting.</strong> <div class="csl-block">Elspeth A Bruford</div> <em>Human genomics</em> (2010-02-01) <a href="https://www.wikidata.org/wiki/Q42790699">https://www.wikidata.org/wiki/Q42790699</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1479-7364-4-3-213">10.1186/1479-7364-4-3-213</a></div></div>
</div>
<div id="ref-5ZFRyTy5" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">130. </div><div class="csl-right-inline"><strong>Ontological realism: A methodology for coordinated evolution of scientific ontologies</strong> <div class="csl-block">Barry Smith, Werner Ceusters</div> <em>Applied Ontology</em> (2010-11-15) <a href="https://www.wikidata.org/wiki/Q28239464">https://www.wikidata.org/wiki/Q28239464</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3233/ao-2010-0079">10.3233/ao-2010-0079</a></div></div>
</div>
<div id="ref-eP47FILn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">131. </div><div class="csl-right-inline"><strong>Multi-level ontology-based conceptual modeling</strong> <div class="csl-block">Victorio A Carvalho, João Paulo A Almeida, Claudenir M Fonseca, Giancarlo Guizzardi</div> <em>Data and Knowledge Engineering</em> (2017-05-01) <a href="https://www.wikidata.org/wiki/Q108926456">https://www.wikidata.org/wiki/Q108926456</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.datak.2017.03.002">10.1016/j.datak.2017.03.002</a></div></div>
</div>
<div id="ref-pFijpXkl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">132. </div><div class="csl-right-inline"><strong>The Human Cell Atlas: Technical approaches and challenges.</strong> <div class="csl-block">Chung Chau Hon, Jay W Shin, Piero Carninci, Michael JT Stubbington</div> <em>Briefings in functional genomics</em> (2017-10-28) <a href="https://www.wikidata.org/wiki/Q48563763">https://www.wikidata.org/wiki/Q48563763</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bfgp/elx029">10.1093/bfgp/elx029</a></div></div>
</div>
<div id="ref-1C6LI68h6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">133. </div><div class="csl-right-inline"><strong>Popper on Definitions</strong> <div class="csl-block">Wilhelm Büttemeyer</div> <em>Zeitschrift fur allgemeine Wissenschaftstheorie. Journal for general philosophy of science</em> (2005-01-01) <a href="https://www.wikidata.org/wiki/Q108925548">https://www.wikidata.org/wiki/Q108925548</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s10838-005-6037-2">10.1007/s10838-005-6037-2</a></div></div>
</div>
<div id="ref-M0lqRYJb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">134. </div><div class="csl-right-inline"><strong>PanglaoDB - A Single Cell Sequencing Resource For Gene Expression Data</strong> <a href="https://panglaodb.se/index.html">https://panglaodb.se/index.html</a></div>
</div>
<div id="ref-T3PlZ3Vh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">135. </div><div class="csl-right-inline"><strong>PanglaoDB: a web server for exploration of mouse and human single-cell RNA sequencing data</strong> <div class="csl-block">Oscar Franzén, Li-Ming Gan, Johan LM Björkegren</div> <em>Database</em> (2019-01-01) <a href="https://doi.org/ggkzxr">https://doi.org/ggkzxr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/database/baz046">10.1093/database/baz046</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30951143">30951143</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6450036">PMC6450036</a></div></div>
</div>
<div id="ref-OGXYtGZ8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">136. </div><div class="csl-right-inline"><strong>Linked Data - Design Issues</strong> <a href="https://www.w3.org/DesignIssues/LinkedData.html">https://www.w3.org/DesignIssues/LinkedData.html</a></div>
</div>
<div id="ref-4AEy2xhQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">137. </div><div class="csl-right-inline"><strong>CellFinder: a cell data repository</strong> <div class="csl-block">Harald Stachelscheid, Stefanie Seltmann, Fritz Lekschas, Jean-Fred Fontaine, Nancy Mah, Mariana Lara Neves, Miguel A Andrade-Navarro, Ulf Leser, Andreas Kurtz</div> <em>Nucleic Acids Research</em> (2013-12-03) <a href="https://www.wikidata.org/wiki/Q28660708">https://www.wikidata.org/wiki/Q28660708</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkt1264">10.1093/nar/gkt1264</a></div></div>
</div>
<div id="ref-imSZTi52" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">138. </div><div class="csl-right-inline"><strong>Type or Individual? Evidence of Large-Scale Conceptual Disarray in Wikidata</strong> <div class="csl-block">Atílio A Dadalto, João Paulo A Almeida, Claudenir M Fonseca, Giancarlo Guizzardi</div> <em>Lecture Notes in Computer Science</em> (2021-01-01) <a href="https://www.wikidata.org/wiki/Q109990743">https://www.wikidata.org/wiki/Q109990743</a></div>
</div>
<div id="ref-RLHsA1U8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">139. </div><div class="csl-right-inline"><strong>CELDA -- an ontology for the comprehensive representation of cells in complex systems</strong> <div class="csl-block">Stefanie Seltmann, Harald Stachelscheid, Alexander Damaschun, Ludger Jansen, Fritz Lekschas, Jean-Fred Fontaine, Throng Nghia Nguyen-Dobinsky, Ulf Leser, Andreas Kurtz</div> <em>BMC Bioinformatics</em> (2013-07-17) <a href="https://www.wikidata.org/wiki/Q21284308">https://www.wikidata.org/wiki/Q21284308</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/1471-2105-14-228">10.1186/1471-2105-14-228</a></div></div>
</div>
<div id="ref-HUJ9NzAb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">140. </div><div class="csl-right-inline"><strong>UniProt</strong> <a href="https://sparql.uniprot.org/sparql">https://sparql.uniprot.org/sparql</a></div>
</div>
<div id="ref-AgOOCR24" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">141. </div><div class="csl-right-inline"><strong>Portal:Semantic Web - WikiPathways</strong> <a href="https://www.wikipathways.org/index.php/Portal:Semantic_Web">https://www.wikipathways.org/index.php/Portal:Semantic_Web</a></div>
</div>
<div id="ref-rhRRCtlA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">142. </div><div class="csl-right-inline"><strong>Cell Markers</strong> <div class="csl-block">Konstantin Yakimchuk</div> <em>Materials and Methods</em> (2013-05-02) <a href="https://doi.org/ghq494">https://doi.org/ghq494</a> <div class="csl-block">DOI: <a href="https://doi.org/10.13070/mm.en.3.183">10.13070/mm.en.3.183</a></div></div>
</div>
<div id="ref-6uWWsiSq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">143. </div><div class="csl-right-inline"><strong>SHOGoiN: Shogoin Human Omics database for the Generation of iPS and Normal cells</strong> <a href="https://stemcellinformatics.org/">https://stemcellinformatics.org/</a></div>
</div>
<div id="ref-SdbIYKUa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">144. </div><div class="csl-right-inline"><strong>Comparison of reference management software - Wikipedia</strong> <a href="https://en.wikipedia.org/wiki/Comparison_of_reference_management_software">https://en.wikipedia.org/wiki/Comparison_of_reference_management_software</a></div>
</div>
<div id="ref-uYuz0opI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">145. </div><div class="csl-right-inline"><strong>Wikipedia, the free encyclopedia</strong> <a href="https://en.wikipedia.org/wiki/Main_Page">https://en.wikipedia.org/wiki/Main_Page</a></div>
</div>
<div id="ref-KNGv5MPp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">146. </div><div class="csl-right-inline"><strong>Clean Coder Blog</strong> <a href="https://blog.cleancoder.com/uncle-bob/2017/12/18/Excuses.html">https://blog.cleancoder.com/uncle-bob/2017/12/18/Excuses.html</a></div>
</div>
<div id="ref-rPKBwmYh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">147. </div><div class="csl-right-inline"><strong>Como fazer um fichamento</strong> <div class="csl-block">Priscilla de Carvalho Nunes disse</div> <em>Blog da Biblioteca da ECA-USP</em> (2019-09-30) <a href="https://bibliotecadaeca.wordpress.com/2019/09/30/como-fazer-um-fichamento/">https://bibliotecadaeca.wordpress.com/2019/09/30/como-fazer-um-fichamento/</a></div>
</div>
<div id="ref-1HBVPtZGp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">148. </div><div class="csl-right-inline"><strong>Come si fa una tesi di laurea</strong> <a href="https://www.wikidata.org/wiki/Q3684178">https://www.wikidata.org/wiki/Q3684178</a></div>
</div>
<div id="ref-15luL9zZC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">149. </div><div class="csl-right-inline"><strong>Unpaywall</strong> <a href="https://unpaywall.org/">https://unpaywall.org/</a></div>
</div>
<div id="ref-13HqB23xH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">150. </div><div class="csl-right-inline"><strong>Clean Code: A Handbook of Agile Software Craftsmanship</strong> <a href="https://www.wikidata.org/wiki/Q109996684">https://www.wikidata.org/wiki/Q109996684</a></div>
</div>
<div id="ref-1A9RvszKC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">151. </div><div class="csl-right-inline"><strong>Wikidata:Tools/Author Disambiguator - Wikidata</strong> <a href="https://www.wikidata.org/wiki/Wikidata:Tools/Author_Disambiguator">https://www.wikidata.org/wiki/Wikidata:Tools/Author_Disambiguator</a></div>
</div>
<div id="ref-6chnW6cc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">152. </div><div class="csl-right-inline"><strong>wbib: A helper for building Wikidata-based literature dashboards via SPARQL queries.</strong> <div class="csl-block">Tiago Lubiana</div> <a href="https://github.com/lubianat/wbib">https://github.com/lubianat/wbib</a></div>
</div>
<div id="ref-1hag8XE6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">153. </div><div class="csl-right-inline"><strong>HCA Latin America - 2021 Workshop</strong> <a href="https://www.humancellatlas.org/hca-latin-america-2021-workshop/">https://www.humancellatlas.org/hca-latin-america-2021-workshop/</a></div>
</div>
<div id="ref-kHL3NVxk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">154. </div><div class="csl-right-inline"><strong>BioHackathon Europe</strong> <a href="https://biohackathon-europe.org/">https://biohackathon-europe.org/</a></div>
</div>
<div id="ref-JgiKEEdq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">155. </div><div class="csl-right-inline"><strong>The Whelming › Tech, tools, and tribulations</strong> <div class="csl-block">Scott Allan Wallick</div> <a href="http://magnusmanske.de/wordpress/">http://magnusmanske.de/wordpress/</a></div>
</div>
<div id="ref-MxIeSJYt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">156. </div><div class="csl-right-inline"><strong>fcoex: FCBF-based Co-Expression Networks for Single Cells</strong> <div class="csl-block">Tiago Lubiana, Helder Nakaya</div> <em>Bioconductor version: Release (3.14)</em> (2022) <a href="https://bioconductor.org/packages/fcoex/">https://bioconductor.org/packages/fcoex/</a></div>
</div>
<div id="ref-CQRJ53gu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">157. </div><div class="csl-right-inline"><strong>Complex Portal 2022: new curation frontiers</strong> <div class="csl-block">Birgit HM Meldal, Livia Perfetto, Colin Combe, Tiago Lubiana, João Vitor Ferreira Cavalcante, Hema Bye-A-Jee, Andra Waagmeester, Noemi del-Toro, Anjali Shrivastava, Elisabeth Barrera, … Sandra Orchard</div> <em>Nucleic Acids Research</em> (2021-10-29) <a href="https://www.wikidata.org/wiki/Q109348309">https://www.wikidata.org/wiki/Q109348309</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkab991">10.1093/nar/gkab991</a></div></div>
</div>
<div id="ref-1DguATd9G" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">158. </div><div class="csl-right-inline"><strong>The Cellosaurus, a cell-line knowledge resource.</strong> <div class="csl-block">Amos Bairoch</div> <em>Journal of Biomolecular Techniques</em> (2018-05-01) <a href="https://www.wikidata.org/wiki/Q54370168">https://www.wikidata.org/wiki/Q54370168</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7171/jbt.18-2902-002">10.7171/jbt.18-2902-002</a></div></div>
</div>
<div id="ref-lMQxhx3q" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">159. </div><div class="csl-right-inline"><strong>User:CellosaurusBot - Wikidata</strong> <a href="https://www.wikidata.org/wiki/User:CellosaurusBot">https://www.wikidata.org/wiki/User:CellosaurusBot</a></div>
</div>
<div id="ref-F2mYjDJ0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">160. </div><div class="csl-right-inline"><strong>The Brazilian Reproducibility Initiative</strong> <div class="csl-block">Ana P Wasilewska-Sampaio, Olavo Bohrer Amaral, Kleber Neves, Ana P Wasilewska-Sampaio, Clarissa FD Carneiro, Olavo Bohrer Amaral, Clarissa FD Carneiro</div> <em>eLife</em> (2019-02-05) <a href="https://www.wikidata.org/wiki/Q61799268">https://www.wikidata.org/wiki/Q61799268</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7554/elife.41602">10.7554/elife.41602</a></div></div>
</div>
<div id="ref-oXMFSv5v" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">161. </div><div class="csl-right-inline"><strong>Sharing intermediate datasets from systematic reviews</strong> <div class="csl-block">Tiago Lubiana, Olavo Bohrer Amaral, Kleber Neves</div> <em>MetaArXiv</em> (2021-11-26) <a href="https://doi.org/gnrf9v">https://doi.org/gnrf9v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.31222/osf.io/vbwa9">10.31222/osf.io/vbwa9</a></div></div>
</div>
<div id="ref-gdYsBE7d" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">162. </div><div class="csl-right-inline"><strong>Wisecube AI | Knowledge Graph Engine</strong> <a href="https://www.wisecube.ai/">https://www.wisecube.ai/</a></div>
</div>
<div id="ref-rkXotO9x" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">163. </div><div class="csl-right-inline"><strong>An overview of the BIOASQ large-scale biomedical semantic indexing and question answering competition</strong> <div class="csl-block">George Tsatsaronis, Georgios Balikas, Prodromos Malakasiotis, Ioannis Partalas, Matthias Zschunke, Michael R Alvers, Dirk Weissenborn, Anastasia Krithara, Sergios Petridis, Dimitris Polychronopoulos, … Georgios Paliouras</div> <em>BMC Bioinformatics</em> (2015-04-30) <a href="https://www.wikidata.org/wiki/Q28646342">https://www.wikidata.org/wiki/Q28646342</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12859-015-0564-6">10.1186/s12859-015-0564-6</a></div></div>
</div>
<div id="ref-12LOzmXRs" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">164. </div><div class="csl-right-inline"><strong>No Budget Science Hack Week</strong> <div class="csl-block">reprodutibilidade</div> <a href="https://www.reprodutibilidade.bio.br/hack-week-2021">https://www.reprodutibilidade.bio.br/hack-week-2021</a></div>
</div>
<div id="ref-IJG65hFm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">165. </div><div class="csl-right-inline"><strong>Biocuration - Wikipedia</strong> <a href="https://en.wikipedia.org/wiki/Biocuration">https://en.wikipedia.org/wiki/Biocuration</a></div>
</div>
<div id="ref-14Wi842eZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">166. </div><div class="csl-right-inline"><strong>biohackathon-projects-2021/projects/32 at main · elixir-europe/biohackathon-projects-2021</strong> <div class="csl-block">GitHub</div> <a href="https://github.com/elixir-europe/biohackathon-projects-2021">https://github.com/elixir-europe/biohackathon-projects-2021</a></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen())
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script>
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
